---
title: "Metagenomics of Parkinson’s disease implicates the gut microbiome in multiple disease mechanisms"
author: "Zachary D Wallen, Ayse Demirkan, Guy Twa, Gwendolyn Cohen, Marissa N Dean, David G Standaert, Timothy Sampson, Haydeh Payami"
output: html_notebook
---
<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>

This R markdown notebook documents code used to perform the analyses described in the manuscript ***Metagenomics of Parkinson’s disease implicates the gut microbiome in multiple disease mechanisms***.

Setting up the R environment to perform analyses:

```{r setup, echo=T, collapse=T}
##### SET OPTIONS ####
knitr::opts_chunk$set(dev="png", dev.args=list(type="cairo-png"), warning=FALSE, message=FALSE, fig.show="hold", digits=2)
options(digits=14, width=100)

##### CREATE FUNCTIONS NEEDED FOR RUNNING CODE #####
# Function to supress warnings and messages
# parameters:
#     x - the command to supress messages/warnings of
suppress <- function(x){invisible(capture.output(suppressMessages(suppressWarnings(x))))}

# Function to perform ANCOM-BC and perform additional actions like grabbing summary stats
# and calculating bias-corrected abundances
# parameters:
#            ps - phyloseq object with otu_table() and sample_data() data
#       formula - formula for model to be tested by ANCOM-BC. make sure categorical variables
#                 have been dummy coded with 1 (for test group) and 0 (reference group) or
#                 else the N calculations will not work correctly.
#   remaining parameters see ANCOM-BC documentation
ANCOMBC.plus <- function(ps, formula, p_adj_method="BH", zero_cut=0.9, group=NULL,
                        lib_cut=0, struc_zero=F, neg_lb=F, tol=1E-5, max_iter=100,
                        conserve=F, alpha=0.05, global=F){
  library(phyloseq)
  library(ANCOMBC)
  ci <- function(coef, se){
    lower.ci <- coef - 1.96*se
    upper.ci <- coef + 1.96*se
    return(c(lower.ci=lower.ci,upper.ci=upper.ci))
  }
  # make sure samples are rows and features are columns
  if (taxa_are_rows(ps)){
    ps <- phyloseq(t(otu_table(ps)), sample_data(ps))
  }
  # run ANCOM-BC
  suppressWarnings(
  ancom.res <- ancombc(ps, formula=formula, p_adj_method=p_adj_method, zero_cut=zero_cut, lib_cut=lib_cut,
                       group=group, struc_zero=struc_zero, neg_lb=neg_lb, tol=tol, max_iter=max_iter,
                       conserve=conserve, alpha=alpha, global=global)
  )
  # calculate bias-corrected abundances
  samp_frac <- ancom.res$samp_frac
  samp_frac[is.na(samp_frac)] <- 0
  ps.adj <- prune_taxa(rownames(ancom.res$feature_table), ps)
  otu_table(ps.adj) <- log(otu_table(ps.adj) + 1) - samp_frac
  # filter for samples with data for variables included in model
  for (var in 1:ncol(ancom.res$res$beta)){
    ps <- phyloseq(otu_table(ps), sample_data(ps)[!is.na(sample_data(ps)[,strsplit(formula, " \\+ ")[[1]][var]]),])
    ps.adj <- phyloseq(otu_table(ps.adj), sample_data(ps.adj)[!is.na(sample_data(ps.adj)[,strsplit(formula, " \\+ ")[[1]][var]]),])
  }
  # get summary of results
  res <- data.frame()
  for (var in 1:ncol(ancom.res$res$beta)){
    # get variable name
    var.name1 <- strsplit(formula, " \\+ ")[[1]][var]
    var.name2 <- colnames(ancom.res$res$beta)[var]
    # get N in each group
    if (length(table(sample_data(ps)[,var.name1])) == 2){
      group.1.index <- sample_data(ps)[,var.name1] == names(table(sample_data(ps)[,var.name1]))[2]
      group.1.index[is.na(group.1.index)] <- FALSE
      group.2.index <- sample_data(ps)[,var.name1] == names(table(sample_data(ps)[,var.name1]))[1]
      group.2.index[is.na(group.2.index)] <- FALSE
      n1 <- colSums(otu_table(ps)[group.1.index,] > 0)
      n2 <- colSums(otu_table(ps)[group.2.index,] > 0)
      mean1 <- exp(colMeans(otu_table(ps.adj)[group.1.index,]))
      mean2 <- exp(colMeans(otu_table(ps.adj)[group.2.index,]))
    }else{
      n1 <- rep(nsamples(ps), ntaxa(ps))
      names(n1) <- taxa_names(ps)
      n2 <- rep(NA, ntaxa(ps))
      names(n2) <- taxa_names(ps)
      mean1 <- exp(colMeans(otu_table(ps.adj)))
      mean2 <- rep(NA, ntaxa(ps.adj))
      names(mean2) <- taxa_names(ps.adj)
    }
    # calculate fold change and confidence interval of fold change
    if(length(table(sample_data(ps)[,var.name1])) == 2){
      FC <- exp(ancom.res$res$beta[,var.name2])
      FC.lower <- c()
      FC.upper <- c()
      for (coef in 1:length(ancom.res$res$beta[,var.name2])){
        FC.lower <- c(FC.lower, exp(ci(ancom.res$res$beta[,var.name2][coef],
                                       ancom.res$res$se[,var.name2][coef])['lower.ci']))
        FC.upper <- c(FC.upper, exp(ci(ancom.res$res$beta[,var.name2][coef],
                                       ancom.res$res$se[,var.name2][coef])['upper.ci']))
      }
    }else{
      FC <- NA
      FC.lower <- NA
      FC.upper <- NA
    }
    # summarize results for variable
    rvar <- data.frame(Variable=var.name1,
                      Feature=rownames(ancom.res$feature_table),
                      N1=n1[rownames(ancom.res$feature_table)],
                      N2=n2[rownames(ancom.res$feature_table)],
                      Mean1=mean1[rownames(ancom.res$feature_table)],
                      Mean2=mean2[rownames(ancom.res$feature_table)],
                      Beta=ancom.res$res$beta[,var.name2],
                      SE=ancom.res$res$se[,var.name2],
                      P=ancom.res$res$p_val[,var.name2],
                      FDR=ancom.res$res$q_val[,var.name2],
                      FC=FC, FC_lower=FC.lower, FC_upper=FC.upper,
                      check.names = F)
    res <- rbind(res, rvar[order(rvar$P),])
    # add untested features if they exist
    if (nrow(rvar) != ntaxa(ps)){
      res <- rbind(res,
             data.frame(Variable=var.name1,
                        Feature=taxa_names(ps)[!(taxa_names(ps) %in% rownames(ancom.res$feature_table))],
                        N1=n1[taxa_names(ps)[!(taxa_names(ps) %in% rownames(ancom.res$feature_table))]],
                        N2=n2[taxa_names(ps)[!(taxa_names(ps) %in% rownames(ancom.res$feature_table))]],
                        Mean1=NA, Mean2=NA, Beta=NA, SE=NA, P=NA, FDR=NA, FC=NA, FC_lower=NA, FC_upper=NA,
                        check.names = F))
    }
  }
  return(list(result.summary=res, ancom.output=ancom.res, bias.corrected.ps=ps.adj))
}

# Function to perform MaAsLin on phyloseq object and perform additional actions like grabbing summary stats
# parameters:
#            ps - phyloseq object with otu_table() and sample_data() data
#        output - path to directory for output
#      metadata - vector listing variable names in sample_data() to include in the model
# remaining parameters see MaAsLin documentation
MaAsLin2.plus <- function(ps, output, metadata, min_abundance=0, min_prevalence=0.1, min_variance=0,
                          normalization='TSS', transform='LOG', analysis_method='LM', max_significance=0.05,
                          random_effects=NULL, fixed_effects=NULL, correction='BH', standardize=T, cores=1,
                          plot_heatmap=T, plot_scatter=T, heatmap_first_n=50, reference=NULL){
  library(phyloseq)
  library(Maaslin2)
  ci <- function(coef, se){
    lower.ci <- coef - 1.96*se
    upper.ci <- coef + 1.96*se
    return(c(lower.ci=lower.ci,upper.ci=upper.ci))
  }
  # make sure samples are rows and features are columns
  if (taxa_are_rows(ps)){
    ps <- phyloseq(t(otu_table(ps)), sample_data(ps))
  }
  # make sure only LOG is chosen for MaAsLin transformation
  if (!(transform == "LOG")){stop('function only supports LOG transformation at this time')}
  # run MaAsLin2
  input_data <- data.frame(otu_table(ps))
  input_metadata <- data.frame(sample_data(ps)[,colnames(sample_data(ps)) %in% metadata])
  fits <- Maaslin2(input_data, input_metadata, output, min_abundance, min_prevalence, min_variance,
                   normalization, transform, analysis_method, max_significance,
                   random_effects, fixed_effects, correction, standardize, cores,
                   plot_heatmap, plot_scatter, heatmap_first_n, reference)
  # put back original feature names
  for (feat in 1:length(fits$results$feature)){
    fits$results$feature[feat] <- taxa_names(ps)[make.names(taxa_names(ps)) == fits$results$feature[feat]]
  }
  # filter for samples with data for variables included in model
  for (var in 1:length(unique(fits$results$metadata))){
    ps <- phyloseq(otu_table(ps), sample_data(ps)[!is.na(sample_data(ps)[,metadata[var]]),])
  }
  # get summary of results
  res <- data.frame()
  for (var in 1:length(unique(fits$results$metadata))){
    # get variable name
    var.name <- metadata[var]
    # get N in each group
    if (length(table(sample_data(ps)[,var.name])) == 2){
      group.1.index <- sample_data(ps)[,var.name] == names(table(sample_data(ps)[,var.name]))[2]
      group.1.index[is.na(group.1.index)] <- FALSE
      group.2.index <- sample_data(ps)[,var.name] == names(table(sample_data(ps)[,var.name]))[1]
      group.2.index[is.na(group.2.index)] <- FALSE
      n1 <- colSums(otu_table(ps)[group.1.index,] > 0)
      n2 <- colSums(otu_table(ps)[group.2.index,] > 0)
      mean1 <- colMeans(otu_table(ps)[group.1.index,])
      mean2 <- colMeans(otu_table(ps)[group.2.index,])
    }else{
      n1 <- rep(sum(table(sample_data(ps)[,var.name])), ntaxa(ps))
      names(n1) <- taxa_names(ps)
      n2 <- rep(NA, ntaxa(ps))
      names(n2) <- taxa_names(ps)
      mean1 <- colMeans(otu_table(ps))
      mean2 <- rep(NA, ntaxa(ps))
      names(mean2) <- taxa_names(ps)
    }
    # calculate fold change and confidence interval of fold change
    if(length(table(sample_data(ps)[,var.name])) == 2){
      FC <- 2^(fits$results$coef[fits$results$metadata == var.name])
      FC.lower <- c()
      FC.upper <- c()
      for (coef in 1:length(fits$results$coef[fits$results$metadata == var.name])){
        FC.lower <- c(FC.lower, 2^(ci(fits$results$coef[fits$results$metadata == var.name][coef],
                                      fits$results$stderr[fits$results$metadata == var.name][coef])['lower.ci']))
        FC.upper <- c(FC.upper, 2^(ci(fits$results$coef[fits$results$metadata == var.name][coef],
                                      fits$results$stderr[fits$results$metadata == var.name][coef])['upper.ci']))
      }
    }else{
      FC <- NA
      FC.lower <- NA
      FC.upper <- NA
    }
    # summarize results for variable
    rvar <- data.frame(Variable=var.name,
                      Feature=fits$results$feature[fits$results$metadata == var.name],
                      N1=n1[fits$results$feature[fits$results$metadata == var.name]],
                      N2=n2[fits$results$feature[fits$results$metadata == var.name]],
                      Mean1=mean1[fits$results$feature[fits$results$metadata == var.name]],
                      Mean2=mean2[fits$results$feature[fits$results$metadata == var.name]],
                      Beta=fits$results$coef[fits$results$metadata == var.name],
                      SE=fits$results$stderr[fits$results$metadata == var.name],
                      P=fits$results$pval[fits$results$metadata == var.name],
                      FDR=p.adjust(fits$results$pval[fits$results$metadata == var.name], method=correction),
                      FC=FC, FC_lower=FC.lower, FC_upper=FC.upper,
                      check.names = F)
    res <- rbind(res, rvar[order(rvar$P),])
    # add untested features if they exist
    if (nrow(rvar) != ntaxa(ps)){
      res <- rbind(res,
             data.frame(Variable=var.name,
                        Feature=taxa_names(ps)[!(taxa_names(ps) %in% fits$results$feature[fits$results$metadata == var.name])],
                        N1=n1[taxa_names(ps)[!(taxa_names(ps) %in% fits$results$feature[fits$results$metadata == var.name])]],
                        N2=n2[taxa_names(ps)[!(taxa_names(ps) %in% fits$results$feature[fits$results$metadata == var.name])]],
                        Mean1=mean1[taxa_names(ps)[!(taxa_names(ps) %in% fits$results$feature[fits$results$metadata == var.name])]],
                        Mean2=mean2[taxa_names(ps)[!(taxa_names(ps) %in% fits$results$feature[fits$results$metadata == var.name])]],
                        Beta=NA, SE=NA, P=NA, FDR=NA, FC=NA, FC_lower=NA, FC_upper=NA,
                        check.names = F)
                   )
    }
  }
return(list(result.summary=res, Maaslin2.output=fits))
}

# Function to compute log2 transform for transforming relative abundances
# parameters:
#    x - a data.frame of relative abundances
log2.trans <- function(x) {
  y <- replace(x, x == 0, min(x[x>0]) / 2)
  return(log2(y))
}

# Function to convert numbers from strings back to numbers in excel output
# parameters:
#       df - the data.frame being outputted into excel
#       wb - the excel workbook object
#    sheet - the excel sheet name
# colnames - does df have column names, TRUE/FALSE
convertNum <- function(df, wb, sheet, colnames) {
  library(foreach)
  cn <- expand.grid(seq_len(ncol(df)), seq_len(nrow(df)))[,1]
  rn <- expand.grid(seq_len(ncol(df)), seq_len(nrow(df)))[,2]
  trash <- foreach(cn = cn, rn = rn) %dopar% {
    if (!is.numeric(df[rn,cn]) && !is.na(as.numeric(as.character(df[rn,cn])))) {
      row.offset <- ifelse(colnames, 1, 0)
      openxlsx::writeData(wb, sheet, as.numeric(as.character(df[rn,cn])), startCol = cn, startRow = row.offset + rn)
    }
  }
}

##### CREATE EXCEL STYLES #####
bold <- openxlsx::createStyle(textDecoration="bold")
center <- openxlsx::createStyle(halign="center", valign="center", wrapText=T)
horizontal_border_med <- openxlsx::createStyle(border="top", borderStyle="medium")
horizontal_border_thin <- openxlsx::createStyle(border="top", borderStyle="thin")

##### LOAD REQUIRED PACKAGES #####
suppress(library(dplyr))
suppress(library(kableExtra))
suppress(library(ggplot2))
suppress(library(scales))
suppress(library(ggpubr))
suppress(library(ggtext))
suppress(library(ggh4x))
suppress(library(reshape2))
suppress(library(ggfortify))
suppress(library(gridExtra))
suppress(library(vegan))
suppress(library(pairwiseCI))
suppress(library(vcd))
suppress(library(readxl))
suppress(library(phyloseq))
suppress(library(car))
suppress(library(ANCOMBC))
suppress(library(Maaslin2))
suppress(library(tibble))
suppress(library(openxlsx))
suppress(library(foreach))
suppress(library(data.table))
suppress(library(igraph))
cat('*** R session information ***
\nR version -', paste(R.version$major, R.version$minor, sep = '.'),
'\n
R packages and versions:
\ndplyr -', paste(packageVersion("dplyr")),
'\nkableExtra -', paste(packageVersion("kableExtra")),
'\nggplot2 -', paste(packageVersion("ggplot2")),
'\nscales -', paste(packageVersion("scales")),
'\nggpubr -', paste(packageVersion("ggpubr")),
'\nggtext -', paste(packageVersion("ggtext")),
'\nggh4x -', paste(packageVersion("ggh4x")),
'\nreshape2 -', paste(packageVersion("reshape2")),
'\nggfortify -', paste(packageVersion("ggfortify")),
'\ngridExtra -', paste(packageVersion("gridExtra")),
'\nvegan -', paste(packageVersion("vegan")),
'\npairwiseCI -', paste(packageVersion("pairwiseCI")),
'\nvcd -', paste(packageVersion("vcd")),
'\nreadxl -', paste(packageVersion("readxl")),
'\nphyloseq -', paste(packageVersion("phyloseq")),
'\ncar -', paste(packageVersion("car")),
'\nANCOMBC -', paste(packageVersion("ANCOMBC")),
'\nMaaslin2 -', paste(packageVersion("Maaslin2")),
'\ntibble -', paste(packageVersion("tibble")),
'\nopenxlsx -', paste(packageVersion("openxlsx")),
'\nforeach -', paste(packageVersion("foreach")),
'\ndata.table -', paste(packageVersion("data.table")),
'\nigraph -', paste(packageVersion("igraph")), '\n')
```

### Processing shotgun sequences using HUMAnN/MetaPhlAn pipeline

Shotgun sequences were bioinformatically processed from raw sequences to taxonomic and functional (gene families and pathway) profiles using a pipeline that involved sequence QC and decontamination with BBTools and taxonomic/functional profiling with MetaPhlAn and HUMAnN (see GitHub repository https://github.com/zwallen/SLURM_Shotgun_Metagenomic_Pipeline). At the end of the pipeline, a sequence tracking report was created to see how many sequence reads passed each step of the pipeline for each sample. Below describes the steps of the pipeline in brief, along with code for generating summaries of the sequence number and proportions of sequences remaining after each quality control step.

```{r load_tracking_report, echo=T}
# prep directory for output
system('
if [ ! -d "output/0.Pre_analysis" ]
then
  mkdir output/0.Pre_analysis
fi
')
system('
if [ ! -d "output/0.Pre_analysis/Sequence_QC_report" ]
then
  mkdir output/0.Pre_analysis/Sequence_QC_report
fi
')

# read in pipeline tracking reports
pipe.report <- read.table('input/remaining_reads_per_step.txt',
                          stringsAsFactors=F, header=T)
```

1. Input raw sequences
    + See below for the distribution of the number of raw sequences for each sample that was sent for shotgun metagenomic sequencing (N = 492 PD, 242 NHC) with the mean (red line), minimum (left blue line), and maximum (right blue line):

```{r input_seqs, echo=T, fig.width=10, fig.height=5}
# plot distribution of number of input reads
mean.summ <- mean(pipe.report$Input)
mean.summ.2 <- mean(log(pipe.report$Input))
min.summ <- min(pipe.report$Input)
min.summ.2 <- min(log(pipe.report$Input))
max.summ <- max(pipe.report$Input)
max.summ.2 <- max(log(pipe.report$Input))

g <- ggplot() +
  geom_histogram(data=pipe.report, aes(x=log(Input)), binwidth = 0.05) +
  geom_vline(xintercept=mean.summ.2, color='red') +
  annotate(geom='text', x=mean.summ.2+0.15, y=52, label=formatC(mean.summ,format='e',digits=2), color='red', size=3) +
  geom_vline(xintercept=min.summ.2, color='blue') +
  annotate(geom='text', x=min.summ.2+0.15, y=52, label=formatC(min.summ,format='e',digits=2), color='blue', size=3) +
  geom_vline(xintercept=max.summ.2, color='blue') +
  annotate(geom='text', x=max.summ.2+0.15, y=52, label=formatC(max.summ,format='e',digits=2), color='blue', size=3) +
  scale_x_continuous(labels = trans_format("exp", format=function(x){formatC(x,format='e',digits=1)})) +
  labs(x='Number of input reads (log scaled)', y='Number of samples') +
  theme(axis.text.x=element_text(angle = 25, hjust = 1))
ggsave('output/0.Pre_analysis/Sequence_QC_report/1.Input_seq_N.pdf', g, device='pdf', width=10, height=5)

g
```

2. Adapter trimming and quality trimming/filtering of sequences using BBDuk
    + Adapter (and PhiX) sequences were removed and quality trimming/filtering of sequence reads was performed using BBDuk with the following parameters:
        + `ref=adapters,phix`: For removing common sequencing adapters and PhiX sequences.
        + `ftm=5`: For trimming sequences to a length that is multiple of 5 (helps in removing extra base if one exists (i.e. if length of reads is 151 bp instead of 150 bp)).
        + `tbo`: In addition to usual kmer adapter trimming, specifies to also trim adapters based on pair overlap detection using BBMerge.
        + `tpe`: Specifies to make sure and trim both reads to the same length.
        + `qtrim=rl`: Quality trim both 5' and 3' end of sequences.
        + `trimq=25` : Quality score to trim up to on sequence ends.
        + `minlen=50`: Filter out sequences that fall below 50 bp in length.
    + The distributions of the mean proportion and number of sequence reads passing adapter trimming and quality trimming/filtering are below:

```{r quality_control, echo=T, fig.width=10, fig.height=5}
# calculated proportion of reads remaining
pipe.report$prop.remaining <- pipe.report$Quality_controlled/pipe.report$Input

# plot distribution of proportion of reads remaining
mean.summ <- mean(pipe.report$prop.remaining)
min.summ <- min(pipe.report$prop.remaining)
max.summ <- max(pipe.report$prop.remaining)

g1 <- ggplot() +
  geom_histogram(data=pipe.report, aes(x=prop.remaining), binwidth = 0.005) +
  geom_vline(xintercept=mean.summ, color='red') +
  annotate(geom='text', x=mean.summ-0.02, y=60, label=paste(round(mean.summ*100,1),'%',sep=''), color='red', size=3) +
  geom_vline(xintercept=min.summ, color='blue') +
  annotate(geom='text', x=min.summ-0.02, y=60, label=paste(round(min.summ*100,1),'%',sep=''), color='blue', size=3) +
  geom_vline(xintercept=max.summ, color='blue') +
  annotate(geom='text', x=max.summ-0.02, y=60, label=paste(round(max.summ*100,1),'%',sep=''), color='blue', size=3) +
  labs(x='Proportion of reads remaining', y='Number of samples')
ggsave('output/0.Pre_analysis/Sequence_QC_report/2.Quality_controlled_seqs_prop.pdf', g1, device='pdf', width=10, height=5)

# plot distribution of number of reads remaining
mean.summ <- mean(pipe.report$Quality_controlled)
mean.summ.2 <- mean(log(pipe.report$Quality_controlled))
min.summ <- min(pipe.report$Quality_controlled)
min.summ.2 <- min(log(pipe.report$Quality_controlled))
max.summ <- max(pipe.report$Quality_controlled)
max.summ.2 <- max(log(pipe.report$Quality_controlled))

g2 <- ggplot() +
  geom_histogram(data=pipe.report, aes(x=log(Quality_controlled)), binwidth = 0.05) +
  geom_vline(xintercept=mean.summ.2, color='red') +
  annotate(geom='text', x=mean.summ.2+0.15, y=52, label=formatC(mean.summ,format='e',digits=2), color='red', size=3) +
  geom_vline(xintercept=min.summ.2, color='blue') +
  annotate(geom='text', x=min.summ.2+0.15, y=52, label=formatC(min.summ,format='e',digits=2), color='blue', size=3) +
  geom_vline(xintercept=max.summ.2, color='blue') +
  annotate(geom='text', x=max.summ.2+0.15, y=52, label=formatC(max.summ,format='e',digits=2), color='blue', size=3) +
  scale_x_continuous(labels = trans_format("exp", format=function(x){formatC(x,format='e',digits=1)})) +
  labs(x='Number of reads remaining (log scaled)', y='Number of samples') +
  theme(axis.text.x=element_text(angle = 25, hjust = 1))
ggsave('output/0.Pre_analysis/Sequence_QC_report/2.Quality_controlled_seqs_N.pdf', g2, device='pdf', width=10, height=5)

g1; g2
```

3. Removal of human host sequence reads using BBSplit/BBMap
    + Human sequence reads were removed from each sequence file by aligning reads to the most recent human genome reference using BBSplit/BBMap.
    + The most recent human genome reference (`GCA_000001405.28_GRCh38.p13_genomic.fna`) was downloaded from the [NCBI FTP site](https://ftp.ncbi.nlm.nih.gov/genomes/genbank/vertebrate_mammalian/Homo_sapiens/reference/GCA_000001405.28_GRCh38.p13/GCA_000001405.28_GRCh38.p13_genomic.fna.gz).
    + BBSplit was ran with default parameters for both indexing the human genome reference file and for mapping sequences.
    + The distributions of the mean proportion and number of quality controlled sequence reads passing decontamination are below:

```{r decontamination, echo=T, fig.width=10, fig.height=5}
# calculated proportion of reads remaining
pipe.report$prop.remaining <- pipe.report$Decontaminated/pipe.report$Quality_controlled

# plot distribution of proportion of reads remaining
mean.summ <- mean(pipe.report$prop.remaining)
min.summ <- min(pipe.report$prop.remaining)
max.summ <- max(pipe.report$prop.remaining)

g1 <- ggplot() +
  geom_histogram(data=pipe.report, aes(x=prop.remaining), binwidth = 0.01) +
  geom_vline(xintercept=mean.summ, color='red') +
  annotate(geom='text', x=mean.summ-0.02, y=600, label=paste(round(mean.summ*100,1),'%',sep=''), color='red', size=3) +
  geom_vline(xintercept=min.summ, color='blue') +
  annotate(geom='text', x=min.summ-0.02, y=600, label=paste(round(min.summ*100,1),'%',sep=''), color='blue', size=3) +
  geom_vline(xintercept=max.summ, color='blue') +
  annotate(geom='text', x=max.summ+0.02, y=600, label=paste(round(max.summ*100,1),'%',sep=''), color='blue', size=3) +
  labs(x='Proportion of reads remaining', y='Number of samples')
ggsave('output/0.Pre_analysis/Sequence_QC_report/3.Decontaminated_seqs_prop.pdf', g1, device='pdf', width=10, height=5)

# plot distribution of number of reads remaining
mean.summ <- mean(pipe.report$Decontaminated)
mean.summ.2 <- mean(log(pipe.report$Decontaminated))
min.summ <- min(pipe.report$Decontaminated)
min.summ.2 <- min(log(pipe.report$Decontaminated))
max.summ <- max(pipe.report$Decontaminated)
max.summ.2 <- max(log(pipe.report$Decontaminated))

g2 <- ggplot() +
  geom_histogram(data=pipe.report, aes(x=log(Decontaminated)), binwidth = 0.05) +
  geom_vline(xintercept=mean.summ.2, color='red') +
  annotate(geom='text', x=mean.summ.2+0.15, y=52, label=formatC(mean.summ,format='e',digits=2), color='red', size=3) +
  geom_vline(xintercept=min.summ.2, color='blue') +
  annotate(geom='text', x=min.summ.2+0.15, y=52, label=formatC(min.summ,format='e',digits=2), color='blue', size=3) +
  geom_vline(xintercept=max.summ.2, color='blue') +
  annotate(geom='text', x=max.summ.2+0.15, y=52, label=formatC(max.summ,format='e',digits=2), color='blue', size=3) +
  scale_x_continuous(labels = trans_format("exp", format=function(x){formatC(x,format='e',digits=1)})) +
  labs(x='Number of reads remaining (log scaled)', y='Number of samples') +
  theme(axis.text.x=element_text(angle = 25, hjust = 1))
ggsave('output/0.Pre_analysis/Sequence_QC_report/3.Decontaminated_seqs_N.pdf', g2, device='pdf', width=10, height=5)

g1; g2
```

4. Removal of low complexity sequences using BBDuk
    + To remove low complexity sequences such as mononucleotide repeats, BBDuk entropy filtering was ran with following parameters:
        + `entropy=0.01`: Entropy threshold for removing low complexity sequences. This value suggested by BBDuk author to remove only monomeric repeats (http://seqanswers.com/forums/showthread.php?t=42776&page=7).
        + `entropywindow=50`: Calculate entropy using a sliding window of this length. Value is the default.
        + `entropyk=5`: Calculate entropy using kmers of this length. Value is the default.
    + The distributions of the mean proportion and number of sequence reads passing entropy filtering are below:

```{r entropy_filtering, echo=T, fig.width=10, fig.height=5}
# calculated proportion of reads remaining
pipe.report$prop.remaining <- pipe.report$Entropy_filtered/pipe.report$Decontaminated

# plot distribution of proportion of reads remaining
mean.summ <- mean(pipe.report$prop.remaining)
min.summ <- min(pipe.report$prop.remaining)
max.summ <- max(pipe.report$prop.remaining)

g1 <- ggplot() +
  geom_histogram(data=pipe.report, aes(x=prop.remaining), binwidth = 0.01) +
  geom_vline(xintercept=mean.summ, color='red') +
  annotate(geom='text', x=mean.summ-0.007, y=600, label=paste(round(mean.summ*100,1),'%',sep=''), color='red', size=3) +
  geom_vline(xintercept=min.summ, color='blue') +
  annotate(geom='text', x=min.summ-0.001, y=600, label=paste(round(min.summ*100,1),'%',sep=''), color='blue', size=3) +
  geom_vline(xintercept=max.summ, color='blue') +
  annotate(geom='text', x=max.summ+0.007, y=600, label=paste(round(max.summ*100,1),'%',sep=''), color='blue', size=3) +
  labs(x='Proportion of reads remaining', y='Number of samples')
ggsave('output/0.Pre_analysis/Sequence_QC_report/4.Low_complexity_filtered_seqs_prop.pdf', g1, device='pdf', width=10, height=5)

# plot distribution of number of reads remaining
mean.summ <- mean(pipe.report$Entropy_filtered)
mean.summ.2 <- mean(log(pipe.report$Entropy_filtered))
min.summ <- min(pipe.report$Entropy_filtered)
min.summ.2 <- min(log(pipe.report$Entropy_filtered))
max.summ <- max(pipe.report$Entropy_filtered)
max.summ.2 <- max(log(pipe.report$Entropy_filtered))

g2 <- ggplot() +
  geom_histogram(data=pipe.report, aes(x=log(Entropy_filtered)), binwidth = 0.05) +
  geom_vline(xintercept=mean.summ.2, color='red') +
  annotate(geom='text', x=mean.summ.2+0.15, y=52, label=formatC(mean.summ,format='e',digits=2), color='red', size=3) +
  geom_vline(xintercept=min.summ.2, color='blue') +
  annotate(geom='text', x=min.summ.2+0.15, y=52, label=formatC(min.summ,format='e',digits=2), color='blue', size=3) +
  geom_vline(xintercept=max.summ.2, color='blue') +
  annotate(geom='text', x=max.summ.2+0.15, y=52, label=formatC(max.summ,format='e',digits=2), color='blue', size=3) +
  scale_x_continuous(labels = trans_format("exp", format=function(x){formatC(x,format='e',digits=1)})) +
  labs(x='Number of reads remaining (log scaled)', y='Number of samples') +
  theme(axis.text.x=element_text(angle = 25, hjust = 1))
ggsave('output/0.Pre_analysis/Sequence_QC_report/4.Low_complexity_filtered_seqs_N.pdf', g2, device='pdf', width=10, height=5)

g1; g2
```

5. Tracking sequence read loss through the pipeline
    + Plotted proportion and number of sequence reads remaining from the starting input for each sample at each step of the pipeline in order to see big picture how samples performed through the pipeline.

```{r pipeline_tracking, echo=T, fig.width=10, fig.height=5}
# calculate proportion of reads remaining at each step from the original read count
prop.remaining <- with(pipe.report, data.frame(Sample=Sample,
                                               Input=1,
                                               Quality_controlled=Quality_controlled/Input,
                                               Decontaminated=Decontaminated/Input,
                                               Entropy_filtered=Entropy_filtered/Input))

# melt data to ready for plotting
pipe.report.melt <- melt(pipe.report[,-ncol(pipe.report)])
prop.remaining.melt <- melt(prop.remaining)

# plot proportion of reads remaining
g1 <- ggplot(data=prop.remaining.melt, aes(x=variable, y=value)) +
  geom_line(aes(group=Sample), alpha=0.3, color='darkgrey') +
  stat_boxplot(geom='errorbar', width=0.25, position=position_dodge(0.75)) +
  geom_boxplot(notch=T, outlier.size=-1) +
  geom_point(alpha=0.3) +
  stat_summary(geom='point', fun.data='mean_se', color='red') +
  theme(axis.text.x = element_text(angle = 25, hjust = 1)) +
  labs(x='Pipeline step', y='Proportion of input reads')
ggsave('output/0.Pre_analysis/Sequence_QC_report/5.Pipeline_seqs_prop.pdf', g1, device='pdf', width=10, height=5)

# plot number of reads remaining
g2 <- ggplot(data=pipe.report.melt, aes(x=variable, y=log(value))) +
  geom_line(aes(group=Sample), alpha=0.3, color='darkgrey') +
  stat_boxplot(geom='errorbar', width=0.25, position=position_dodge(0.75)) +
  geom_boxplot(notch=T, outlier.size=-1) +
  geom_point(alpha=0.3) +
  stat_summary(geom='point', fun.data='mean_se', color='red') +
  scale_y_continuous(labels = trans_format("exp", format=function(x){formatC(x,format='e',digits=1)})) +
  theme(axis.text.x = element_text(angle = 25, hjust = 1)) +
  labs(x='Pipeline step', y='Number of reads (log scaled)') +
  theme(axis.text.x=element_text(angle = 25, hjust = 1))
ggsave('output/0.Pre_analysis/Sequence_QC_report/5.Pipeline_seqs_N.pdf', g2, device='pdf', width=10, height=5)

g1; g2
```

7. Post quality control exclusions
    + Duplicate PD sample, N = 1
    + NHC samples whose subjects reported to have a neurological condition, N = 8
    + PD sample with high human DNA contamination and low sequence count (< 10M) after QC, N = 1.
    + This made the final set of quality controlled sequences range from 12M - 285M sequences per sample for 490 PD and 234 NHC.

6. Taxonomic and functional profiling using MetaPhlAn3 and HUMAnN3
    + Quality controlled, decontaminated, and low complexity filtered sequences were profiled for taxonomic and functional content using `MetaPhlAn v 3.0.14` that performs marker gene based taxonomic profiling to get relative abundances of species present in each sample, then using `HUMAnN v 3.0.0` to determine the microbial gene family and metabolic pathway content present in each sample using the UniRef and MetaCyc databases.
    + MetaPhlAn and HUMAnN were used with the most up to date ChocoPhlAn database (`full_chocophlan.v296_201901b`) and UniRef database with sequences grouped at 90% identity (UniRef90, `uniref90_annotated_v201901b_full`). These were supplied to the `--nucleotide-database` and `--protein-database` when running the `humann` command.
    + All options for MetaPhlAn and HUMAnN were left as default.
    + MetaPhlAn was ran a second time adding `--unknown_estimation`. This flag was enabled so the relative abundances of taxa would take the unknown content of a samples metagenome (portion of sequenced metagenome that is not contained in the MetaPhlAn database) into account (important for calculating count data).
    + When running of second run of MetaPhlAn was complete, estimated counts for each taxa were derived by multiplying the relative abundances by the total read count reported by bowtie2 intermediate file from running MetaPhlAn (formula: `(relative abundance / 100) x nread from bowtie2`).

### Creating MetaPhlAn sample profiles for each taxonomic level

To have all samples in one table and to have relative abundances and counts for species plus their higher taxomic levels broken up into separate tables for referencing, broke tables up into taxonomic levels. All relative abundance and count tables were compiled into an excel where each tab is a separate table (see files **0.Pre_analysis/MetaPhlAn_rel_ab_profiles.xlsx** and **0.Pre_analysis/MetaPhlan_count_profiles.xlsx**).
    + Note: these files were created for ease of referencing only, and should not be used in analysis as excel might round up large numbers and skew the actual relative abundances or count data.

```{r metaphlan_profiles, echo=T}
# prep directory for output
system('
if [ ! -d "output/0.Pre_analysis" ]
then
  mkdir output/0.Pre_analysis
fi
')
system('
if [ ! -d "output/0.Pre_analysis/MetaPhlAn_profiles" ]
then
  mkdir output/0.Pre_analysis/MetaPhlAn_profiles
fi
')

# read in tables
abun <- read.table('input/metaphlan_counts.tsv',
                   header=T, stringsAsFactors=F, sep='\t', check.names=F)

ra <- read.table('input/metaphlan_rel_ab.tsv',
                 header=T, stringsAsFactors=F, sep='\t', check.names=F)

# read in metadata
metadata <- data.frame(read_xlsx('input/subject_metadata.xlsx'))
rownames(metadata) <- metadata$UAMRMC_ID

# make table sample x species
rownames(abun) <- abun$clade_name
abun <- data.frame(t(abun[,-c(1,2)]), check.names=F)

rownames(ra) <- ra$clade_name
ra <- data.frame(t(ra[,-c(1,2)]), check.names=F)

# modify sample IDs to match metadata
for (i in 1:nrow(abun)){
  rownames(abun)[i] <- strsplit(rownames(abun)[i], '_')[[1]][1]
}
ids.to.mod <- rownames(abun)[grep("^S", rownames(abun))]
for (i in 1:length(ids.to.mod)){
  rownames(abun)[grep(ids.to.mod[i], rownames(abun))] <- metadata$UAMRMC_ID[grep(ids.to.mod[i], metadata$UAB_ID)]
}

for (i in 1:nrow(ra)){
  rownames(ra)[i] <- strsplit(rownames(ra)[i], '_')[[1]][1]
}
ids.to.mod <- rownames(ra)[grep("^S", rownames(ra))]
for (i in 1:length(ids.to.mod)){
  rownames(ra)[grep(ids.to.mod[i], rownames(ra))] <- metadata$UAMRMC_ID[grep(ids.to.mod[i], metadata$UAB_ID)]
}

# order by sample ids
abun <- abun[order(rownames(abun)),]

ra <- ra[order(rownames(ra)),]

# compile count data into phyloseq objects split up by taxonomic levels
abun.sub <- abun[,grep("s__|UNKNOWN", colnames(abun))]
abun.ps.s <- phyloseq(otu_table(as.matrix(abun.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(abun.sub)))))
abun.sub <- abun[,intersect(grep("s__", colnames(abun), invert=T), grep("g__|UNKNOWN", colnames(abun)))]
abun.ps.g <- phyloseq(otu_table(as.matrix(abun.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(abun.sub)))))
abun.sub <- abun[,intersect(grep("g__", colnames(abun), invert=T), grep("f__|UNKNOWN", colnames(abun)))]
abun.ps.f <- phyloseq(otu_table(as.matrix(abun.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(abun.sub)))))
abun.sub <- abun[,intersect(grep("f__", colnames(abun), invert=T), grep("o__|UNKNOWN", colnames(abun)))]
abun.ps.o <- phyloseq(otu_table(as.matrix(abun.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(abun.sub)))))
abun.sub <- abun[,intersect(grep("o__", colnames(abun), invert=T), grep("c__|UNKNOWN", colnames(abun)))]
abun.ps.c <- phyloseq(otu_table(as.matrix(abun.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(abun.sub)))))
abun.sub <- abun[,intersect(grep("c__", colnames(abun), invert=T), grep("p__|UNKNOWN", colnames(abun)))]
abun.ps.p <- phyloseq(otu_table(as.matrix(abun.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(abun.sub)))))
abun.sub <- abun[,grep("p__", colnames(abun), invert=T)]
abun.ps.k <- phyloseq(otu_table(as.matrix(abun.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(abun.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(abun.sub)))))

# compile relative abundance data into phyloseq objects split up by taxonomic levels
ra.sub <- ra[,grep("s__|UNKNOWN", colnames(ra))]
ra.ps.s <- phyloseq(otu_table(as.matrix(ra.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(ra.sub)))))
ra.sub <- ra[,intersect(grep("s__", colnames(ra), invert=T), grep("g__|UNKNOWN", colnames(ra)))]
ra.ps.g <- phyloseq(otu_table(as.matrix(ra.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(ra.sub)))))
ra.sub <- ra[,intersect(grep("g__", colnames(ra), invert=T), grep("f__|UNKNOWN", colnames(ra)))]
ra.ps.f <- phyloseq(otu_table(as.matrix(ra.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(ra.sub)))))
ra.sub <- ra[,intersect(grep("f__", colnames(ra), invert=T), grep("o__|UNKNOWN", colnames(ra)))]
ra.ps.o <- phyloseq(otu_table(as.matrix(ra.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(ra.sub)))))
ra.sub <- ra[,intersect(grep("o__", colnames(ra), invert=T), grep("c__|UNKNOWN", colnames(ra)))]
ra.ps.c <- phyloseq(otu_table(as.matrix(ra.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(ra.sub)))))
ra.sub <- ra[,intersect(grep("c__", colnames(ra), invert=T), grep("p__|UNKNOWN", colnames(ra)))]
ra.ps.p <- phyloseq(otu_table(as.matrix(ra.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(ra.sub)))))
ra.sub <- ra[,grep("p__", colnames(ra), invert=T)]
ra.ps.k <- phyloseq(otu_table(as.matrix(ra.sub), taxa_are_rows=F),
                      sample_data(metadata),
                      tax_table(as.matrix(
                      data.frame(Kingdom=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[1]}),
                                 Phylum=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[2]}),
                                 Class=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[3]}),
                                 Order=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[4]}),
                                 Family=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[5]}),
                                 Genus=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[6]}),
                                 Species=sapply(strsplit(colnames(ra.sub), "\\|"),function(x){x[7]}),
                                 check.names=F, row.names=colnames(ra.sub)))))

# create excels
wb <- createWorkbook()
addWorksheet(wb, 'Kingdom')
addWorksheet(wb, 'Phylum')
addWorksheet(wb, 'Class')
addWorksheet(wb, 'Order')
addWorksheet(wb, 'Family')
addWorksheet(wb, 'Genus')
addWorksheet(wb, 'Species')
writeData(wb, 'Kingdom', rownames_to_column(data.frame(otu_table(abun.ps.k),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Phylum', rownames_to_column(data.frame(otu_table(abun.ps.p),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Class', rownames_to_column(data.frame(otu_table(abun.ps.c),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Order', rownames_to_column(data.frame(otu_table(abun.ps.o),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Family', rownames_to_column(data.frame(otu_table(abun.ps.f),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Genus', rownames_to_column(data.frame(otu_table(abun.ps.g),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Species', rownames_to_column(data.frame(otu_table(abun.ps.s),check.names=F), "UAMRMC ID"),
          keepNA=T)
saveWorkbook(wb, 'output/0.Pre_analysis/MetaPhlAn_profiles/MetaPhlAn_count_profiles.xlsx',
             overwrite=T)

wb <- createWorkbook()
addWorksheet(wb, 'Kingdom')
addWorksheet(wb, 'Phylum')
addWorksheet(wb, 'Class')
addWorksheet(wb, 'Order')
addWorksheet(wb, 'Family')
addWorksheet(wb, 'Genus')
addWorksheet(wb, 'Species')
writeData(wb, 'Kingdom', rownames_to_column(data.frame(otu_table(ra.ps.k),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Phylum', rownames_to_column(data.frame(otu_table(ra.ps.p),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Class', rownames_to_column(data.frame(otu_table(ra.ps.c),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Order', rownames_to_column(data.frame(otu_table(ra.ps.o),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Family', rownames_to_column(data.frame(otu_table(ra.ps.f),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Genus', rownames_to_column(data.frame(otu_table(ra.ps.g),check.names=F), "UAMRMC ID"),
          keepNA=T)
writeData(wb, 'Species', rownames_to_column(data.frame(otu_table(ra.ps.s),check.names=F), "UAMRMC ID"),
          keepNA=T)
saveWorkbook(wb, 'output/0.Pre_analysis/MetaPhlAn_profiles/MetaPhlAn_rel_ab_profiles.xlsx',
             overwrite=T)
```

### Subject metdata variable differences between PD vs NHC

* To determine what subject metadata variables are significantly different between PD and NHC subjects, tested each variable for association with PD using Fisher's exact test (via `fisher.test` function in R) for categorical variables and Wilcoxon rank-sum test (via `wilcox.test` function in R) for quantitative variables. Odds ratios and confidence interals were calculated via the `fisher.test` function. If any 2x2 tables of categorical variables contained 0, then the function `Prop.or` from the `pairwiseCI` R package specifying `CImethod='Woolf'` was used to calculate odds ratios and confidence intervals. Results can be found in **1.Metadata/Subject_characteristics.xlsx** file. This file contains data showcased in **Table 1** of the manuscript.

```{r subject_data, echo=T}
# prep directory for output
system('
if [ ! -d "output/1.Metadata" ]
then
  mkdir output/1.Metadata
fi
')

# get metadata for PD and NHCs
subject.data <- metadata

# create result data.frame
results <- data.frame()

# samples passing QC
results <- rbind(results, data.frame(Category = '',
                                     Metadata = "Number of subjects who passed sequence and metadata QC",
                                     `PD N` = table(subject.data$Case_status)['PD'],
                                     `PD summary stats` = "-",
                                     `NHC N` = table(subject.data$Case_status)['Control'],
                                     `NHC summary stats` = "-",
                                     `Total N` = length(na.omit(subject.data$Case_status)),
                                     P = "-", `OR [95%CI]` = "-", check.names=F))
# age
P.t <- length(na.omit(subset(subject.data, Case_status == "PD")$Age_at_collection))
P.avg <- mean(na.omit(subset(subject.data, Case_status == "PD")$Age_at_collection))
P.sd <- sd(na.omit(subset(subject.data, Case_status == "PD")$Age_at_collection))
C.t <- length(na.omit(subset(subject.data, Case_status == "Control")$Age_at_collection))
C.avg <- mean(na.omit(subset(subject.data, Case_status == "Control")$Age_at_collection))
C.sd <- sd(na.omit(subset(subject.data, Case_status == "Control")$Age_at_collection))
p <- wilcox.test(subset(subject.data, Case_status == "PD")$Age_at_collection,
                 subset(subject.data, Case_status == "Control")$Age_at_collection)$p.value
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Age",
                                     `PD N` = P.t,
                                     `PD summary stats` = paste(round(P.avg, 1),round(P.sd, 1), sep = "±"),
                                     `NHC N` = C.t,
                                     `NHC summary stats` = paste(round(C.avg, 1),round(C.sd, 1), sep = "±"),
                                     `Total N` = P.t+C.t,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]` = "-",
                                     check.names=F))
# sex
P.f <- table(subset(subject.data, Case_status == "PD")$Sex)['F']
P.m <- table(subset(subject.data, Case_status == "PD")$Sex)['M']
C.f <- table(subset(subject.data, Case_status == "Control")$Sex)['F']
C.m <- table(subset(subject.data, Case_status == "Control")$Sex)['M']
p <- fisher.test(matrix(c(P.m,P.f,C.m,C.f), nrow = 2))$p.value
or <- fisher.test(matrix(c(P.m,P.f,C.m,C.f), nrow = 2))$estimate
ci <- paste(round(fisher.test(matrix(c(P.m,P.f,C.m,C.f), nrow = 2))$conf.int[1],1),
            round(fisher.test(matrix(c(P.m,P.f,C.m,C.f), nrow = 2))$conf.int[2],1), sep='-')
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Sex (N & % male)",
                                     `PD N` = P.f+P.m,
                                     `PD summary stats` = paste(P.m," ","(",round(P.m/(P.f+P.m)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.f+C.m,
                                     `NHC summary stats` = paste(C.m," ","(",round(C.m/(C.f+C.m)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.f+P.m+C.f+C.m,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# hispanic or latino
P.y <- table(subset(subject.data, Case_status == "PD")$Hispanic_or_Latino)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Hispanic_or_Latino)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Hispanic_or_Latino)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Hispanic_or_Latino)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = 'Ancestry',
                                     Metadata ="Hispanic or Latino",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# race
P.y <- table(subset(subject.data, Case_status == "PD")$Race)['White']
P.n <- sum(table(subset(subject.data, Case_status == "PD")$Race))-P.y
C.y <- table(subset(subject.data, Case_status == "Control")$Race)['White']
C.n <- sum(table(subset(subject.data, Case_status == "Control")$Race))-C.y
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Race (N & % White)",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# jewish ancestry
P.y <- table(subset(subject.data, Case_status == "PD")$Jewish_ancestory)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Jewish_ancestory)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Jewish_ancestory)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Jewish_ancestory)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Jewish",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# bmi
P.t <- length(na.omit(subset(subject.data, Case_status == "PD")$BMI))
P.avg <- mean(na.omit(subset(subject.data, Case_status == "PD")$BMI))
P.sd <- sd(na.omit(subset(subject.data, Case_status == "PD")$BMI))
C.t <- length(na.omit(subset(subject.data, Case_status == "Control")$BMI))
C.avg <- mean(na.omit(subset(subject.data, Case_status == "Control")$BMI))
C.sd <- sd(na.omit(subset(subject.data, Case_status == "Control")$BMI))
p <- wilcox.test(subset(subject.data, Case_status == "PD")$BMI,
                 subset(subject.data, Case_status == "Control")$BMI)$p.value
results <- rbind(results, data.frame(Category = 'Weight',
                                     Metadata ="BMI",
                                     `PD N` = P.t,
                                     `PD summary stats` = paste(round(P.avg, 1),round(P.sd, 1), sep = "±"),
                                     `NHC N` = C.t,
                                     `NHC summary stats` = paste(round(C.avg, 1),round(C.sd, 1), sep = "±"),
                                     `Total N` = P.t+C.t,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]` = "-",
                                     check.names=F))
# lost 10lbs in past year
P.y <- table(subset(subject.data, Case_status == "PD")$Loss_10lbs_in_last_year)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Loss_10lbs_in_last_year)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Loss_10lbs_in_last_year)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Loss_10lbs_in_last_year)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Lost >10 pounds in past year",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# gained 10lbs in past year
P.y <- table(subset(subject.data, Case_status == "PD")$Gained_10lbs_in_last_year)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Gained_10lbs_in_last_year)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Gained_10lbs_in_last_year)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Gained_10lbs_in_last_year)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Gained >10 pounds in past year",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# fruits or vegetables daily
P.y <- table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_FRUITS_or_VEGETABLES)["At least once a day"]
P.n <- sum(table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_FRUITS_or_VEGETABLES))-P.y
C.y <- table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_FRUITS_or_VEGETABLES)["At least once a day"]
C.n <- sum(table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_FRUITS_or_VEGETABLES))-C.y
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = 'Diet',
                                     Metadata ="Fruits or vegetables daily",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# poultry, beef, pork, seafood, eggs daily
P.y <- table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_POULTRY_BEEF_PORK_SEAFOOD_EGGS)["At least once a day"]
P.n <- sum(table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_POULTRY_BEEF_PORK_SEAFOOD_EGGS))-P.y
C.y <- table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_POULTRY_BEEF_PORK_SEAFOOD_EGGS)["At least once a day"]
C.n <- sum(table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_POULTRY_BEEF_PORK_SEAFOOD_EGGS))-C.y
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Poultry, beef, pork, seafood, eggs daily",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# nuts daily
P.y <- table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_NUTS)["At least once a day"]
P.n <- sum(table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_NUTS))-P.y
C.y <- table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_NUTS)["At least once a day"]
C.n <- sum(table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_NUTS))-C.y
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Nuts daily",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# yogurt at least a few times a week
P.y <- table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_YOGURT)["At least once a day"]+
       table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_YOGURT)["Few times a week"]
P.n <- sum(table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_YOGURT))-P.y
C.y <- table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_YOGURT)["At least once a day"]+
       table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_YOGURT)["Few times a week"]
C.n <- sum(table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_YOGURT))-C.y
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Yogurt at least a few times a week",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# grains daily
P.y <- table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_GRAINS)["At least once a day"]
P.n <- sum(table(subset(subject.data, Case_status == "PD")$How_often_do_you_eat_GRAINS))-P.y
C.y <- table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_GRAINS)["At least once a day"]
C.n <- sum(table(subset(subject.data, Case_status == "Control")$How_often_do_you_eat_GRAINS))-C.y
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Grains daily",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# alcohol
P.y <- table(subset(subject.data, Case_status == "PD")$Do_you_drink_alcohol)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Do_you_drink_alcohol)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Do_you_drink_alcohol)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Do_you_drink_alcohol)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Alcohol",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# do you smoke
P.y <- table(subset(subject.data, Case_status == "PD")$Do_you_smoke)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Do_you_smoke)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Do_you_smoke)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Do_you_smoke)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Tobacco",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# caffeine
P.y <- table(subset(subject.data, Case_status == "PD")$Do_you_drink_caffeinated_beverages)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Do_you_drink_caffeinated_beverages)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Do_you_drink_caffeinated_beverages)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Do_you_drink_caffeinated_beverages)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Caffeine",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# constipation day of stool collection
P.y <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_constipation)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_constipation)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_constipation)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_constipation)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = 'GI health on day of stool collection',
                                     Metadata ="Constipation (no bowel movement) in ≥3 days prior to stool collection",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# diarrhea day of stool collection
P.y <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_diarrhea)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_diarrhea)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_diarrhea)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_diarrhea)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Diarrhea",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# abdominal pain day of stool collection
P.y <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_abdominal_pain)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_abdominal_pain)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_abdominal_pain)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_abdominal_pain)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Abdominal pain",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# excess gas day of stool collection
P.y <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_excess_gas)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_excess_gas)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_excess_gas)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_excess_gas)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Excess gas",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# bloating day of stool collection
P.y <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_bloating)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_bloating)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_bloating)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_bloating)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Bloating",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# GI discomfort day of stool collection
P.y <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_digestion_issue)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Day_of_stool_collection_digestion_issue)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_digestion_issue)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Day_of_stool_collection_digestion_issue)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="GI discomfort on day of stool collection (yes to any of the five items)",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# bristol stool chart
P.t <- length(na.omit(subset(subject.data, Case_status == "PD")$Bristol_stool_chart))
P.avg <- mean(na.omit(subset(subject.data, Case_status == "PD")$Bristol_stool_chart))
P.sd <- sd(na.omit(subset(subject.data, Case_status == "PD")$Bristol_stool_chart))
C.t <- length(na.omit(subset(subject.data, Case_status == "Control")$Bristol_stool_chart))
C.avg <- mean(na.omit(subset(subject.data, Case_status == "Control")$Bristol_stool_chart))
C.sd <- sd(na.omit(subset(subject.data, Case_status == "Control")$Bristol_stool_chart))
p <- wilcox.test(subset(subject.data, Case_status == "PD")$Bristol_stool_chart,
                 subset(subject.data, Case_status == "Control")$Bristol_stool_chart)$p.value
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Bristol stool chart",
                                     `PD N` = P.t,
                                     `PD summary stats` = paste(round(P.avg, 1),round(P.sd, 1), sep = "±"),
                                     `NHC N` = C.t,
                                     `NHC summary stats` = paste(round(C.avg, 1),round(C.sd, 1), sep = "±"),
                                     `Total N` = P.t+C.t,
                                     P = formatC(p, format = "e", digits = 1), `OR [95%CI]` = "-", check.names=F))
# constipation in the past 3 months
P.y <- table(subset(subject.data, Case_status == "PD")$Constipation)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Constipation)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Constipation)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Constipation)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = 'GI health in 3 months prior to stool collection',
                                     Metadata ="Constipation (< 3 bowel movements per week)",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# diarrhea (once a week or more)
P.y <- table(subset(subject.data, Case_status == "PD")$Diarrhea)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Diarrhea)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Diarrhea)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Diarrhea)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Diarrhea (once a week or more)",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# colitis
P.y <- table(subset(subject.data, Case_status == "PD")$Colitis)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Colitis)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Colitis)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Colitis)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = 'GI disease',
                                     Metadata ="Colitis",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# IBS
P.y <- table(subset(subject.data, Case_status == "PD")$IBS)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$IBS)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$IBS)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$IBS)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Irritable bowel syndrome",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# Crohn's disease
P.y <- table(subset(subject.data, Case_status == "PD")$Crohns_disease)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Crohns_disease)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Crohns_disease)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Crohns_disease)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Crohn's disease",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 1),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 1),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# IBD
P.y <- table(subset(subject.data, Case_status == "PD")$IBD)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$IBD)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$IBD)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$IBD)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Inflammatory bowel disease",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# ulcers in past 3 months
P.y <- table(subset(subject.data, Case_status == "PD")$Ulcer_past_3_months)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Ulcer_past_3_months)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Ulcer_past_3_months)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Ulcer_past_3_months)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Ulcers in the past 3 months",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 1),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# SIBO
P.y <- table(subset(subject.data, Case_status == "PD")$SIBO)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$SIBO)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$SIBO)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$SIBO)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Small instestinal bacterial overgrowth",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 1),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# Celiac disease
P.y <- table(subset(subject.data, Case_status == "PD")$Celiac_disease)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Celiac_disease)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Celiac_disease)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Celiac_disease)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Celiac disease",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 1),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# GI cancer in past 3 months
P.y <- table(subset(subject.data, Case_status == "PD")$GI_cancer_past_3_months)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$GI_cancer_past_3_months)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$GI_cancer_past_3_months)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$GI_cancer_past_3_months)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="GI cancer in the last 3 months",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 1),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# intestinal disease
P.y <- table(subset(subject.data, Case_status == "PD")$Intestinal_disease)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Intestinal_disease)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Intestinal_disease)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Intestinal_disease)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="GI disease (yes to any of the eight items)",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# indigestion drugs
P.y <- table(subset(subject.data, Case_status == "PD")$Indigestion_drugs)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Indigestion_drugs)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Indigestion_drugs)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Indigestion_drugs)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = 'Medications (currentlty taking at time of stool collection, unless noted)',
                                     Metadata ="Indigestion drugs",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# antibiotics
P.y <- table(subset(subject.data, Case_status == "PD")$Antibiotics_current)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Antibiotics_current)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Antibiotics_current)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Antibiotics_current)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Antibiotics",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# antibiotics in past 3 months
P.y <- table(subset(subject.data, Case_status == "PD")$Antibiotics_past_3_months)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Antibiotics_past_3_months)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Antibiotics_past_3_months)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Antibiotics_past_3_months)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Antibiotics in past 3 months",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# laxatives
P.y <- table(subset(subject.data, Case_status == "PD")$Laxatives)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Laxatives)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Laxatives)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Laxatives)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Laxatives",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# anti-inflammatory drugs
P.y <- table(subset(subject.data, Case_status == "PD")$Anti_inflammatory_drugs)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Anti_inflammatory_drugs)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Anti_inflammatory_drugs)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Anti_inflammatory_drugs)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Anti-inflammatory drugs",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# probiotics
P.y <- table(subset(subject.data, Case_status == "PD")$Probiotic)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Probiotic)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Probiotic)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Probiotic)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Probiotics",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# radiation or chemotherapy
P.y <- table(subset(subject.data, Case_status == "PD")$Radiation_Chemo)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Radiation_Chemo)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Radiation_Chemo)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Radiation_Chemo)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Radiation or chemotherapy",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# blood thinners
P.y <- table(subset(subject.data, Case_status == "PD")$Blood_thinners)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Blood_thinners)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Blood_thinners)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Blood_thinners)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Blood thinners",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# cholesterol medication
P.y <- table(subset(subject.data, Case_status == "PD")$Cholesterol_med)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Cholesterol_med)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Cholesterol_med)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Cholesterol_med)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Cholesterol medication",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# blood pressure medication
P.y <- table(subset(subject.data, Case_status == "PD")$Blood_pressure_med)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Blood_pressure_med)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Blood_pressure_med)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Blood_pressure_med)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Blood pressure medication",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# thyroid medication
P.y <- table(subset(subject.data, Case_status == "PD")$Thyroid_med)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Thyroid_med)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Thyroid_med)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Thyroid_med)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Thyroid medication",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# asthma or COPD medication
P.y <- table(subset(subject.data, Case_status == "PD")$Asthma_or_COPD_med)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Asthma_or_COPD_med)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Asthma_or_COPD_med)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Asthma_or_COPD_med)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Asthma or COPD medication",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# diabetes medication
P.y <- table(subset(subject.data, Case_status == "PD")$Diabetes_med)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Diabetes_med)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Diabetes_med)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Diabetes_med)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Diabetes medication",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# pain medication
P.y <- table(subset(subject.data, Case_status == "PD")$Pain_med)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Pain_med)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Pain_med)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Pain_med)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Pain medication",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# depression, anxiety, mood medication
P.y <- table(subset(subject.data, Case_status == "PD")$Depression_anxiety_mood_med)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Depression_anxiety_mood_med)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Depression_anxiety_mood_med)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Depression_anxiety_mood_med)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Depression, anxiety, mood medication",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# birth control or estrogen (females only)
P.y <- table(subset(subject.data, Case_status == "PD" & Sex == "F")$Birth_control_or_estrogen)['Y']
P.n <- table(subset(subject.data, Case_status == "PD" & Sex == "F")$Birth_control_or_estrogen)['N']
C.y <- table(subset(subject.data, Case_status == "Control" & Sex == "F")$Birth_control_or_estrogen)['Y']
C.n <- table(subset(subject.data, Case_status == "Control" & Sex == "F")$Birth_control_or_estrogen)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Birth control or estrogen (females)",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# antihistamines
P.y <- table(subset(subject.data, Case_status == "PD")$Antihistamines)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Antihistamines)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Antihistamines)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Antihistamines)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Antihistamines",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# Co Q 10
P.y <- table(subset(subject.data, Case_status == "PD")$Co_Q_10)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Co_Q_10)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Co_Q_10)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Co_Q_10)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Co-Q 10",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# sleep aid
P.y <- table(subset(subject.data, Case_status == "PD")$Sleep_aid)['Y']
P.n <- table(subset(subject.data, Case_status == "PD")$Sleep_aid)['N']
C.y <- table(subset(subject.data, Case_status == "Control")$Sleep_aid)['Y']
C.n <- table(subset(subject.data, Case_status == "Control")$Sleep_aid)['N']
if (is.na(P.n)){P.n <- 0}
if (is.na(P.y)){P.y <- 0}
if (is.na(C.n)){C.n <- 0}
if (is.na(C.y)){C.y <- 0}
p <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$p.value
if (any(c(P.y, P.n, C.y, C.n)==0)){
  or <- Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$estimate
  ci <- paste(round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[1],1),
              round(Prop.or(x=c(P.y,P.n), y=c(C.y,C.n), CImethod='Woolf')$conf.int[2],1), sep='-')
}else{
  or <- fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$estimate
  ci <- paste(round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[1],1),
              round(fisher.test(matrix(c(P.y,P.n,C.y,C.n), nrow = 2))$conf.int[2],1), sep='-')
}
results <- rbind(results, data.frame(Category = '',
                                     Metadata ="Sleep aid",
                                     `PD N` = P.n+P.y,
                                     `PD summary stats` = paste(P.y," ","(",round(P.y/(P.n+P.y)*100, 0),"%",")",
                                                                sep = ""),
                                     `NHC N` = C.n+C.y,
                                     `NHC summary stats` = paste(C.y," ","(",round(C.y/(C.n+C.y)*100, 0),"%",")",
                                                                 sep = ""),
                                     `Total N` = P.n+P.y+C.n+C.y,
                                     P = formatC(p, format = "e", digits = 1),
                                     `OR [95%CI]`=paste(round(or, 1),' [',ci,']',sep=''),
                                     check.names=F))
# add variable numbers
results <- data.frame(Index=c('',1:(nrow(results)-1)), results, check.names=F)

# write out results
wb <- createWorkbook()

addWorksheet(wb, 'Subject characteristics')
writeData(wb, 'Subject characteristics', results, keepNA=T)
setColWidths(wb, 'Subject characteristics', cols=1:ncol(results), widths=c(10,20,55,10,15,10,15,10,10,12)) ### format cells
addStyle(wb, 'Subject characteristics', cols=1:ncol(results), rows=1:(nrow(results)+1), gridExpand=T, style=center, stack=T)
mergeCells(wb, 'Subject characteristics', cols=2, rows=2:4)
mergeCells(wb, 'Subject characteristics', cols=2, rows=5:7)
mergeCells(wb, 'Subject characteristics', cols=2, rows=8:10)
mergeCells(wb, 'Subject characteristics', cols=2, rows=11:18)
mergeCells(wb, 'Subject characteristics', cols=2, rows=19:25)
mergeCells(wb, 'Subject characteristics', cols=2, rows=26:27)
mergeCells(wb, 'Subject characteristics', cols=2, rows=28:36)
mergeCells(wb, 'Subject characteristics', cols=2, rows=37:55)
addStyle(wb, 'Subject characteristics', cols=1:ncol(results), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'Subject characteristics', cols=1:ncol(results), rows=c(1,2,(nrow(results)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
addStyle(wb, 'Subject characteristics', cols=1:ncol(results), rows=c(5,8,11,19,26,28,37), gridExpand=T,
         style=horizontal_border_thin, stack=T)
convertNum(results, wb, 'Subject characteristics', TRUE) ### convert numbers from strings back to numbers

saveWorkbook(wb, 'output/1.Metadata/Subject_characteristics.xlsx', overwrite=T) ### write out

# print table
kable(results, col.names = colnames(results),
      row.names=F, align='c', caption = 'Subject characteristics') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='500px', width='1000px')
```

* To determine if any of the PD-associated subject variables are collinear, performed logistic regression (via the `glm` function in R with `family='binomial'`) with all variables in the model, and calculated variance inflation factor (VIF) (using `vif` function from the `car` package).

```{r vif_subject_data, echo=T}
# perform logistic regression model with all PD-associated variables in model
subject.data$Case_status <- dplyr::recode(subject.data$Case_status, PD=1L, Control=0L)
mod <- glm(Case_status ~ Age_at_collection + Sex + Day_of_stool_collection_digestion_issue +
                         Bristol_stool_chart + Loss_10lbs_in_last_year + Do_you_drink_alcohol + Constipation + Diarrhea +
                         Laxatives + Probiotic + Pain_med + Depression_anxiety_mood_med +
                         Antihistamines + Sleep_aid,
                         data=subject.data, family='binomial')

# calculate vif for each variable
var.vif <- car::vif(mod)

# report results of model and vif
results <- data.frame()
suppress(
for (i in 1:length(var.vif)){
  results <- rbind(results, data.frame(OR=paste(round(exp(summary(mod)$coefficients[i+1,1]),2), ' [',
                                          paste(round(exp(confint(mod)[i+1,1]),2), round(exp(confint(mod)[i+1,2]),2), sep='-'), ']',
                                                sep=''),
                                       P=formatC(summary(mod)$coefficients[i+1,4], format = "e", digits = 1),
                                       VIF=round(var.vif[i],2)))
})
rownames(results) <- c("Age (years)",
                      "Sex (Male vs Female)",
                      "GI discomfort on the day of stool collection (Yes vs No)",
                      "Bristol stool chart (1-7)",
                      "Lost >10 pounds in past year (Yes vs No)",
                      "Alcohol (Yes vs No)",
                      "Constipation (<3 bowel movements per week) in the past 3 months (Yes vs No)",
                      "Diarrhea (once a week or more) (Yes vs No)",
                      "Currently taking laxatives (Yes vs No)",
                      "Currently taking probiotics (Yes vs No)",
                      "Currently taking pain medication (Yes vs No)",
                      "Currently taking depression, anxiety, mood medication (Yes vs No)",
                      "Currently taking antihistamines (Yes vs No)",
                      "Currently taking sleep aid")
results <- rownames_to_column(results, "Variable")

addWorksheet(wb, 'Collinearity PD variables')
writeData(wb, 'Collinearity PD variables', results, keepNA=T, colNames=T)
setColWidths(wb, 'Collinearity PD variables', cols=1:ncol(results), widths=c(60,14,10,10)) ### format cells
addStyle(wb, 'Collinearity PD variables', cols=1:ncol(results), rows=1:(nrow(results)+1), gridExpand=T, style=center, stack=T)
addStyle(wb, 'Collinearity PD variables', cols=1:ncol(results), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'Collinearity PD variables', cols=1:ncol(results), rows=c(1,2,(nrow(results)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
convertNum(results, wb, 'Collinearity PD variables', TRUE) ### convert numbers from strings back to numbers

saveWorkbook(wb, 'output/1.Metadata/Subject_characteristics.xlsx', overwrite=T) ### write out

# print table
kable(results, col.names = colnames(results),
      row.names=F, align='c', caption = 'Variance inflation factors of PD variables') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='500px', width='1000px')
```

### Examining inter-sample differences in microbiome compositions

To observe inter-sample differences in gut microbiome compositions (beta-diversity), principal component analysis (PCA) was performed to visually inspect differences in microbiome compositions between samples. Case status was statistically tested for association with inter-sample differences in microbiome compositions using `adonis2` adjusting for technical covariates (sample collection method and total read count per sample standardized using the `scale` function in R).

* To perform the PCA, species counts from MetaPhlAn were transformed using the clr transformation (formula: `log(x+1)-mean(log(x+1))`). PCA was then performed using the `prcomp` function with default parameters. PC1 and PC2 were then plotted with convex hull areas for each group using the `autoplot` function from `ggfortify`. File **2.Gut_microbiome_composition/PCA_case_status.pdf** contains the PCA plot which is showcased in **Supplementary Fig. 2a** in the manuscript.

```{r PCA, echo=T, fig.width=5, fig.height=5}
# prep directory for output
system('
if [ ! -d "output/2.Gut_microbiome_composition" ]
then
  mkdir output/2.Gut_microbiome_composition
fi
')

# transform abundances to clr
abun.clr <- transform_sample_counts(filter_taxa(abun.ps.s, function(x){sum(x>0)>0}, TRUE),
                                    function(x){log(x+1)-mean(log(x+1))})

# perform PCA
pca <- prcomp(otu_table(abun.clr))

# plot PC1 and PC2 coloring by case status
pd <- pca$x[grep('DP', rownames(pca$x)),1:2][
        chull(pca$x[grep('DP', rownames(pca$x)),1:2]),]
hc <- pca$x[grep('DC', rownames(pca$x)),1:2][
        chull(pca$x[grep('DC', rownames(pca$x)),1:2]),]
suppress(
g <- autoplot(pca, data=data.frame(sample_data(abun.clr)), colour='Case_status',
              shape='Case_status', scale=F, frame=T) +
  geom_polygon(data=pd, alpha=0.01, color="#00BFC4", fill="#00BFC4") +
  geom_polygon(data=hc, alpha=0.01, color="#E69F00", fill="#E69F00") +
  theme_bw() +
  scale_colour_manual(labels=c('NHC','PD'), values=c("#E69F00", "#00BFC4")) +
  scale_fill_manual(labels=c('NHC','PD'), values=c("#E69F00", "#00BFC4")) +
  scale_shape_manual(labels=c('NHC','PD'), values=c(17,16)) +
  labs(fill="Case status", color="Case status", shape="Case status")
)
ggsave('output/2.Gut_microbiome_composition/PCA_case_status.pdf', g, device='pdf', width=5, height=5)

g
```

* To test if case status significantly associates with beta-diversity, the function `adonis2` from `vegan` was used to test association of case status adjusting for sampling method, and total read count per sample (standardized using the `scale` function) on inter-sample variation in microbiome compositions. Significance of the associations was determined using 9999 permutations. All variables were adjusted for one another in a marginal model by setting `by='margin'`. Aitchison distance (Euclidean distance of clr transformed data) was used as the distance matrix outcome. Results can be found in file **2.Gut_microbiome_composition/PERMANOVA_results.xlsx**, and are reported in the main manuscript text.

```{r permanova, echo=T}
# perform adonis
sample_data(abun.clr)$total_seqs_scaled <- scale(sample_data(abun.clr)$total_sequences)
set.seed(1234)
fit <- adonis2(otu_table(abun.clr) ~ Case_status + total_seqs_scaled + collection_method,
                data=data.frame(sample_data(abun.clr)), method='euclidean', by='margin', perm=9999)

# print results
results <- data.frame(Variable=c('case status','sequence depth (standardized)','collection method'),
                      PERMANOVA_Results=c(paste('r2=',round(fit$R2[1],3),
                                                ', P=',formatC(fit$`Pr(>F)`[1],format='e',digits=0),
                                                sep=''),
                                          paste('r2=',round(fit$R2[2],3),
                                                ', P=',formatC(fit$`Pr(>F)`[2],format='e',digits=0),
                                                sep=''),
                                          paste('r2=',round(fit$R2[3],3),
                                                ', P=',formatC(fit$`Pr(>F)`[3],format='e',digits=0),
                                                sep='')))
wb <- createWorkbook()

addWorksheet(wb, 'PERMANOVA results')
writeData(wb, 'PERMANOVA results', results, keepNA=T)
setColWidths(wb, 'PERMANOVA results', cols=1:ncol(results), widths=c(20,20)) ### format cells
addStyle(wb, 'PERMANOVA results', cols=1:ncol(results), rows=1:(nrow(results)+1), gridExpand=T, style=center, stack=T)
addStyle(wb, 'PERMANOVA results', cols=1:ncol(results), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'PERMANOVA results', cols=1:ncol(results), rows=c(1,2,(nrow(results)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)

saveWorkbook(wb, 'output/2.Gut_microbiome_composition/PERMANOVA_results.xlsx', overwrite=T) ### write out

# print table
kable(results, col.names = colnames(results),
      row.names=F, align='c', caption = 'PERMANOVA results for PD vs NHC adjusted for sequence depth and collection method') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='200px', width='500')
```

### Testing differences in enterotypes between PD and NHC

To determine if PD patients had a different distribution of enterotype frequencies than NHC, enterotype profiling was performed using the web-based EMBL enterotype classifier (http://enterotypes.org/), then differences in enterotype frequencies between PD and NHC tested using the `chisq.test` function to perform Pearson's Chi-squared test. The enterotype classifier uses the orignal entertoype definitions, classifying each sample as the enterotype *Bacteroides*, *Firmicutes*, or *Prevotella*.

* To perform enterotype profiling, the MetaPhlAn relative abundance file was subsetted for only genus level entries, then uploaded to the web-based EMBL enterotype classifier. The raw results were downloaded and used to perform the Chi-squared test, and generate the mosaic plot. The Chi-squred results can be found in the file **2.Gut_microbiome_composition/Enterotype_chisq_results.xlsx**, and mosaic plot found in the file **2.Gut_microbiome_composition/Enterotype_mosaic_plot.pdf**. These results are reported in the manuscript text and in **Supplementary Fig. 2b**.

```{r entertoypes, echo=T, fig.width=10, fig.height=3}
# read in enterotype profiles
et <- read.table('input/enterotypes.txt', header=T)

# make a quick column for case status
et$case_status <- NA
et$case_status[grep('P', rownames(et))] <- "PD"
et$case_status[grep('C', rownames(et))] <- "NHC"

# perform chi-squared test
chisq.res <- chisq.test(table(et$case_status, et$ET))

# print results
results <- data.frame()
results <- data.frame(`Case status`=c(rownames(table(et$case_status, et$ET))[2], rownames(table(et$case_status, et$ET))[1]),
                      Bacteroides=c(paste(table(et$case_status, et$ET)[2,1],' (',
                                          round(table(et$case_status, et$ET)[2,1]/sum(table(et$case_status, et$ET)[2,])*100,0),'%)',
                                          sep=''),
                                    paste(table(et$case_status, et$ET)[1,1],' (',
                                          round(table(et$case_status, et$ET)[1,1]/sum(table(et$case_status, et$ET)[1,])*100,0),'%)',
                                          sep='')),
                      Firmicutes=c(paste(table(et$case_status, et$ET)[2,2],' (',
                                         round(table(et$case_status, et$ET)[2,2]/sum(table(et$case_status, et$ET)[2,])*100,0),'%)',
                                         sep=''),
                                   paste(table(et$case_status, et$ET)[1,2],' (',
                                         round(table(et$case_status, et$ET)[1,2]/sum(table(et$case_status, et$ET)[1,])*100,0),'%)',
                                         sep='')),
                      Prevotella=c(paste(table(et$case_status, et$ET)[2,3],' (',
                                         round(table(et$case_status, et$ET)[2,3]/sum(table(et$case_status, et$ET)[2,])*100,0),'%)',
                                         sep=''),
                                   paste(table(et$case_status, et$ET)[1,3],' (',
                                         round(table(et$case_status, et$ET)[1,3]/sum(table(et$case_status, et$ET)[1,])*100,0),'%)',
                                         sep='')),
                      `Chi-squared results`=c('',
                                              paste('X2=',round(chisq.res$statistic,1),
                                                    ', P=',formatC(chisq.res$p.value,format='e',digits=0),
                                                    sep='')),
                      check.names=F)
wb <- createWorkbook()

addWorksheet(wb, 'Chi-squared results')
writeData(wb, 'Chi-squared results', results, keepNA=T)
setColWidths(wb, 'Chi-squared results', cols=1:ncol(results), widths=c(rep(15,4),20)) ### format cells
addStyle(wb, 'Chi-squared results', cols=1:ncol(results), rows=1:(nrow(results)+1), gridExpand=T, style=center, stack=T)
addStyle(wb, 'Chi-squared results', cols=1:ncol(results), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'Chi-squared results', cols=1:ncol(results), rows=c(1,2,(nrow(results)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)

saveWorkbook(wb, 'output/2.Gut_microbiome_composition/Enterotype_chisq_results.xlsx', overwrite=T) ### write out

# print table
kable(results, col.names = colnames(results),
      row.names=F, align='c', caption = 'Chi-squared results for differences in enterotype distribution') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='200px', width='500')

# create mosaic plot
pdf('output/2.Gut_microbiome_composition/Enterotype_mosaic_plot.pdf', height=3, width=10)
counts <- table(dplyr::recode(et$case_status, PD=1L, NHC=2L), et$ET)
rownames(counts) <- c("PD", "NHC")
colnames(counts) <- c("Bacteroides","Firmicutes","Prevotella")
dimnames(counts) <- list(`Case status`=c("PD", "NHC"),
                         `Enterotype`=c("Bacteroides", "Firmicutes", "Prevotella"))
percents <- rbind(paste(round(table(et$case_status, et$ET)[2,]/sum(table(et$case_status, et$ET)[2,])*100,0),'%',sep=''),
                  paste(round(table(et$case_status, et$ET)[1,]/sum(table(et$case_status, et$ET)[1,])*100,0),'%',sep=''))
rownames(percents) <- c("PD", "NHC")
colnames(percents) <- c("Bacteroides","Firmicutes","Prevotella")
dimnames(percents) <- list(`Case status`=c("PD", "NHC"),
                           `Enterotype`=c("Bacteroides", "Firmicutes", "Prevotella"))
mosaic(counts, main=NULL, highlighting='Enterotype', highlighting_fill=c('grey50','dodgerblue3','firebrick'),
       spacing=spacing_equal(sp=0.5), labeling=labeling_border(varname=FALSE, rot_labels=0), keep_aspect_ratio=FALSE, pop=FALSE)
  labeling_cells(text=percents, gp_text=gpar(col='white'), margin=0)(counts)
trash <- dev.off()

mosaic(counts, main=NULL, highlighting='Enterotype', highlighting_fill=c('grey50','dodgerblue3','firebrick'),
       spacing=spacing_equal(sp=0.5), labeling=labeling_border(varname=FALSE, rot_labels=0), keep_aspect_ratio=FALSE, pop=FALSE)
  labeling_cells(text=percents, gp_text=gpar(col='white'), margin=0)(counts)
```

### Differential abundance analysis of species and genera for PD vs NHC

To determine what species and genera are differentially abundant between PD and NHC samples, differential abundance analysis was performed using two methods: 1) ANCOM-BC with count data [relative abundance with unknown estimation x total reads] and 2) linear regression with log2 transformed relative abundances (without unknown estimation) as implemented in MaAsLin2.

* To perform differential abundance analysis using ANCOM-BC with counts, counts were used as input for the `ancombc` function of the `ANCOMBC` R package. The ANCOM-BC `formula` included case status (PD vs NHC), collection method (swab vs OMNIgene GUT kit), and total sequence count (taken from `#nread` line of bowtie2 intermediate files produced by MetaPhlAn) standardized using the `scale` function in R with default parameters. All parameters were left as default except for the FDR adjustment which was made to be the Benjamini-Hochberg (`BH`) method, and the `zero_cut` which was made `0.95` to make the effective sample size for analysis 37 samples.
* To peform differential abundance analysis using linear regression with log2 transformed relative abundances, relative abundances from MetaPhlAn were divided by 100 to convert to proportions and used as input to the `Maaslin2` function. All parameters were left as default except for `min_prevalence` which was set to `0.05` to make the effective sample size 37, `normalization` which was set to `NONE` as we are already inputting relative abundances, `standardize` which was set to `FALSE` as this is being done prior to MaAsLin2, and `max_significance` which was set to `0.05`. The MaAsLin2 `fixed_effects` model included case status (PD vs NHC), collection method (swab vs OMNIgene GUT kit), and total sequence count (taken from `#nread` line of bowtie2 intermediate files produced by MetaPhlAn) standardized using the `scale` function.
* Results for the analyses can be found in result file **3.Taxonomic_associations/MaAsLin2_ANCOMBC_MWAS_PDvsNHC.xlsx**. Data from this file are showcased in **Tables 2 and 3** and **Supplementary Tables 1 and 2** of the manuscript.

```{r DA_analysis, echo=T}
# prep directory for output
system('
if [ ! -d "output/3.Taxonomic_associations" ]
then
  mkdir output/3.Taxonomic_associations
  mkdir output/3.Taxonomic_associations/MaAsLin2_MWAS_PDvsNHC
  mkdir output/3.Taxonomic_associations/MaAsLin2_MWAS_PDvsNHC/Species
  mkdir output/3.Taxonomic_associations/MaAsLin2_MWAS_PDvsNHC/Genera
fi
')

# recode categorical variable to get correct effect direction and scale numeric data
sample_data(abun.ps.s)$Case_status <- dplyr::recode(sample_data(abun.ps.s)$Case_status, PD=1, Control=0)
sample_data(abun.ps.g)$Case_status <- dplyr::recode(sample_data(abun.ps.g)$Case_status, PD=1, Control=0)
sample_data(abun.ps.s)$collection_method <- dplyr::recode(sample_data(abun.ps.s)$collection_method, swab=1, `OMNIgene GUT`=0)
sample_data(abun.ps.g)$collection_method <- dplyr::recode(sample_data(abun.ps.g)$collection_method, swab=1, `OMNIgene GUT`=0)
sample_data(abun.ps.s)$total_seqs_scaled <- scale(sample_data(abun.ps.s)$total_sequences)
sample_data(abun.ps.g)$total_seqs_scaled <- scale(sample_data(abun.ps.g)$total_sequences)

sample_data(ra.ps.s)$Case_status <- dplyr::recode(sample_data(ra.ps.s)$Case_status, PD=1, Control=0)
sample_data(ra.ps.g)$Case_status <- dplyr::recode(sample_data(ra.ps.g)$Case_status, PD=1, Control=0)
sample_data(ra.ps.s)$collection_method <- dplyr::recode(sample_data(ra.ps.s)$collection_method, swab=1, `OMNIgene GUT`=0)
sample_data(ra.ps.g)$collection_method <- dplyr::recode(sample_data(ra.ps.g)$collection_method, swab=1, `OMNIgene GUT`=0)
sample_data(ra.ps.s)$total_seqs_scaled <- scale(sample_data(ra.ps.s)$total_sequences)
sample_data(ra.ps.g)$total_seqs_scaled <- scale(sample_data(ra.ps.g)$total_sequences)

# perform differential abundance analysis using ANCOM-BC with count data
ancom.s <- ANCOMBC.plus(ps=abun.ps.s, formula="Case_status + collection_method + total_seqs_scaled",
                        p_adj_method="BH", zero_cut=0.95, lib_cut=0, group=NULL,
                        struc_zero=F, neg_lb=F, tol=1E-5, max_iter=100, conserve=F, alpha=0.05, global=F)
ancom.g <- ANCOMBC.plus(ps=abun.ps.g, formula="Case_status + collection_method + total_seqs_scaled",
                        p_adj_method="BH", zero_cut=0.95, lib_cut=0, group=NULL,
                        struc_zero=F, neg_lb=F, tol=1E-5, max_iter=100, conserve=F, alpha=0.05, global=F)

# perform differential abundance analysis using linear regression with log2 transformed relative abundances
suppress(
lm.s <- MaAsLin2.plus(ps=phyloseq(otu_table(ra.ps.s)/100, sample_data(ra.ps.s)),
                      output='output/3.Taxonomic_associations/MaAsLin2_MWAS_PDvsNHC/Species',
                      metadata=c('Case_status','collection_method','total_seqs_scaled'),
                      min_abundance=0, min_prevalence=0.05, min_variance=0,
                      normalization='NONE', transform='LOG', analysis_method='LM', max_significance=0.05,
                      random_effects=NULL, fixed_effects=NULL, correction='BH', standardize=F, cores=1,
                      plot_heatmap=F, plot_scatter=F, heatmap_first_n=50, reference=NULL)
)
suppress(
lm.g <- MaAsLin2.plus(ps=phyloseq(otu_table(ra.ps.g)/100, sample_data(ra.ps.g)),
                      output='output/3.Taxonomic_associations/MaAsLin2_MWAS_PDvsNHC/Genera',
                      metadata=c('Case_status','collection_method','total_seqs_scaled'),
                      min_abundance=0, min_prevalence=0.05, min_variance=0,
                      normalization='NONE', transform='LOG', analysis_method='LM', max_significance=0.05,
                      random_effects=NULL, fixed_effects=NULL, correction='BH', standardize=F, cores=1,
                      plot_heatmap=F, plot_scatter=F, heatmap_first_n=50, reference=NULL)
)

# initialize workbook
wb <- createWorkbook()

# write results for species
res.summ <- data.frame(Variable=lm.s$result.summary$Variable,
                       Kingdom=gsub('_', ' ', gsub('k__', '', sapply(strsplit(lm.s$result.summary$Feature, "\\|"), function(x){x[1]}))),
                       Phylum=gsub('_', ' ', gsub('p__', '', sapply(strsplit(lm.s$result.summary$Feature, "\\|"), function(x){x[2]}))),
                       Class=gsub('_', ' ', gsub('c__', '', sapply(strsplit(lm.s$result.summary$Feature, "\\|"), function(x){x[3]}))),
                       Order=gsub('_', ' ', gsub('o__', '', sapply(strsplit(lm.s$result.summary$Feature, "\\|"), function(x){x[4]}))),
                       Family=gsub('_', ' ', gsub('f__', '', sapply(strsplit(lm.s$result.summary$Feature, "\\|"), function(x){x[5]}))),
                       Genus=gsub('_', ' ', gsub('g__', '', sapply(strsplit(lm.s$result.summary$Feature, "\\|"), function(x){x[6]}))),
                       Species=gsub('_', ' ', gsub('s__', '', sapply(strsplit(lm.s$result.summary$Feature, "\\|"), function(x){x[7]}))),
                       `N PD`=lm.s$result.summary$N1, `N NHC`=lm.s$result.summary$N2, space_1='',
                       `RA in PD`=lm.s$result.summary$Mean1, `RA in NHC`=lm.s$result.summary$Mean2,
                       lm.s$result.summary[,c('Beta','SE','P','FDR','FC')],
                       `FC lower`=lm.s$result.summary$FC_lower, `FC upper`=lm.s$result.summary$FC_upper, space_2='', check.names=F)
res.summ <- merge(res.summ,
                  data.frame(Variable=ancom.s$result.summary$Variable,
                             Kingdom=gsub('_', ' ', gsub('k__', '', sapply(strsplit(ancom.s$result.summary$Feature, "\\|"), function(x){x[1]}))),
                             Phylum=gsub('_', ' ', gsub('p__', '', sapply(strsplit(ancom.s$result.summary$Feature, "\\|"), function(x){x[2]}))),
                             Class=gsub('_', ' ', gsub('c__', '', sapply(strsplit(ancom.s$result.summary$Feature, "\\|"), function(x){x[3]}))),
                             Order=gsub('_', ' ', gsub('o__', '', sapply(strsplit(ancom.s$result.summary$Feature, "\\|"), function(x){x[4]}))),
                             Family=gsub('_', ' ', gsub('f__', '', sapply(strsplit(ancom.s$result.summary$Feature, "\\|"), function(x){x[5]}))),
                             Genus=gsub('_', ' ', gsub('g__', '', sapply(strsplit(ancom.s$result.summary$Feature, "\\|"), function(x){x[6]}))),
                             Species=gsub('_', ' ', gsub('s__', '', sapply(strsplit(ancom.s$result.summary$Feature, "\\|"), function(x){x[7]}))),
                             `N PD`=ancom.s$result.summary$N1, `N NHC`=ancom.s$result.summary$N2,
                             `BC-OA in PD`=ancom.s$result.summary$Mean1, `BC-OA in NHC`=ancom.s$result.summary$Mean2,
                             ancom.s$result.summary[,c('Beta','SE','P','FDR','FC')],
                             `FC lower`=ancom.s$result.summary$FC_lower, `FC upper`=ancom.s$result.summary$FC_upper, check.names=F),
                  by=c('Variable','Kingdom','Phylum','Class','Order','Family','Genus','Species','N PD','N NHC'),
                  suffix=c('_m','_a'), all=T, sort=F)
res.summ <- res.summ[res.summ$Variable=='Case_status',-1]
res.summ <- rbind(data.frame(Kingdom='', Phylum='', Class='', Order='', Family='', Genus='', Species='', `N PD`='', `N NHC`='', space_1='',
                             `RA in PD`='MaAsLin2 results', `RA in NHC`='', Beta_m='', SE_m='', P_m='', FDR_m='',
                             FC_m='', `FC lower_m`='', `FC upper_m`='', space_2='',
                             `BC-OA in PD`='ANCOM-BC results', `BC-OA in NHC`='', Beta_a='', SE_a='', P_a='', FDR_a='',
                             FC_a='', `FC lower_a`='', `FC upper_a`='',
                             check.names=F),
                  data.frame(Kingdom='Kingdom', Phylum='Phylum', Class='Class', Order='Order', Family='Famiy', Genus='Genus', Species='Species',
                             `N PD`='N PD', `N NHC`='N NHC', space_1='', `RA in PD`='RA in PD', `RA in NHC`='RA in NHC', Beta_m='Beta',
                             SE_m='SE', P_m='P', FDR_m='FDR', FC_m='FC', `FC lower_m`='FC lower', `FC upper_m`='FC upper', space_2='',
                             `BC-OA in PD`='BC-OA in PD', `BC-OA in NHC`='BC-OA in NHC',
                             Beta_a='Beta', SE_a='SE', P_a='P', FDR_a='FDR', FC_a='FC',
                             `FC lower_a`='FC lower', `FC upper_a`='FC upper', check.names=F),
                  res.summ)
res.summ[3:(nrow(res.summ)-1),c(grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1),
                            grep('ANCOM', res.summ[1,]):ncol(res.summ))][
         is.na(res.summ[3:(nrow(res.summ)-1),c(grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1),
                                           grep('ANCOM', res.summ[1,]):ncol(res.summ))])] <- 'NT'
addWorksheet(wb, 'Species results')
writeData(wb, 'Species results', res.summ, keepNA=F, colNames=F)
setColWidths(wb, 'Species results', cols=1:ncol(res.summ), widths=c(10, rep(22,6), rep(11,2), 2, rep(11,9), 2, rep(11,9))) ### format cells
mergeCells(wb, 'Species results', cols=grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1), rows=1)
mergeCells(wb, 'Species results', cols=grep('ANCOM', res.summ[1,]):ncol(res.summ), rows=1)
addStyle(wb, 'Species results', cols=1:ncol(res.summ), rows=1:2, style=bold, stack=T, gridExpand=T) ### font
addStyle(wb, 'Species results', cols=1:ncol(res.summ), rows=c(1,3,(nrow(res.summ)+1)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
addStyle(wb, 'Species results', cols=grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1), rows=2,
         style=horizontal_border_thin, stack=T)
addStyle(wb, 'Species results', cols=grep('ANCOM', res.summ[1,]):ncol(res.summ), rows=2, style=horizontal_border_thin, stack=T)
convertNum(res.summ, wb, 'Species results', FALSE) ### convert numbers from strings back to numbers

# print table
res.summ[3:nrow(res.summ),c(11,12,15,16,25,26)] <- apply(res.summ[3:nrow(res.summ),c(11,12,15,16,25,26)],2,
                                                         function(x){formatC(as.numeric(x), format='e', digits=0)})
res.summ[3:nrow(res.summ),c(13,14,17:19,21:24,27:29)] <- apply(res.summ[3:nrow(res.summ),c(13,14,17:19,21:24,27:29)],2,
                                                               function(x){round(as.numeric(x), 2)})
kable(res.summ, col.names=NULL, row.names=F, align='c', caption = 'Species MWAS results for MaAsLin2 and ANCOM-BC') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% row_spec(1,bold=T,hline_after=F) %>% row_spec(2,bold=T,hline_after=T) %>% scroll_box(height='500px', width='1000px')

# write results for genera
res.summ <- data.frame(Variable=lm.g$result.summary$Variable,
                       Kingdom=gsub('_', ' ', gsub('k__', '', sapply(strsplit(lm.g$result.summary$Feature, "\\|"), function(x){x[1]}))),
                       Phylum=gsub('_', ' ', gsub('p__', '', sapply(strsplit(lm.g$result.summary$Feature, "\\|"), function(x){x[2]}))),
                       Class=gsub('_', ' ', gsub('c__', '', sapply(strsplit(lm.g$result.summary$Feature, "\\|"), function(x){x[3]}))),
                       Order=gsub('_', ' ', gsub('o__', '', sapply(strsplit(lm.g$result.summary$Feature, "\\|"), function(x){x[4]}))),
                       Family=gsub('_', ' ', gsub('f__', '', sapply(strsplit(lm.g$result.summary$Feature, "\\|"), function(x){x[5]}))),
                       Genus=gsub('_', ' ', gsub('g__', '', sapply(strsplit(lm.g$result.summary$Feature, "\\|"), function(x){x[6]}))),
                       `N PD`=lm.g$result.summary$N1, `N NHC`=lm.g$result.summary$N2, space_1='',
                       `RA in PD`=lm.g$result.summary$Mean1, `RA in NHC`=lm.g$result.summary$Mean2,
                       lm.g$result.summary[,c('Beta','SE','P','FDR','FC')],
                       `FC lower`=lm.g$result.summary$FC_lower, `FC upper`=lm.g$result.summary$FC_upper, space_2='', check.names=F)
res.summ <- merge(res.summ,
                  data.frame(Variable=ancom.g$result.summary$Variable,
                             Kingdom=gsub('_', ' ', gsub('k__', '', sapply(strsplit(ancom.g$result.summary$Feature, "\\|"), function(x){x[1]}))),
                             Phylum=gsub('_', ' ', gsub('p__', '', sapply(strsplit(ancom.g$result.summary$Feature, "\\|"), function(x){x[2]}))),
                             Class=gsub('_', ' ', gsub('c__', '', sapply(strsplit(ancom.g$result.summary$Feature, "\\|"), function(x){x[3]}))),
                             Order=gsub('_', ' ', gsub('o__', '', sapply(strsplit(ancom.g$result.summary$Feature, "\\|"), function(x){x[4]}))),
                             Family=gsub('_', ' ', gsub('f__', '', sapply(strsplit(ancom.g$result.summary$Feature, "\\|"), function(x){x[5]}))),
                             Genus=gsub('_', ' ', gsub('g__', '', sapply(strsplit(ancom.g$result.summary$Feature, "\\|"), function(x){x[6]}))),
                             `N PD`=ancom.g$result.summary$N1, `N NHC`=ancom.g$result.summary$N2,
                             `BC-OA in PD`=ancom.g$result.summary$Mean1, `BC-OA in NHC`=ancom.g$result.summary$Mean2,
                             ancom.g$result.summary[,c('Beta','SE','P','FDR','FC')],
                             `FC lower`=ancom.g$result.summary$FC_lower, `FC upper`=ancom.g$result.summary$FC_upper, check.names=F),
                  by=c('Variable','Kingdom','Phylum','Class','Order','Family','Genus','N PD','N NHC'),
                  suffix=c('_m','_a'), all=T, sort=F)
res.summ <- res.summ[res.summ$Variable=='Case_status',-1]
res.summ <- rbind(data.frame(Kingdom='', Phylum='', Class='', Order='', Family='', Genus='', `N PD`='', `N NHC`='', space_1='',
                             `RA in PD`='MaAsLin2 results', `RA in NHC`='', Beta_m='', SE_m='', P_m='', FDR_m='',
                             FC_m='', `FC lower_m`='', `FC upper_m`='', space_2='',
                             `BC-OA in PD`='ANCOM-BC results', `BC-OA in NHC`='', Beta_a='', SE_a='', P_a='', FDR_a='',
                             FC_a='', `FC lower_a`='', `FC upper_a`='',
                             check.names=F),
                  data.frame(Kingdom='Kingdom', Phylum='Phylum', Class='Class', Order='Order', Family='Famiy', Genus='Genus',
                             `N PD`='N PD', `N NHC`='N NHC', space_1='', `RA in PD`='RA in PD', `RA in NHC`='RA in NHC', Beta_m='Beta',
                             SE_m='SE', P_m='P', FDR_m='FDR', FC_m='FC', `FC lower_m`='FC lower', `FC upper_m`='FC upper', space_2='',
                             `BC-OA in PD`='BC-OA in PD', `BC-OA in NHC`='BC-OA in NHC',
                             Beta_a='Beta', SE_a='SE', P_a='P', FDR_a='FDR', FC_a='FC',
                             `FC lower_a`='FC lower', `FC upper_a`='FC upper', check.names=F),
                  res.summ)
res.summ[3:(nrow(res.summ)-1),c(grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1),
                            grep('ANCOM', res.summ[1,]):ncol(res.summ))][
         is.na(res.summ[3:(nrow(res.summ)-1),c(grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1),
                                           grep('ANCOM', res.summ[1,]):ncol(res.summ))])] <- 'NT'
addWorksheet(wb, 'Genus results')
writeData(wb, 'Genus results', res.summ, keepNA=F, colNames=F)
setColWidths(wb, 'Genus results', cols=1:ncol(res.summ), widths=c(10, rep(22,5), rep(11,2), 2, rep(11,9), 2, rep(11,9))) ### format cells
mergeCells(wb, 'Genus results', cols=grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1), rows=1)
mergeCells(wb, 'Genus results', cols=grep('ANCOM', res.summ[1,]):ncol(res.summ), rows=1)
addStyle(wb, 'Genus results', cols=1:ncol(res.summ), rows=1:2, style=bold, stack=T, gridExpand=T) ### font
addStyle(wb, 'Genus results', cols=1:ncol(res.summ), rows=c(1,3,(nrow(res.summ)+1)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
addStyle(wb, 'Genus results', cols=grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1), rows=2,
         style=horizontal_border_thin, stack=T)
addStyle(wb, 'Genus results', cols=grep('ANCOM', res.summ[1,]):ncol(res.summ), rows=2, style=horizontal_border_thin, stack=T)
convertNum(res.summ, wb, 'Genus results', FALSE) ### convert numbers from strings back to numbers

# output results
saveWorkbook(wb, 'output/3.Taxonomic_associations/MaAsLin2_ANCOMBC_MWAS_PDvsNHC.xlsx',
             overwrite=T)

# print table
res.summ[3:nrow(res.summ),c(10,11,14,15,24,25)] <- apply(res.summ[3:nrow(res.summ),c(10,11,14,15,24,25)],2,
                                                         function(x){formatC(as.numeric(x), format='e', digits=0)})
res.summ[3:nrow(res.summ),c(12,13,16:18,20:23,26:28)] <- apply(res.summ[3:nrow(res.summ),c(12,13,16:18,20:23,26:28)],2,
                                                               function(x){round(as.numeric(x), 2)})
kable(res.summ, col.names=NULL, row.names=F, align='c', caption = 'Genus MWAS results for MaAsLin2 and ANCOM-BC') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% row_spec(1,bold=T,hline_after=F) %>% row_spec(2,bold=T,hline_after=T) %>% scroll_box(height='500px', width='1000px')
```

* Species counts for species that were tested, found significant, and found elevated or reduced for each significant genus from differential abundance analysis were also calculated to observe heterogeneity of genera and their association with PD. This was also done for genera not found significant in differential abunance analysis, but had a significant species in the species differential abundance analysis. See file **3.Taxonomic_associations/Species_counts_for_tested_genera.xlsx** for these results. This data is showcased in **Table 3** of the manuscript.

```{r species_counts, echo=T}
# get names of significant taxa and tested taxa
ancom.fdr.0.05 <- ancom.s$result.summary$Feature[ancom.s$result.summary$Variable == 'Case_status' &
                                                 ancom.s$result.summary$FDR < 0.05 & !is.na(ancom.s$result.summary$FDR)]
ancom.fdr.0.15 <- ancom.s$result.summary$Feature[ancom.s$result.summary$Variable == 'Case_status' &
                                                 ancom.s$result.summary$FDR < 0.15 & !is.na(ancom.s$result.summary$FDR)]
lm.fdr.0.05 <- lm.s$result.summary$Feature[lm.s$result.summary$Variable == 'Case_status' &
                                           lm.s$result.summary$FDR < 0.05 & !is.na(lm.s$result.summary$FDR)]
lm.fdr.0.15 <- lm.s$result.summary$Feature[lm.s$result.summary$Variable == 'Case_status' &
                                           lm.s$result.summary$FDR < 0.15 & !is.na(lm.s$result.summary$FDR)]
sig.species <- unique(c(ancom.fdr.0.05[ancom.fdr.0.05 %in% lm.fdr.0.15], lm.fdr.0.05[lm.fdr.0.05 %in% ancom.fdr.0.15]))

tested.species <- taxa_names(filter_taxa(ra.ps.s, function(x){sum(x>0)>(0.05*nsamples(ra.ps.s))}, TRUE))

ancom.fdr.0.05 <- ancom.g$result.summary$Feature[ancom.g$result.summary$Variable == 'Case_status' &
                                                 ancom.g$result.summary$FDR < 0.05 & !is.na(ancom.g$result.summary$FDR)]
ancom.fdr.0.15 <- ancom.g$result.summary$Feature[ancom.g$result.summary$Variable == 'Case_status' &
                                                 ancom.g$result.summary$FDR < 0.15 & !is.na(ancom.g$result.summary$FDR)]
lm.fdr.0.05 <- lm.g$result.summary$Feature[lm.g$result.summary$Variable == 'Case_status' &
                                           lm.g$result.summary$FDR < 0.05 & !is.na(lm.g$result.summary$FDR)]
lm.fdr.0.15 <- lm.g$result.summary$Feature[lm.g$result.summary$Variable == 'Case_status' &
                                           lm.g$result.summary$FDR < 0.15 & !is.na(lm.g$result.summary$FDR)]
sig.genera <- unique(c(ancom.fdr.0.05[ancom.fdr.0.05 %in% lm.fdr.0.15], lm.fdr.0.05[lm.fdr.0.05 %in% ancom.fdr.0.15]))

tested.genera <- taxa_names(filter_taxa(ra.ps.g, function(x){sum(x>0)>(0.05*nsamples(ra.ps.g))}, TRUE))

# create table giving species count for each significant genus
species.counts <- data.frame()
for (genus in 1:length(tested.genera)){
  genus.name <- strsplit(tested.genera[genus], '\\|')[[1]][6]
  species.n <- length(tested.species[grep(genus.name, tested.species)])
  elev.species <- length(
                  lm.s$result.summary$Feature[lm.s$result.summary$Variable == 'Case_status' &
                                              lm.s$result.summary$Feature %in% sig.species[grep(genus.name, sig.species)] &
                                              lm.s$result.summary$FC > 1])
  red.species <- length(
                 lm.s$result.summary$Feature[lm.s$result.summary$Variable == 'Case_status' &
                                             lm.s$result.summary$Feature %in% sig.species[grep(genus.name, sig.species)] &
                                             lm.s$result.summary$FC < 1])
  if (tested.genera[genus] %in% sig.genera){
    genus.fc <- ifelse(lm.g$result.summary$FC[lm.g$result.summary$Variable == 'Case_status' &
                                              lm.g$result.summary$Feature == tested.genera[genus]] < 1, 'Reduced', 'Elevated')
  }else{
    genus.fc <- 'Missed'
  }
  species.counts <- rbind(species.counts, data.frame(`PD-associated genera`=gsub('_', ' ', gsub('g__', '', genus.name)),
                                                     `N species tested`=species.n,
                                                     `N PD assoc species`=elev.species+red.species,
                                                     `N species elevated`=elev.species,
                                                     `N species reduced`=red.species,
                                                     `Genus level MWAS`=genus.fc,
                                                     check.names=F))
}

# remove genera who were not significant and did not have any significant species
species.counts <- species.counts[species.counts$`N PD assoc species` > 0 |
                                 species.counts$`Genus level MWAS` != 'Missed',]

# sort by genus name and put missed genera at bottom
species.counts <- species.counts[order(species.counts$`PD-associated genera`),]
species.counts <- rbind(species.counts[species.counts$`Genus level MWAS` != 'Missed',],
                        data.frame(`PD-associated genera`='Association at species level, missed at genus level',
                                   `N species tested`='',
                                   `N PD assoc species`='',
                                   `N species elevated`='',
                                   `N species reduced`='',
                                   `Genus level MWAS`='',
                                   check.names=F),
                        species.counts[species.counts$`Genus level MWAS` == 'Missed',])

# write results
wb <- createWorkbook()

addWorksheet(wb, 'Species counts')
writeData(wb, 'Species counts', species.counts, keepNA=T, colNames=T)
setColWidths(wb, 'Species counts', cols=1:ncol(species.counts), widths=c(29, rep(18, (ncol(species.counts)-1)))) ### format cells
mergeCells(wb, 'Species counts', cols=1:ncol(species.counts), rows=36)
addStyle(wb, 'Species counts', cols=1:ncol(species.counts), rows=c(1,36), style=bold, gridExpand=T, stack=T) ### font
addStyle(wb, 'Species counts', cols=1:ncol(species.counts), rows=c(1,2,36,37,(nrow(species.counts)+2)),  ### borders
         style=horizontal_border_med, gridExpand=T, stack=T)
convertNum(species.counts, wb, 'Species counts', TRUE) ### convert numbers from strings back to numbers

saveWorkbook(wb, 'output/3.Taxonomic_associations/Species_counts_for_tested_genera.xlsx', ### write out
             overwrite=T)

# print table
kable(species.counts, col.names = colnames(species.counts),
      row.names=F, align='c', caption = 'Species counts for MWAS significant, and non-significant, genera') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='500px', width='1000px')
```

* Log2 transformed relative abundances and natural log transformed bias-adjusted abundances (from ANCOM-BC) were plotted as boxplots for the 84 PD-associated species to see distribution of the data, along with fold changes from MaAsLin2 and ANCOM-BC. See file **3.Taxonomic_associations/PD_associated_species_figure.pdf** for plot. This is **Supplementary Fig. 3** of the manuscript.

```{r species_plot, echo=T, fig.width=12, fig.height=30}
# grab relative abundances and bias-adjusted abundances of KO groups and pathways
ra.spp <- data.frame(otu_table(ra.ps.s)/100, check.names=F)
ba.spp <- data.frame(otu_table(ancom.s$bias.corrected.ps), check.names=F)

# pull out plotting data for PD-associated species
spp.ra.data <- ra.spp[,colnames(ra.spp) %in% sig.species, FALSE]
spp.ra.fc.data <- lm.s$result.summary[lm.s$result.summary$Feature %in% sig.species &
                                      lm.s$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')]
spp.ra.fc.data <- spp.ra.fc.data[order(sapply(spp.ra.fc.data$FC, function(x){ifelse(x<1,1/x,x)}), decreasing=F),]
spp.ra.data <- spp.ra.data[,rownames(spp.ra.fc.data), FALSE]
spp.ra.fc.data <- data.frame(plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=gsub('_',' ',sapply(strsplit(rownames(spp.ra.fc.data), "\\|s__"), function(x){x[2]})),
                             spp.ra.fc.data)
spp.ra.data <- data.frame(plot='log2(Relative abundances)', Case_status=sample_data(ra.ps.s)$Case_status, spp.ra.data, check.names=F)

spp.ba.data <- ba.spp[,colnames(ba.spp) %in% sig.species, FALSE]
spp.ba.fc.data <- ancom.s$result.summary[ancom.s$result.summary$Feature %in% sig.species &
                                         ancom.s$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')]
spp.ba.fc.data <- spp.ba.fc.data[rownames(spp.ra.fc.data),, FALSE]
spp.ba.data <- spp.ba.data[,rownames(spp.ra.fc.data), FALSE]
spp.ba.fc.data <- data.frame(plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=gsub('_',' ',sapply(strsplit(rownames(spp.ba.fc.data), "\\|s__"), function(x){x[2]})),
                             spp.ba.fc.data)
spp.ba.data <- data.frame(plot='log(Bias-adjusted abundances)', Case_status=sample_data(ra.ps.s)$Case_status, spp.ba.data, check.names=F)

# combine MaAsLin2 and ANCOM-BC data
fc.plot.data <- data.frame(rbind(spp.ra.fc.data, spp.ba.fc.data))
ab.plot.data <- merge(spp.ra.data, spp.ba.data, all=T, sort=F)
colnames(ab.plot.data)[3:ncol(ab.plot.data)] <- gsub('_',' ',
                                                     sapply(strsplit(colnames(ab.plot.data)[3:ncol(ab.plot.data)], "\\|s__"),
                                                            function(x){x[2]}))

# prep fold change data for plotting
fc.plot.data$line <- factor(fc.plot.data$line, levels=rev(unique(fc.plot.data$line)))
fc.plot.data$variable <- factor(fc.plot.data$variable, levels=unique(fc.plot.data$variable))
fc.plot.data$color[fc.plot.data$FC < 1] <- 'elevated'
fc.plot.data$color[fc.plot.data$FC > 1] <- 'depleted'
fc.plot.data$FC_mod[fc.plot.data$FC > 1] <- fc.plot.data$FC[fc.plot.data$FC > 1]-1
fc.plot.data$FC_mod[fc.plot.data$FC < 1] <- -((1/fc.plot.data$FC[fc.plot.data$FC < 1])-1)
fc.plot.data$FC_lower_mod[fc.plot.data$FC_lower > 1] <- fc.plot.data$FC_lower[fc.plot.data$FC_lower > 1]-1
fc.plot.data$FC_lower_mod[fc.plot.data$FC_lower < 1] <- -((1/fc.plot.data$FC_lower[fc.plot.data$FC_lower < 1])-1)
fc.plot.data$FC_upper_mod[fc.plot.data$FC_upper > 1] <- fc.plot.data$FC_upper[fc.plot.data$FC_upper > 1]-1
fc.plot.data$FC_upper_mod[fc.plot.data$FC_upper < 1] <- -((1/fc.plot.data$FC_upper[fc.plot.data$FC_upper < 1])-1)

# prep abundance data for plotting
ab.plot.data$Case_status <- dplyr::recode(ab.plot.data$Case_status, '1' = 'PD', '0' = 'NHC')
ab.plot.data$Case_status <- factor(ab.plot.data$Case_status, levels=rev(unique(ab.plot.data$Case_status)))
ab.plot.data$plot <- factor(ab.plot.data$plot, levels=unique(ab.plot.data$plot))
ab.plot.data.melt <- reshape2::melt(ab.plot.data)
ab.plot.data.melt <- ab.plot.data.melt[!is.na(ab.plot.data.melt$value),]
ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log2(Relative abundances)'] <-
      log2(replace(ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log2(Relative abundances)'],
                   ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log2(Relative abundances)']==0,
                   min(ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log2(Relative abundances)' &
                                               ab.plot.data.melt$value > 0])/2))-20
ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log(Bias-adjusted abundances)'] <-
      ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log(Bias-adjusted abundances)']+20

# merge data
plot.data <- merge(ab.plot.data.melt, fc.plot.data, all=T, sort=F)
plot.data$plot <- factor(plot.data$plot, levels=unique(plot.data$plot))
plot.data$variable <- factor(gsub('Candidatus Methanomassiliicoccus intestinalis',
                                  'Candidatus Methanomassiliicoccus\nintestinalis', plot.data$variable),
                                  levels=gsub('Candidatus Methanomassiliicoccus intestinalis',
                                              'Candidatus Methanomassiliicoccus\nintestinalis', unique(plot.data$variable)))

# plot data
breaks <- c(-40,-35,-30,-25,-20,-15,-10,-5,-2,0,2,5,10,20,25,30,35)
break_labels <- c(paste(breaks[1:5]+20,'\n(',gsub('e\\+00','',gsub('e-0','e-',formatC(2^(breaks[1:5]+20),format='e',digits=0))),')',sep=''),
                  gsub('1x','0x', paste(abs(breaks[6:13])+1, 'x',sep='')),
                  paste(breaks[14:17]-20,'\n(',round(exp(breaks[14:17]-20),1),')',sep=''))

g <- ggplot(data=plot.data[grep('log', plot.data$plot),],
            aes(x=variable, y=value, fill=as.character(Case_status))) +
        geom_boxplot(notch=F, outlier.size = 0.5) +
        geom_errorbar(inherit.aes=F, data=plot.data[plot.data$plot=='Absolute fold change with 95%CI',],
                      aes(x=variable, ymin=FC_lower_mod, ymax=FC_upper_mod, color=color, linetype=line),
                      width=0, position=position_dodge(0.75), size=0.75) +
        geom_point(inherit.aes=F, data=plot.data[plot.data$plot=='Absolute fold change with 95%CI',],
                   aes(x=variable, y=FC_mod, color=color, pch=line), position=position_dodge(0.75), size=1.75) +
        geom_hline(data=plot.data[plot.data$plot=='Absolute fold change with 95%CI',], aes(yintercept=0),
                   size=0.5, linetype='dashed', alpha=0.5) +
        facet_nested(. ~ plot, scales='free', space='free_y', switch='y',
                     strip=strip_nested(text_y=list(element_text(angle=0))),
                     labeller=labeller(group=label_wrap_gen(width=10), sub_group=label_wrap_gen(width=10))) +
        scale_x_discrete(position='bottom') +
        scale_y_continuous(position='right', breaks=breaks, labels=break_labels) +
        coord_flip() +
        scale_fill_manual(values=c("#E69F00", "#00BFC4")) +
        scale_color_manual(values=c("blue", "red"), labels=c("elevated","depleted")) +
        scale_linetype_manual(values=c("11", "solid")) +
        scale_shape_manual(values=c(16, 15)) +
        guides(fill=guide_legend(order=1, title="Subject group", title.position="top"),
               color=guide_legend(order=2, title="Fold change direction", title.position="top"),
               linetype=guide_legend(title="Fold change source", title.position="top", reverse=T),
               pch=guide_legend(title="Fold change source", title.position="top", reverse=T)) +
        theme(legend.position="top", legend.key=element_blank(), legend.title=element_text(size=8), legend.text=element_text(size=8),
              axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.y=element_text(size=8),
              strip.text=element_text(size=8), strip.background=element_rect(fill='gray90', color='gray'), strip.placement="outside",
              panel.spacing.y=unit(0.5, "lines"))
ggsave('output/3.Taxonomic_associations/PD_associated_species_figure.pdf', g, device='pdf', width=12, height=30)

g
```

* To see how PD-species associations are affected when adusting for age and sex and extrinsic PD-associated subject data (variables associated with PD from earlier subject metadata analysis that are exposure variables not intrinsically or biologically related to the disease), re-ran MaAsLin2 adjusting for these variables.
    + Note: data for pain meds and sleep aid were missing for 5 subjects who had their stool samples collected with sterile swabs, therefore, these subjects were excluded from these analyses to control for collection method instead of adjusting for it in the model.
* MaAsLin2 was ran once adjusting for age and sex, and then again adjusting for the 7 potential confounding variables.
* After running the confounding analysis, a table was created tagging which model variables associated with each taxon.
* Results can be found in the file **3.Taxonomic_associations/MaAsLin2_PD_taxa_adj_covariates.xlsx**, and corresponding MaAsLin2 output folder. Results from this file are reported in the manuscript (associatin adjusted for age and sex) and showcased in **Table 2** and **Supplementary Table 3** of the manuscript (confounding analysis).

```{r age_sex_extrinsic_PD_variables, echo=T}
# prep directory for output
system('
if [ ! -d "output/3.Taxonomic_associations/MaAsLin2_PD_taxa_adj_covariates" ]
then
  mkdir output/3.Taxonomic_associations/MaAsLin2_PD_taxa_adj_covariates
  mkdir output/3.Taxonomic_associations/MaAsLin2_PD_taxa_adj_covariates/Sex_Age
  mkdir output/3.Taxonomic_associations/MaAsLin2_PD_taxa_adj_covariates/Confounding_variables
fi
')

# recode categorical variable to get correct effect direction and scale numeric variables
sample_data(ra.ps.s)$Sex <- dplyr::recode(sample_data(ra.ps.s)$Sex, M=1, F=0)
sample_data(ra.ps.s)$Do_you_drink_alcohol <- dplyr::recode(sample_data(ra.ps.s)$Do_you_drink_alcohol, Y=1, N=0)
sample_data(ra.ps.s)$Laxatives <- dplyr::recode(sample_data(ra.ps.s)$Laxatives, Y=1, N=0)
sample_data(ra.ps.s)$Probiotic <- dplyr::recode(sample_data(ra.ps.s)$Probiotic, Y=1, N=0)
sample_data(ra.ps.s)$Pain_med <- dplyr::recode(sample_data(ra.ps.s)$Pain_med, Y=1, N=0)
sample_data(ra.ps.s)$Depression_anxiety_mood_med <- dplyr::recode(sample_data(ra.ps.s)$Depression_anxiety_mood_med, Y=1, N=0)
sample_data(ra.ps.s)$Antihistamines <- dplyr::recode(sample_data(ra.ps.s)$Antihistamines, Y=1, N=0)
sample_data(ra.ps.s)$Sleep_aid <- dplyr::recode(sample_data(ra.ps.s)$Sleep_aid, Y=1, N=0)
sample_data(ra.ps.s)$age_scaled <- scale(sample_data(ra.ps.s)$Age_at_collection)

# perform differential abundance analysis adjusting for sex and age using linear regression with log2 transformed relative abundances
variables <- c('Case_status', 'total_seqs_scaled', 'collection_method', 'Sex', 'age_scaled')

ps <- phyloseq(otu_table(prune_taxa(sig.species, ra.ps.s))/100, sample_data(ra.ps.s))
suppress(
lm.s.adj <- MaAsLin2.plus(ps=ps, metadata=variables,
                      output='output/3.Taxonomic_associations/MaAsLin2_PD_taxa_adj_covariates/Sex_Age',
                      min_abundance=0, min_prevalence=0.05, min_variance=0,
                      normalization='NONE', transform='LOG', analysis_method='LM', max_significance=0.05,
                      random_effects=NULL, fixed_effects=NULL, correction='BH', standardize=F, cores=1,
                      plot_heatmap=F, plot_scatter=F, heatmap_first_n=50, reference=NULL)
)

# write results for sex and age analysis
wb <- createWorkbook()

res.summ <- data.frame(Variable=lm.s.adj$result.summary$Variable,
                       Species=gsub('_', ' ', gsub('s__', '', sapply(strsplit(lm.s.adj$result.summary$Feature, "\\|"), function(x){x[7]}))),
                       lm.s.adj$result.summary[,c('Beta','SE','P','FDR','FC')],
                       `FC lower`=lm.s.adj$result.summary$FC_lower, `FC upper`=lm.s.adj$result.summary$FC_upper, check.names=F)
res.summ <- res.summ[order(res.summ$Species),]
res.summ$Variable <- gsub('Case_status', 'Case status', res.summ$Variable)
res.summ$Variable <- gsub('total_seqs_scaled', 'Total sequence count (standardized)', res.summ$Variable)
res.summ$Variable <- gsub('collection_method', 'Stool collection method', res.summ$Variable)
res.summ$Variable <- gsub('age_scaled', 'Age (standardized)', res.summ$Variable)
addWorksheet(wb, 'Sex age results')
writeData(wb, 'Sex age results', res.summ, keepNA=T, colNames=T)
setColWidths(wb, 'Sex age results', cols=1:ncol(res.summ), widths=c(22, 34, rep(10,7))) ### format cells
addStyle(wb, 'Sex age results', cols=1:ncol(res.summ), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'Sex age results', cols=1:ncol(res.summ), rows=c(1,2,(nrow(res.summ)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)

# print table
res.summ[,3:ncol(res.summ)] <- apply(res.summ[,3:ncol(res.summ)],2,function(x){round(as.numeric(x), 2)})
kable(res.summ, col.names = colnames(res.summ),
      row.names=F, align='c', caption = 'Association analysis with PD-associated species adjusting for sex and age') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='500px', width='1000px')

# perform differential abundance analysis with confounding variables using linear regression with log2 transformed relative abundances
variables <- c('Case_status', 'total_seqs_scaled', 'Do_you_drink_alcohol', 'Laxatives', 'Probiotic',
               'Pain_med', 'Depression_anxiety_mood_med', 'Antihistamines', 'Sleep_aid')

ps <- phyloseq(otu_table(prune_taxa(sig.species, subset_samples(ra.ps.s, collection_method==0)))/100, sample_data(ra.ps.s))
suppress(
lm.s.adj <- MaAsLin2.plus(ps=ps, metadata=variables,
                      output='output/3.Taxonomic_associations/MaAsLin2_PD_taxa_adj_covariates/Confounding_variables',
                      min_abundance=0, min_prevalence=0.05, min_variance=0,
                      normalization='NONE', transform='LOG', analysis_method='LM', max_significance=0.05,
                      random_effects=NULL, fixed_effects=NULL, correction='BH', standardize=F, cores=1,
                      plot_heatmap=F, plot_scatter=F, heatmap_first_n=50, reference=NULL)
)

# make table of what taxa associated with what variable
var.breakdown <- data.frame(Feature=unique(lm.s.adj$result.summary$Feature), `Associated variable`=NA, check.names=F)
for (taxa in 1:nrow(var.breakdown)){
  taxa.name <- var.breakdown$Feature[taxa]
  sig.var <- lm.s.adj$result.summary[lm.s.adj$result.summary$Feature == taxa.name & lm.s.adj$result.summary$FDR < 0.15,]
  if (nrow(sig.var) > 0){
    sig.var.lab <- c()
    for (var in 1:nrow(sig.var)){
      var.res <- sig.var[var,]
      if (!(is.na(var.res$FC))){
        if (var.res$FC > 1 & var.res$FDR < 0.05){sig.var.lab <- c(sig.var.lab, paste(var.res$Variable,'++',sep=''))}
        if (var.res$FC > 1 & var.res$FDR >= 0.05){sig.var.lab <- c(sig.var.lab, paste(var.res$Variable,'+',sep=''))}
        if (var.res$FC < 1 & var.res$FDR < 0.05){sig.var.lab <- c(sig.var.lab, paste(var.res$Variable,'--',sep=''))}
        if (var.res$FC < 1 & var.res$FDR >= 0.05){sig.var.lab <- c(sig.var.lab, paste(var.res$Variable,'-',sep=''))}
      }
    }
    var.breakdown$`Associated variable`[taxa] <- paste(sig.var.lab, collapse=',')
  }
}
var.breakdown$`Associated variable`[is.na(var.breakdown$`Associated variable`)] <- ''

# replace variable names with smaller names
var.breakdown$`Associated variable` <- gsub('Case_status', 'PD', var.breakdown$`Associated variable`)
var.breakdown$`Associated variable` <- gsub('Depression_anxiety_mood_med', 'Mood med', var.breakdown$`Associated variable`)
var.breakdown$`Associated variable` <- gsub('Do_you_drink_alcohol', 'Alcohol', var.breakdown$`Associated variable`)
var.breakdown$`Associated variable` <- gsub('_', ' ', var.breakdown$`Associated variable`)

# write results for confounding analysis
res.summ <- data.frame(Variable=lm.s.adj$result.summary$Variable,
                       Species=gsub('_', ' ', gsub('s__', '', sapply(strsplit(lm.s.adj$result.summary$Feature, "\\|"), function(x){x[7]}))),
                       lm.s.adj$result.summary[,c('Beta','SE','P','FDR','FC')],
                       `FC lower`=lm.s.adj$result.summary$FC_lower, `FC upper`=lm.s.adj$result.summary$FC_upper, check.names=F)
res.summ <- res.summ[order(res.summ$Species),]
res.summ$Variable <- gsub('Case_status', 'Case status', res.summ$Variable)
res.summ$Variable <- gsub('total_seqs_scaled', 'Total sequence count (standardized)', res.summ$Variable)
res.summ$Variable <- gsub('Do_you_drink_alcohol', 'Alcohol', res.summ$Variable)
res.summ$Variable <- gsub('Pain_med', 'Pain medication', res.summ$Variable)
res.summ$Variable <- gsub('Depression_anxiety_mood_med', 'Depression, anxiety, mood medication', res.summ$Variable)
res.summ$Variable <- gsub('Sleep_aid', 'Sleep aid', res.summ$Variable)
addWorksheet(wb, 'Confounder var results')
writeData(wb, 'Confounder var results', res.summ, keepNA=T, colNames=T)
setColWidths(wb, 'Confounder var results', cols=1:ncol(res.summ), widths=c(22, 34, rep(10,7))) ### format cells
addStyle(wb, 'Confounder var results', cols=1:ncol(res.summ), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'Confounder var results', cols=1:ncol(res.summ), rows=c(1,2,(nrow(res.summ)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)

# print table
res.summ[,3:ncol(res.summ)] <- apply(res.summ[,3:ncol(res.summ)],2,function(x){round(as.numeric(x), 2)})
kable(res.summ, col.names = colnames(res.summ),
      row.names=F, align='c', caption = 'Association analysis with PD-associated species and potential confounding variables') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='500px', width='1000px')

# write results for associating variables
res.summ <- data.frame(Species=gsub('_', ' ', gsub('s__', '', sapply(strsplit(var.breakdown$Feature, "\\|"), function(x){x[7]}))),
                       `Associated variable`=var.breakdown$`Associated variable`, check.names=F)
res.summ <- res.summ[order(res.summ$Species),]
addWorksheet(wb, 'Assoc confound var')
writeData(wb, 'Assoc confound var', res.summ, keepNA=T, colNames=T)
setColWidths(wb, 'Assoc confound var', cols=1:ncol(res.summ), widths=c(22, 30)) ### format cells
addStyle(wb, 'Assoc confound var', cols=1:ncol(res.summ), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'Assoc confound var', cols=1:ncol(res.summ), rows=c(1,2,(nrow(res.summ)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)

saveWorkbook(wb, 'output/3.Taxonomic_associations/MaAsLin2_PD_taxa_adj_covariates.xlsx', ### write out
             overwrite=T)

# print table
kable(res.summ, col.names = colnames(res.summ),
      row.names=F, align='c', caption = 'Confounding variables associating with PD-associated species') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='500px', width='1000px')
```

* To test if *Prevotella* species previously classified into the sub-genus "*Prevotella*" in SILVA v 132 (used in previous 16S analysis from *Wallen et al. 2020*) would be signficantly associated with PD as a group in this dataset, collapsed relative abundances of *Prevotella* species in this dataset that were mapped to ASVs in *Prevotella* sub-genus of *Wallen et al. 2020* datasets (*P. buccalis*, *P. timonensis*, *P. bivia*, *P. disiens*, and *P. oralis*) into one group and tested for association with PD using linear regression (via `lm` function) with log2 transformed relative abundances (as done with MaAsLin2). Results can be found in file **3.Taxonomic_associations/Additional_association_analyses**. These results are reported in the main manuscript text.

```{r silva_prevotella, echo=T}
# define target species
target_taxa <- c("Prevotella_buccalis","Prevotella_timonensis","Prevotella_bivia","Prevotella_disiens","Prevotella_oralis")

# collapse species relative abundances into one group
ra.mod <- data.frame(otu_table(ra.ps.s)/100)
colnames(ra.mod) <- sapply(strsplit(as.character(colnames(ra.mod)), "s__"), function(x){x[2]})
ra.mod <- data.frame(target_cluster=rowSums(ra.mod[,colnames(ra.mod) %in% target_taxa]),
                     ra.mod[,!(colnames(ra.mod) %in% target_taxa)])

# log or log2 transform
log2.ra <- data.frame(apply(ra.mod, 2, log2.trans))

# perform linear regression
ra.lm <- lm(log2.ra$target_cluster ~ Case_status + collection_method + total_seqs_scaled,
             data=data.frame(sample_data(ra.ps.s)))

# report results
FC <- paste(round(2^summary(ra.lm)$coefficients[2,1],2), ' [',
            round(2^confint(ra.lm)[2,1],2),'-',round(2^confint(ra.lm)[2,2],2),']',sep='')
mod.results <- data.frame(Grouping="Prevotella (SILVAv132)",
                          `N PD`=sum(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 1] > 0),
                          `N NHC`=sum(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 0] > 0),
                          `RA in PD`=formatC(mean(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 1]),format='e',digits=1),
                          `RA in NHC`=formatC(mean(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 0]),format='e',digits=1),
                          Beta=round(summary(ra.lm)$coefficients[2,1],2),
                          SE=round(summary(ra.lm)$coefficients[2,2],2),
                          P=formatC(summary(ra.lm)$coefficients[2,4],format='e',digits=1),
                          FC=FC,
                          check.names=F)

# print results
wb <- createWorkbook()

addWorksheet(wb, 'Prevotella (SILVAv132)')
writeData(wb, 'Prevotella (SILVAv132)', mod.results, keepNA=T, colNames=T)
setColWidths(wb, 'Prevotella (SILVAv132)', cols=1:ncol(mod.results), widths=c(18, rep(7,7), 13)) ### format cells
addStyle(wb, 'Prevotella (SILVAv132)', cols=1:ncol(mod.results), rows=1:(nrow(mod.results)+1), gridExpand=T, style=center, stack=T)
addStyle(wb, 'Prevotella (SILVAv132)', cols=1:ncol(mod.results), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'Prevotella (SILVAv132)', cols=1:ncol(mod.results), rows=c(1,2,(nrow(mod.results)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
convertNum(mod.results, wb, 'Prevotella (SILVAv132)', TRUE) ### convert numbers from strings back to numbers

saveWorkbook(wb, 'output/3.Taxonomic_associations/Additional_association_analyses.xlsx', ### write out
             overwrite=T)

# print table
kable(mod.results, col.names = colnames(mod.results),
      row.names=F, align='c', caption = 'Association of SILVA v 132 defined Prevotella with PD') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='200px', width='1000px')
```

### Constructing co-occurrence networks

In order to get an inferred ecological picture of the PD and NHC gut microbiome, co-occurence networks were constructed using SparCC. To construct the correlation networks, pairwise correlations and corresponding P-values were calculated using SparCC ([FastSpar](https://github.com/scwatts/fastspar) C++ implementation) on species count data. These correlations were visulized as a network using the GUI program `Gephi`.

* SparCC (`FastSpar`) was performed using 100 iterations (double the default) to get inter-random seed stable correlation calculations with the default `--threshold` parameter of 0.1.
* To calculate p-values for SparCC correlations, the input PD and NHC data were randomly permuted 1,000 times to make 1,000 random datasets. SparCC correlations were then calculated on these random datasets. P-values were calculated by comparing SparCC correlations computed on random datasets to those computed on the real data to see how many instances real data correlations were better than those derived from random.

```{r sparcc, echo=T}
# separate to PD and healthy NHC data
ps.pd.s <- filter_taxa(prune_taxa(taxa_names(abun.ps.s)[grep('UNKNOWN',taxa_names(abun.ps.s),invert=T)],
                                  subset_samples(abun.ps.s, Case_status == 1)),
                       function(x){sum(x > 0) > 0}, TRUE)
ps.hc.s <- filter_taxa(prune_taxa(taxa_names(abun.ps.s)[grep('UNKNOWN',taxa_names(abun.ps.s),invert=T)],
                                  subset_samples(abun.ps.s, Case_status == 0)),
                       function(x){sum(x > 0) > 0}, TRUE)

# format for input to SparCC
pd.s <- data.frame(OTU_id = sapply(strsplit(as.character(taxa_names(ps.pd.s)), "s__"),
                                   function(x){x[2]}),
                   t(otu_table(ps.pd.s)), check.names = F)

hc.s <- data.frame(OTU_id = sapply(strsplit(as.character(taxa_names(ps.hc.s)), "s__"),
                                   function(x){x[2]}),
                   t(otu_table(ps.hc.s)), check.names = F)

# prep directory and output input tables
system('
if [ ! -d "output/4.a.Network_analysis" ]
then
  mkdir output/4.a.Network_analysis
fi
')

write.table(pd.s, 'output/4.a.Network_analysis/Species_SparCC_PD_table.txt',
            row.names = F, quote = F, sep = '\t')

write.table(hc.s, 'output/4.a.Network_analysis/Species_SparCC_HC_table.txt',
            row.names = F, quote = F, sep = '\t')

# calculate SparCC correlations
system('
fastspar --iterations 100 \\
--threads 10 \\
--threshold 0.1 \\
--seed 1234 \\
--otu_table output/4.a.Network_analysis/Species_SparCC_PD_table.txt \\
--correlation output/4.a.Network_analysis/Species_SparCC_PD_Cor_Matrix.txt \\
--covariance output/4.a.Network_analysis/Species_SparCC_PD_Cov_Matrix.txt
')

system('
fastspar --iterations 100 \\
--threads 10 \\
--threshold 0.1 \\
--seed 1234 \\
--otu_table output/4.a.Network_analysis/Species_SparCC_HC_table.txt \\
--correlation output/4.a.Network_analysis/Species_SparCC_HC_Cor_Matrix.txt \\
--covariance output/4.a.Network_analysis/Species_SparCC_HC_Cov_Matrix.txt
')

# create directory for temporary permuted data
system('
if [ ! -d "temp_data" ]
then
  mkdir temp_data
  mkdir temp_data/ErrorOut
  mkdir temp_data/Output
fi
')

# create randomly permuted datasets, calculated correlations, and calculate p-values
system('
fastspar_bootstrap --otu_table output/4.a.Network_analysis/Species_SparCC_PD_table.txt \\
--threads 20 \\
--number 1000 \\
--seed 1234 \\
--prefix temp_data/PD_temp_data
')

system('
fastspar_bootstrap --otu_table output/4.a.Network_analysis/Species_SparCC_HC_table.txt \\
--threads 20 \\
--number 1000 \\
--seed 1234 \\
--prefix temp_data/HC_temp_data
')

system('
echo "#!/bin/bash" > bash_script.sh
echo "#SBATCH --partition=express" >> bash_script.sh
echo "#SBATCH --job-name=SparCC" >> bash_script.sh
echo "#SBATCH --error=temp_data/ErrorOut/SparCC_%A_%a.err" >> bash_script.sh
echo "#SBATCH --output=temp_data/Output/SparCC_%A_%a.out" >> bash_script.sh
echo "#SBATCH --time=2:00:00" >> bash_script.sh
echo "#SBATCH --ntasks=1" >> bash_script.sh
echo "#SBATCH --cpus-per-task=1" >> bash_script.sh
echo "#SBATCH --mem-per-cpu=4000" >> bash_script.sh
echo "#SBATCH --mail-type=FAIL" >> bash_script.sh
echo "#SBATCH --mail-user=wallenz@uab.edu" >> bash_script.sh
echo "#SBATCH --array=0-999" >> bash_script.sh
echo "#SBATCH --wait" >> bash_script.sh
echo " " >> bash_script.sh
echo "source ~/miniconda3/etc/profile.d/conda.sh" >> bash_script.sh
echo "conda activate fastspar" >> bash_script.sh
echo " " >> bash_script.sh
echo "fastspar --iterations 100 \\\\" >> bash_script.sh
echo "--threads 10 \\\\" >> bash_script.sh
echo "--threshold 0.1 \\\\" >> bash_script.sh
echo "--seed 1234 \\\\" >> bash_script.sh
echo "--otu_table temp_data/PD_temp_data_\\${SLURM_ARRAY_TASK_ID}.tsv \\\\" >> bash_script.sh
echo "--correlation temp_data/PD_Cor_Mat_\\${SLURM_ARRAY_TASK_ID}.tsv \\\\" >> bash_script.sh
echo "--covariance temp_data/PD_Cov_Mat_\\${SLURM_ARRAY_TASK_ID}.tsv" >> bash_script.sh
echo " " >> bash_script.sh
echo "fastspar --iterations 100 \\\\" >> bash_script.sh
echo "--threads 10 \\\\" >> bash_script.sh
echo "--threshold 0.1 \\\\" >> bash_script.sh
echo "--seed 1234 \\\\" >> bash_script.sh
echo "--otu_table temp_data/HC_temp_data_\\${SLURM_ARRAY_TASK_ID}.tsv \\\\" >> bash_script.sh
echo "--correlation temp_data/HC_Cor_Mat_\\${SLURM_ARRAY_TASK_ID}.tsv \\\\" >> bash_script.sh
echo "--covariance temp_data/HC_Cov_Mat_\\${SLURM_ARRAY_TASK_ID}.tsv" >> bash_script.sh
')

system('sbatch bash_script.sh')

system('
fastspar_pvalues --threads 20 \\
--otu_table output/4.a.Network_analysis/Species_SparCC_PD_table.txt \\
--correlation output/4.a.Network_analysis/Species_SparCC_PD_Cor_Matrix.txt \\
--prefix temp_data/PD_Cor_Mat_ \\
--permutations 1000 \\
--outfile output/4.a.Network_analysis/Species_SparCC_PD_Pval_Matrix.txt
')

system('
fastspar_pvalues --threads 20 \\
--otu_table output/4.a.Network_analysis/Species_SparCC_HC_table.txt \\
--correlation output/4.a.Network_analysis/Species_SparCC_HC_Cor_Matrix.txt \\
--prefix temp_data/HC_Cor_Mat_ \\
--permutations 1000 \\
--outfile output/4.a.Network_analysis/Species_SparCC_HC_Pval_Matrix.txt
')

# clean up
system('
rm -r temp_data
rm bash_script.sh
')
```

* Once SparCC correlations and p-values were computed, they were formatted into node and edge data.frames and files that could be imported into `igraph` and the visualization program `Gephi`.
* Before importing into `igraph`, SparCC correlations were filtered for species correlations that resulted in a permuted p-value of < 0.05.
* Once imported into `igraph`, edges were further filtered for those with *r* > 0.2, and nodes were then filtered for those who had edges remaining.
* Communities of nodes (species) were then detected using the Louvain algorithm via `cluster_louvain` function in `igraph`.
* Degree for each species was calculated using the `degree` function in `igraph`.
* Node and edge CSV files were outputted from R, and PD and NHC networks were plotted in `Gephi` using the force directed `Force Atlas 2` algorithm to position nodes, coloring first by species cluster memberships (detected with Louvain algorithm) then by PD-associated species differentiating between elevated (blue) vs reduced (red) species.
* SparCC correlations and p-values can be found in directory **4.a.Network_analysis**, and are showcased in **Supplementary Tables 4 and 5** of the manuscript. Results for degree of each species and detected communities of species can be found in the same directory, and are showcased in **Supplementary Table 6** of the manuscript.
* `Gephi` generated files can be found in the directory **4.b.Gephi_network_plots**. This holds network plots that are showcased in **Figure 2** and **Supplementary Fig. 4** of the manuscript.

```{r species_networks, echo=T}
# read in SparCC correlations and pvalues
pd.cor.s <- read.table('output/4.a.Network_analysis/Species_SparCC_PD_Cor_Matrix.txt',
                       row.names = "#OTU ID", header = TRUE, stringsAsFactors = FALSE,
                       check.names = FALSE, comment.char = '', sep = '\t')
pd.pval.s <- read.table('output/4.a.Network_analysis/Species_SparCC_PD_Pval_Matrix.txt',
                        row.names = "#OTU ID", header = TRUE, stringsAsFactors = FALSE,
                        check.names = FALSE, comment.char = '', sep = '\t')

hc.cor.s <- read.table('output/4.a.Network_analysis/Species_SparCC_HC_Cor_Matrix.txt',
                       row.names = "#OTU ID", header = TRUE, stringsAsFactors = FALSE,
                       check.names = FALSE, comment.char = '', sep = '\t')
hc.pval.s <- read.table('output/4.a.Network_analysis/Species_SparCC_HC_Pval_Matrix.txt',
                        row.names = "#OTU ID", header = TRUE, stringsAsFactors = FALSE,
                        check.names = FALSE, comment.char = '', sep = '\t')

# subset and merge result data for case status results
res.s <- merge(ancom.s$result.summary[ancom.s$result.summary$Variable == 'Case_status' &
                                      ancom.s$result.summary$Feature != 'UNKNOWN',],
               lm.s$result.summary[lm.s$result.summary$Variable == 'Case_status',],
               by=c('Variable','Feature','N1','N2'), suffix=c('_ancombc','_maaslin'))
res.s <- res.s[order(res.s$P_ancom, decreasing=T),]

# create node files
nodes.s <- data.frame(Id = sapply(strsplit(as.character(res.s$Feature), "s__"), function(x){x[2]}),
                      Label = sapply(strsplit(as.character(res.s$Feature), "s__"), function(x){x[2]}),
                      ANCOMBC_Beta = res.s$Beta_ancombc,
                      ANCOMBC_FDR = res.s$FDR_ancombc,
                      MaAsLin2_Beta = res.s$Beta_maaslin,
                      MaAsLin2_FDR = res.s$FDR_maaslin,
                      row.names = NULL)
nodes.s$`PD-associated` <- 'No'
nodes.s$`PD-associated`[((nodes.s$ANCOMBC_FDR < 0.05 & nodes.s$MaAsLin2_FDR < 0.15) |
                    (nodes.s$ANCOMBC_FDR < 0.15 & nodes.s$MaAsLin2_FDR < 0.05)) &
                    (nodes.s$ANCOMBC_Beta > 0 & nodes.s$MaAsLin2_Beta > 0)] <- 'Yes_Increased'
nodes.s$`PD-associated`[((nodes.s$ANCOMBC_FDR < 0.05 & nodes.s$MaAsLin2_FDR < 0.15) |
                    (nodes.s$ANCOMBC_FDR < 0.15 & nodes.s$MaAsLin2_FDR < 0.05)) &
                    (nodes.s$ANCOMBC_Beta < 0 & nodes.s$MaAsLin2_Beta < 0)] <- 'Yes_Decreased'
nodes.s$`PD-associated`[is.na(nodes.s$ANCOMBC_FDR) & is.na(nodes.s$MaAsLin2_FDR)] <- 'Not_tested'
nodes.s <- nodes.s[,grep('Beta|FDR', colnames(nodes.s), invert=T)]

# create edge files for PD and NHC
corr.direction <- c()
corr.direction[pd.cor.s[lower.tri(pd.cor.s)] > 0] <- "+"
corr.direction[pd.cor.s[lower.tri(pd.cor.s)] < 0] <- "-"
pd.edges.s <- data.frame(Source = t(combn(rownames(pd.cor.s), 2))[,1],
                         Target = t(combn(rownames(pd.cor.s), 2))[,2],
                         Weight = abs(pd.cor.s[lower.tri(pd.cor.s)]),
			                   `Direction of correlation` = corr.direction,
                         `Correlation P-value` = pd.pval.s[lower.tri(pd.pval.s)],
			                   check.names = F)

corr.direction <- c()
corr.direction[hc.cor.s[lower.tri(hc.cor.s)] > 0] <- "+"
corr.direction[hc.cor.s[lower.tri(hc.cor.s)] < 0] <- "-"
hc.edges.s <- data.frame(Source = t(combn(rownames(hc.cor.s), 2))[,1],
                         Target = t(combn(rownames(hc.cor.s), 2))[,2],
                         Weight = abs(hc.cor.s[lower.tri(hc.cor.s)]),
			                   `Direction of correlation` = corr.direction,
                         `Correlation P-value` = hc.pval.s[lower.tri(hc.pval.s)],
			                   check.names = F)

# tag edges that contains a significant taxon
pd.edges.s$`PD-associated`[pd.edges.s$Source %in% nodes.s$Id[grep('Yes', nodes.s$`PD-associated`)] |
                             pd.edges.s$Target %in% nodes.s$Id[grep('Yes', nodes.s$`PD-associated`)]] <- "Yes"
pd.edges.s$`PD-associated`[is.na(pd.edges.s$`PD-associated`)] <- "No"

hc.edges.s$`PD-associated`[hc.edges.s$Source %in% nodes.s$Id[grep('Yes', nodes.s$`PD-associated`)] |
                             hc.edges.s$Target %in% nodes.s$Id[grep('Yes', nodes.s$`PD-associated`)]] <- "Yes"
hc.edges.s$`PD-associated`[is.na(hc.edges.s$`PD-associated`)] <- "No"

# filter edges for significant correlations (permuted pvalue < 0.05)
pd.edges.s <- pd.edges.s[pd.edges.s$`Correlation P-value` < 0.05,]

hc.edges.s <- hc.edges.s[hc.edges.s$`Correlation P-value` < 0.05,]

# import data into igraph
pd.igraph.s <- graph_from_data_frame(pd.edges.s, directed = FALSE, vertices = nodes.s)
E(pd.igraph.s)$weight <- E(pd.igraph.s)$Weight

hc.igraph.s <- graph_from_data_frame(hc.edges.s, directed = FALSE, vertices = nodes.s)
E(hc.igraph.s)$weight <- E(hc.igraph.s)$Weight

# remove edges with correlations < 0.2
pd.igraph.s <- delete_edges(pd.igraph.s, which(E(pd.igraph.s)$weight < 0.2))

hc.igraph.s <- delete_edges(hc.igraph.s, which(E(hc.igraph.s)$weight < 0.2))

# remove nodes with degree of 0
V(pd.igraph.s)$Degree <- degree(pd.igraph.s, normalized=F)
pd.igraph.s <- delete_vertices(pd.igraph.s, V(pd.igraph.s)$Degree == 0)

V(hc.igraph.s)$Degree <- degree(hc.igraph.s, normalized=F)
hc.igraph.s <- delete_vertices(hc.igraph.s, V(hc.igraph.s)$Degree == 0)

# calculate community membership and modularity of networks
pd.clusters <- cluster_louvain(pd.igraph.s)
V(pd.igraph.s)$Cluster <- pd.clusters$membership

hc.clusters <- cluster_louvain(hc.igraph.s)
V(hc.igraph.s)$Cluster <- hc.clusters$membership

# add degree and community memberships to node files
nodes.s <- merge(nodes.s,
                 data.frame(Id=V(pd.igraph.s)$name,
                            `Degree in PD`=V(pd.igraph.s)$Degree,
                            `Cluster in PD`=V(pd.igraph.s)$Cluster,
                            check.names=F),
                 by='Id', all=T)
nodes.s <- merge(nodes.s,
                 data.frame(Id=V(hc.igraph.s)$name,
                            `Degree in NHC`=V(hc.igraph.s)$Degree,
                            `Cluster in NHC`=V(hc.igraph.s)$Cluster,
                            check.names=F),
                 by='Id', all=T)
nodes.s$`Degree in PD`[is.na(nodes.s$`Degree in PD`)] <- 0
nodes.s$`Degree in NHC`[is.na(nodes.s$`Degree in NHC`)] <- 0
nodes.s$`Cluster in PD`[is.na(nodes.s$`Cluster in PD`)] <- "none"
nodes.s$`Cluster in NHC`[is.na(nodes.s$`Cluster in NHC`)] <- "none"

# output node and edge files
write.csv(nodes.s, "output/4.a.Network_analysis/Species_SparCC_node_file.csv",
          quote=F, row.names=F)
write.csv(pd.edges.s, "output/4.a.Network_analysis/Species_SparCC_PD_edge_file.csv",
          quote=F, row.names=F)
write.csv(hc.edges.s, "output/4.a.Network_analysis/Species_SparCC_HC_edge_file.csv",
          quote=F, row.names=F)
```

**PD network showing algorithm defined clusters and PD-associated species**
```{r PD_network_image, echo=F, out.width="1000px", out.height="1000px"}
knitr::include_graphics(c("output/4.b.Gephi_network_plots/SparCC_PD_Species_w_clusters_labeled.pdf",
                          "output/4.b.Gephi_network_plots/SparCC_PD_Species_w_sig_taxa.pdf"),
                        auto_pdf=T)
```

**NHC network showing algorithm defined clusters and PD-associated species**
```{r NHC_network_image, echo=F, out.width="1000px", out.height="1000px"}
knitr::include_graphics(c("output/4.b.Gephi_network_plots/SparCC_HC_Species_w_clusters_labeled.pdf",
                          "output/4.b.Gephi_network_plots/SparCC_HC_Species_w_sig_taxa.pdf"),
                        auto_pdf=T)
```

* After networks were constructed, we noted a poly-microbial cluster of species (cluster 17 in the PD network) that resembled that of a cluster of opportunitstic pathogens noted in our previous 16S analysis (*Wallen et al. 2020 npj Parkinsons Dis*). As done in *Wallen et al. 2020*, to test if all of the species in this cluster are significantly enriched in PD (since only *P. asaccharolytica* was the only member of this cluster prevalent enough to be tested in MWAS), collpased relative abundances of cluster members (as shown in PD network cluster 17) and tested for association with PD using linear regression with log2 transformed relative abundances (as done in MaAsLin2) adjusting for total sequence count per sample and collection method. Results can be found in file **3.Taxonomic_associations/Additional_association_analyses.xlsx**, and are showcased in **Supplementary Table 7** of the manuscript.

```{r cluster_17, echo=T}
### full cluster ###
# collapse species relative abundances into one group
ra.mod <- data.frame(otu_table(ra.ps.s)/100)
colnames(ra.mod) <- sapply(strsplit(as.character(colnames(ra.mod)), "s__"), function(x){x[2]})
ra.mod <- data.frame(target_cluster=rowSums(ra.mod[,colnames(ra.mod) %in% pd.clusters$names[pd.clusters$membership == 17]]),
                     ra.mod[,!(colnames(ra.mod) %in% pd.clusters$names[pd.clusters$membership == 17])])

# log or log2 transform
log2.ra <- data.frame(apply(ra.mod, 2, log2.trans))

# perform linear regression
ra.lm <- lm(log2.ra$target_cluster ~ Case_status + collection_method + total_seqs_scaled,
             data=data.frame(sample_data(ra.ps.s)))

# report results
FC <- paste(round(2^summary(ra.lm)$coefficients[2,1],2), ' [',
            round(2^confint(ra.lm)[2,1],2),'-',round(2^confint(ra.lm)[2,2],2),']',sep='')
mod.results <- data.frame(Grouping="All 19 species in cluster #17",
                          `N PD`=sum(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 1] > 0),
                          `N NHC`=sum(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 0] > 0),
                          `RA in PD`=formatC(mean(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 1]),format='e',digits=1),
                          `RA in NHC`=formatC(mean(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 0]),format='e',digits=1),
                          Beta=round(summary(ra.lm)$coefficients[2,1],2),
                          SE=round(summary(ra.lm)$coefficients[2,2],2),
                          P=formatC(summary(ra.lm)$coefficients[2,4],format='e',digits=1),
                          FC=FC,
                          check.names=F)

### reduced cluster ###
# collapse species relative abundances into one group
ra.mod <- data.frame(otu_table(ra.ps.s)/100)
colnames(ra.mod) <- sapply(strsplit(as.character(colnames(ra.mod)), "s__"), function(x){x[2]})
ra.mod <- data.frame(target_cluster=rowSums(ra.mod[,(colnames(ra.mod) %in% pd.clusters$names[pd.clusters$membership == 17]) &
                                                    !(colnames(ra.mod) %in% nodes.s$Id[grep("Yes", nodes.s$`PD-associated`)])]),
                     ra.mod[,!(colnames(ra.mod) %in% pd.clusters$names[pd.clusters$membership == 17]) |
                               colnames(ra.mod) %in% nodes.s$Id[grep("Yes", nodes.s$`PD-associated`)]])

# log or log2 transform
log2.ra <- data.frame(apply(ra.mod, 2, log2.trans))

# perform linear regression
ra.lm <- lm(log2.ra$target_cluster ~ Case_status + collection_method + total_seqs_scaled,
             data=data.frame(sample_data(ra.ps.s)))

# report results
FC <- paste(round(2^summary(ra.lm)$coefficients[2,1],2), ' [',
            round(2^confint(ra.lm)[2,1],2),'-',round(2^confint(ra.lm)[2,2],2),']',sep='')
mod.results <- rbind(mod.results,
               data.frame(Grouping="Cluster #17 excluding P. asaccharolytica",
                          `N PD`=sum(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 1] > 0),
                          `N NHC`=sum(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 0] > 0),
                          `RA in PD`=formatC(mean(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 1]),format='e',digits=1),
                          `RA in NHC`=formatC(mean(ra.mod$target_cluster[sample_data(ra.ps.s)$Case_status == 0]),format='e',digits=1),
                          Beta=round(summary(ra.lm)$coefficients[2,1],2),
                          SE=round(summary(ra.lm)$coefficients[2,2],2),
                          P=formatC(summary(ra.lm)$coefficients[2,4],format='e',digits=1),
                          FC=FC,
                          check.names=F))

### P. asaccharolytica ###
# perform linear regression
ra.lm <- lm(log2.ra$Porphyromonas_asaccharolytica ~ Case_status + collection_method + total_seqs_scaled,
            data=data.frame(sample_data(ra.ps.s)))

# report results
FC <- paste(round(2^summary(ra.lm)$coefficients[2,1],2), ' [',
            round(2^confint(ra.lm)[2,1],2),'-',round(2^confint(ra.lm)[2,2],2),']',sep='')
mod.results <- rbind(mod.results,
               data.frame(Grouping="P. asaccharolytica only",
                          `N PD`=sum(ra.mod$Porphyromonas_asaccharolytica[sample_data(ra.ps.s)$Case_status == 1] > 0),
                          `N NHC`=sum(ra.mod$Porphyromonas_asaccharolytica[sample_data(ra.ps.s)$Case_status == 0] > 0),
                          `RA in PD`=formatC(mean(ra.mod$Porphyromonas_asaccharolytica[sample_data(ra.ps.s)$Case_status == 1]),
                                             format='e',digits=1),
                          `RA in NHC`=formatC(mean(ra.mod$Porphyromonas_asaccharolytica[sample_data(ra.ps.s)$Case_status == 0]),
                                              format='e',digits=1),
                          Beta=round(summary(ra.lm)$coefficients[2,1],2),
                          SE=round(summary(ra.lm)$coefficients[2,2],2),
                          P=formatC(summary(ra.lm)$coefficients[2,4],format='e',digits=1),
                          FC=FC,
                          check.names=F))

addWorksheet(wb, 'Cluster 17')
writeData(wb, 'Cluster 17', mod.results, keepNA=T, colNames=T)
setColWidths(wb, 'Cluster 17', cols=1:ncol(mod.results), widths=c(24, rep(7,7), 13)) ### format cells
addStyle(wb, 'Cluster 17', cols=1:ncol(mod.results), rows=1:(nrow(mod.results)+1), gridExpand=T, style=center, stack=T)
addStyle(wb, 'Cluster 17', cols=1:ncol(mod.results), rows=1, style=bold, stack=T) ### font
addStyle(wb, 'Cluster 17', cols=1:ncol(mod.results), rows=c(1,2,(nrow(mod.results)+2)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
convertNum(mod.results, wb, 'Cluster 17', TRUE) ### convert numbers from strings back to numbers

saveWorkbook(wb, 'output/3.Taxonomic_associations/Additional_association_analyses.xlsx', ### write out
             overwrite=T)

# print table
kable(mod.results, col.names = colnames(mod.results),
      row.names=F, align='c', caption = 'Association of PD network cluster #17 with PD') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% scroll_box(height='200px', width='1000px')
```

### Differential abundance analysis of microbial pathways and genes

During processing of the shotgun metagenomic sequences, microbial metabolic pathways (MetaCyc) and gene families (UniProt90) RPK abundances were inferred using HUMAnN. Like taxa, would like to test for pathways and gene families that are enriched or depleted in PD, therefore, differential abundance analysis was performed using MaAsLin2 and ANCOM-BC to test for association of pathway and gene family abundances with PD.

* To reduce the number of gene families being analyzed, default HUMAnN gene families (UniRef90) were converted to KEGG ortholog (KO) groups using the `humann_regroup_table` script specifying the mapping file to be `map_ko_uniref90.txt.gz`. This was performed on gene family tables for each sample.
    + Note: on average per sample, 4-7% of gene families were able to be regrouped into KO groups
    + Per sample KO group abundances were then merged to create one file for all samples

```
$ humann_regroup_table -i ${sample}_genefamilies.tsv -c full_mapping_v201901b/map_ko_uniref90.txt.gz -o ${sample}_humann_KOs.tsv
```

* Pathway and KO group abundance tables were outputted as stratified (contains abundances broken up by contributing species), therefore, merged pathway and KO group abundance files were subsetted for only the community level abundances using the following commands:

```
$ grep -v '|' humann_pathabundance.tsv > humann_pathabundance_unstratified.tsv
$ grep -v '|' humann_KOs.tsv > humann_KOs_unstratified.tsv
```

* Then, more informative names were given to KO groups using the `humann_rename_table` script specifying the naming file to be `map_ko_name.txt.gz`. This was performed on the merged KO table.

```
$ humann_rename_table -i humann_KOs_unstratified.tsv -c full_mapping_v201901b/map_ko_name.txt.gz -o humann_KOs_unstratified_named.tsv
```

* To peform the differential abundance analyses, abundances of detected pathways and KO groups were provided to ANCOM-BC and MaAsLin2. Parameters and model formula used for ANCOM-BC and MaAsLin2 were kept the same as what was used in taxonomic-based analyses, except KO group and pathway abundances were total sum scaled prior to analyzing with MaAsLin2 since abundances are being used as input this time instead of relative abundances. Results for the analyses can be found in result file **5.Gene_pathway_associations/MaAsLin2_ANCOMBC_MWAS_PDvsNHC.xlsx**, and are showcased in **Supplementary Tables 8 and 9** of the manuscript.

```{r DA_genes_pathways, echo=T}
# prep directory for output
system('
if [ ! -d "output/5.Gene_pathway_associations" ]
then
  mkdir output/5.Gene_pathway_associations
  mkdir output/5.Gene_pathway_associations/MaAsLin2_MWAS_PDvsNHC
  mkdir output/5.Gene_pathway_associations/MaAsLin2_MWAS_PDvsNHC/KOs
  mkdir output/5.Gene_pathway_associations/MaAsLin2_MWAS_PDvsNHC/MetaCyc_pathways
fi
')

### GENE FAMILIES ###
# read in tables
gene <- read.table('input/humann_KOs_unstratified_named.tsv', comment.char='',
                   quote='', header=T, stringsAsFactors=F, sep='\t', check.names=F)

# read in metadata
metadata <- data.frame(read_xlsx('input/subject_metadata.xlsx'))
rownames(metadata) <- metadata$UAMRMC_ID

# make table sample x species
rownames(gene) <- gene$`# Gene Family`
gene <- data.frame(t(gene[,-1]), check.names=F)

# modify sample IDs to match metadata
for (i in 1:nrow(gene)){
  rownames(gene)[i] <- strsplit(rownames(gene)[i], '_')[[1]][1]
}
ids.to.mod <- rownames(gene)[grep("^S", rownames(gene))]
for (i in 1:length(ids.to.mod)){
  rownames(gene)[grep(ids.to.mod[i], rownames(gene))] <- metadata$UAMRMC_ID[grep(ids.to.mod[i], metadata$UAB_ID)]
}

# compile count data into phyloseq object
gene.ps <- phyloseq(otu_table(as.matrix(gene), taxa_are_rows=F), sample_data(metadata))

# recode categorical variable to get correct effect direction and scale total sequence count
sample_data(gene.ps)$Case_status <- dplyr::recode(sample_data(gene.ps)$Case_status, PD=1, Control=0)
sample_data(gene.ps)$collection_method <- dplyr::recode(sample_data(gene.ps)$collection_method, swab=1, `OMNIgene GUT`=0)
sample_data(gene.ps)$total_seqs_scaled <- scale(sample_data(gene.ps)$total_sequences)

# perform differential abundance analysis using ANCOM-BC with abundance data
ancom.gene <- ANCOMBC.plus(ps=gene.ps, formula="Case_status + collection_method + total_seqs_scaled",
                           p_adj_method="BH", zero_cut=0.95, lib_cut=0, group=NULL,
                           struc_zero=F, neg_lb=F, tol=1E-5, max_iter=100, conserve=F, alpha=0.05, global=F)

# perform differential abundance analysis using linear regression with log2 transformed relative abundances
suppress(
lm.gene <- MaAsLin2.plus(ps=transform_sample_counts(gene.ps, function(x){x/sum(x)}),
                         output='output/5.Gene_pathway_associations/MaAsLin2_MWAS_PDvsNHC/KOs',
                         metadata=c('Case_status','collection_method','total_seqs_scaled'),
                         min_abundance=0, min_prevalence=0.05, min_variance=0,
                         normalization='NONE', transform='LOG', analysis_method='LM', max_significance=0.05,
                         random_effects=NULL, fixed_effects=NULL, correction='BH', standardize=F, cores=1,
                         plot_heatmap=F, plot_scatter=F, heatmap_first_n=50, reference=NULL)
)

### PATHWAYS ###
# read in tables
path <- read.table('input/humann_pathabundance_unstratified.tsv', comment.char='',
                   quote='', header=T, stringsAsFactors=F, sep='\t', check.names=F)

# read in metadata
metadata <- data.frame(read_xlsx('input/subject_metadata.xlsx'))
rownames(metadata) <- metadata$UAMRMC_ID

# make table sample x species
rownames(path) <- path$`# Pathway`
path <- data.frame(t(path[,-1]), check.names=F)

# modify sample IDs to match metadata
for (i in 1:nrow(path)){
  rownames(path)[i] <- strsplit(rownames(path)[i], '_')[[1]][1]
}
ids.to.mod <- rownames(path)[grep("^S", rownames(path))]
for (i in 1:length(ids.to.mod)){
  rownames(path)[grep(ids.to.mod[i], rownames(path))] <- metadata$UAMRMC_ID[grep(ids.to.mod[i], metadata$UAB_ID)]
}

# compile count data into phyloseq object
path.ps <- phyloseq(otu_table(as.matrix(path), taxa_are_rows=F), sample_data(metadata))

# recode categorical variable to get correct effect direction and scale total sequence count
sample_data(path.ps)$Case_status <- dplyr::recode(sample_data(path.ps)$Case_status, PD=1, Control=0)
sample_data(path.ps)$collection_method <- dplyr::recode(sample_data(path.ps)$collection_method, swab=1, `OMNIgene GUT`=0)
sample_data(path.ps)$total_seqs_scaled <- scale(sample_data(path.ps)$total_sequences)

# perform differential abundance analysis using ANCOM-BC with count data
ancom.path <- ANCOMBC.plus(ps=path.ps, formula="Case_status + collection_method + total_seqs_scaled",
                           p_adj_method="BH", zero_cut=0.95, lib_cut=0, group=NULL,
                           struc_zero=F, neg_lb=F, tol=1E-5, max_iter=100, conserve=F, alpha=0.05, global=F)

# perform differential abundance analysis using linear regression with log2 transformed relative abundances
suppress(
lm.path <- MaAsLin2.plus(ps=transform_sample_counts(path.ps, function(x){x/sum(x)}),
                         output='output/5.Gene_pathway_associations/MaAsLin2_MWAS_PDvsNHC/MetaCyc_pathways',
                         metadata=c('Case_status','collection_method','total_seqs_scaled'),
                         min_abundance=0, min_prevalence=0.05, min_variance=0,
                         normalization='TSS', transform='LOG', analysis_method='LM', max_significance=0.05,
                         random_effects=NULL, fixed_effects=NULL, correction='BH', standardize=F, cores=1,
                         plot_heatmap=F, plot_scatter=F, heatmap_first_n=50, reference=NULL)
)

# initialize workbook
wb <- createWorkbook()

# write results for KO groups
res.summ <- data.frame(Variable=lm.gene$result.summary$Variable,
                       `KEGG ID`=sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}),
                       `KEGG ortholog group`=sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[2]}),
                       `N PD`=lm.gene$result.summary$N1, `N NHC`=lm.gene$result.summary$N2, space_1='',
                       `RA in PD`=lm.gene$result.summary$Mean1, `RA in NHC`=lm.gene$result.summary$Mean2,
                       lm.gene$result.summary[,c('Beta','SE','P','FDR','FC')],
                       `FC lower`=lm.gene$result.summary$FC_lower, `FC upper`=lm.gene$result.summary$FC_upper, space_2='', check.names=F)
res.summ <- merge(res.summ,
                  data.frame(Variable=ancom.gene$result.summary$Variable,
                             `KEGG ID`=sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}),
                             `KEGG ortholog group`=sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[2]}),
                             `N PD`=ancom.gene$result.summary$N1, `N NHC`=ancom.gene$result.summary$N2,
                             `BC-OA in PD`=ancom.gene$result.summary$Mean1, `BC-OA in NHC`=ancom.gene$result.summary$Mean2,
                             ancom.gene$result.summary[,c('Beta','SE','P','FDR','FC')],
                             `FC lower`=ancom.gene$result.summary$FC_lower, `FC upper`=ancom.gene$result.summary$FC_upper, check.names=F),
                  by=c('Variable','KEGG ID','KEGG ortholog group','N PD','N NHC'),
                  suffix=c('_m','_a'), all=T, sort=F)
res.summ <- res.summ[res.summ$Variable=='Case_status',-1]
res.summ <- rbind(data.frame(`KEGG ID`='', `KEGG ortholog group`='', `N PD`='', `N NHC`='', space_1='',
                             `RA in PD`='MaAsLin2 results', `RA in NHC`='', Beta_m='', SE_m='', P_m='', FDR_m='',
                             FC_m='', `FC lower_m`='', `FC upper_m`='', space_2='',
                             `BC-OA in PD`='ANCOM-BC results', `BC-OA in NHC`='', Beta_a='', SE_a='', P_a='', FDR_a='',
                             FC_a='', `FC lower_a`='', `FC upper_a`='',
                             check.names=F),
                  data.frame(`KEGG ID`='KEGG ID', `KEGG ortholog group`='KEGG ortholog group', `N PD`='N PD', `N NHC`='N NHC', space_1='',
                             `RA in PD`='RA in PD', `RA in NHC`='RA in NHC', Beta_m='Beta',
                             SE_m='SE', P_m='P', FDR_m='FDR', FC_m='FC', `FC lower_m`='FC lower', `FC upper_m`='FC upper', space_2='',
                             `BC-OA in PD`='BC-OA in PD', `BC-OA in NHC`='BC-OA in NHC',
                             Beta_a='Beta', SE_a='SE', P_a='P', FDR_a='FDR', FC_a='FC',
                             `FC lower_a`='FC lower', `FC upper_a`='FC upper', check.names=F),
                  res.summ)
res.summ[3:nrow(res.summ),c(grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1),
                            grep('ANCOM', res.summ[1,]):ncol(res.summ))][
         is.na(res.summ[3:nrow(res.summ),c(grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1),
                                           grep('ANCOM', res.summ[1,]):ncol(res.summ))])] <- 'NT'
addWorksheet(wb, 'KO group results')
writeData(wb, 'KO group results', res.summ, keepNA=F, colNames=F)
setColWidths(wb, 'KO group results', cols=1:ncol(res.summ), widths=c(10, 82, rep(11,2), 2, rep(11,9), 2, rep(11,9))) ### format cells
mergeCells(wb, 'KO group results', cols=grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1), rows=1)
mergeCells(wb, 'KO group results', cols=grep('ANCOM', res.summ[1,]):ncol(res.summ), rows=1)
addStyle(wb, 'KO group results', cols=1:ncol(res.summ), rows=1:2, style=bold, stack=T, gridExpand=T) ### font
addStyle(wb, 'KO group results', cols=1:ncol(res.summ), rows=c(1,3,(nrow(res.summ)+1)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
addStyle(wb, 'KO group results', cols=grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1), rows=2,
         style=horizontal_border_thin, stack=T)
addStyle(wb, 'KO group results', cols=grep('ANCOM', res.summ[1,]):ncol(res.summ), rows=2, style=horizontal_border_thin, stack=T)
convertNum(res.summ, wb, 'KO group results', FALSE) ### convert numbers from strings back to numbers

# print table
res.summ[3:nrow(res.summ),c(6,7,10,11,20,21)] <- apply(res.summ[3:nrow(res.summ),c(6,7,10,11,20,21)],2,
                                                       function(x){formatC(as.numeric(x), format='e', digits=0)})
res.summ[3:nrow(res.summ),c(8,9,12:14,16:19,22:24)] <- apply(res.summ[3:nrow(res.summ),c(8,9,12:14,16:19,22:24)],2,
                                                             function(x){round(as.numeric(x), 2)})
kable(res.summ, col.names=NULL, row.names=F, align='c', caption = 'KO group differential abundance results for MaAsLin2 and ANCOM-BC') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% row_spec(1,bold=T,hline_after=F) %>% row_spec(2,bold=T,hline_after=T) %>% scroll_box(height='500px', width='1000px')

# write results for pathways
res.summ <- data.frame(Variable=lm.path$result.summary$Variable,
                       `MetaCyc ID`=sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}),
                       `Pathway`=sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[2]}),
                       `N PD`=lm.path$result.summary$N1, `N NHC`=lm.path$result.summary$N2, space_1='',
                       `RA in PD`=lm.path$result.summary$Mean1, `RA in NHC`=lm.path$result.summary$Mean2,
                       lm.path$result.summary[,c('Beta','SE','P','FDR','FC')],
                       `FC lower`=lm.path$result.summary$FC_lower, `FC upper`=lm.path$result.summary$FC_upper, space_2='', check.names=F)
res.summ <- merge(res.summ,
                  data.frame(Variable=ancom.path$result.summary$Variable,
                             `MetaCyc ID`=sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}),
                             `Pathway`=sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[2]}),
                             `N PD`=ancom.path$result.summary$N1, `N NHC`=ancom.path$result.summary$N2,
                             `BC-OA in PD`=ancom.path$result.summary$Mean1, `BC-OA in NHC`=ancom.path$result.summary$Mean2,
                             ancom.path$result.summary[,c('Beta','SE','P','FDR','FC')],
                             `FC lower`=ancom.path$result.summary$FC_lower, `FC upper`=ancom.path$result.summary$FC_upper, check.names=F),
                  by=c('Variable','MetaCyc ID','Pathway','N PD','N NHC'),
                  suffix=c('_m','_a'), all=T, sort=F)
res.summ <- res.summ[res.summ$Variable=='Case_status',-1]
res.summ <- rbind(data.frame(`MetaCyc ID`='', `Pathway`='', `N PD`='', `N NHC`='', space_1='',
                             `RA in PD`='MaAsLin2 results', `RA in NHC`='', Beta_m='', SE_m='', P_m='', FDR_m='',
                             FC_m='', `FC lower_m`='', `FC upper_m`='', space_2='',
                             `BC-OA in PD`='ANCOM-BC results', `BC-OA in NHC`='', Beta_a='', SE_a='', P_a='', FDR_a='',
                             FC_a='', `FC lower_a`='', `FC upper_a`='',
                             check.names=F),
                  data.frame(`MetaCyc ID`='MetaCyc ID', `Pathway`='Pathway', `N PD`='N PD', `N NHC`='N NHC', space_1='',
                             `RA in PD`='RA in PD', `RA in NHC`='RA in NHC', Beta_m='Beta',
                             SE_m='SE', P_m='P', FDR_m='FDR', FC_m='FC', `FC lower_m`='FC lower', `FC upper_m`='FC upper', space_2='',
                             `BC-OA in PD`='BC-OA in PD', `BC-OA in NHC`='BC-OA in NHC',
                             Beta_a='Beta', SE_a='SE', P_a='P', FDR_a='FDR', FC_a='FC',
                             `FC lower_a`='FC lower', `FC upper_a`='FC upper', check.names=F),
                  res.summ)
res.summ[3:nrow(res.summ),c(grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1),
                            grep('ANCOM', res.summ[1,]):ncol(res.summ))][
         is.na(res.summ[3:nrow(res.summ),c(grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1),
                                           grep('ANCOM', res.summ[1,]):ncol(res.summ))])] <- 'NT'
addWorksheet(wb, 'Pathway results')
writeData(wb, 'Pathway results', res.summ, keepNA=F, colNames=F)
setColWidths(wb, 'Pathway results', cols=1:ncol(res.summ), widths=c(10, 82, rep(11,2), 2, rep(11,9), 2, rep(11,9))) ### format cells
mergeCells(wb, 'Pathway results', cols=grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1), rows=1)
mergeCells(wb, 'Pathway results', cols=grep('ANCOM', res.summ[1,]):ncol(res.summ), rows=1)
addStyle(wb, 'Pathway results', cols=1:ncol(res.summ), rows=1:2, style=bold, stack=T, gridExpand=T) ### font
addStyle(wb, 'Pathway results', cols=1:ncol(res.summ), rows=c(1,3,(nrow(res.summ)+1)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
addStyle(wb, 'Pathway results', cols=grep('MaAsLin2', res.summ[1,]):(grep('space_2', colnames(res.summ))-1), rows=2,
         style=horizontal_border_thin, stack=T)
addStyle(wb, 'Pathway results', cols=grep('ANCOM', res.summ[1,]):ncol(res.summ), rows=2, style=horizontal_border_thin, stack=T)
convertNum(res.summ, wb, 'Pathway results', FALSE) ### convert numbers from strings back to numbers

# output results
saveWorkbook(wb, 'output/5.Gene_pathway_associations/MaAsLin2_ANCOMBC_MWAS_PDvsNHC.xlsx',
             overwrite=T)

# print table
res.summ[3:nrow(res.summ),c(6,7,10,11,20,21)] <- apply(res.summ[3:nrow(res.summ),c(6,7,10,11,20,21)],2,
                                                       function(x){formatC(as.numeric(x), format='e', digits=0)})
res.summ[3:nrow(res.summ),c(8,9,12:14,16:19,22:24)] <- apply(res.summ[3:nrow(res.summ),c(8,9,12:14,16:19,22:24)],2,
                                                             function(x){round(as.numeric(x), 2)})
kable(res.summ, col.names=NULL, row.names=F, align='c', caption = 'Pathway differential abundance results for MaAsLin2 and ANCOM-BC') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% row_spec(1,bold=T,hline_after=F) %>% row_spec(2,bold=T,hline_after=T) %>% scroll_box(height='500px', width='1000px')
```

* To determine if broad reduction in sporulation KO groups is associated with constipation, or not, tested association of these KO groups with constipation in PD and NHC groups by collapsing the relative abundances of differentially abundant sporulation KO groups (FDR < 0.05 in both MaAsLin2 and ANCOM-BC), and testing for association with constipation in PD and NHC subjects separately using linear regression with log2 transformed relative abundances. Results can be found in file **5.Gene_pathway_associations/Additional_association_analyses.xlsx**, and are reported in the manuscript text.

```{r spore_KOs, echo=T}
# recode variable to get correct effect direction
sample_data(gene.ps)$Constipation <- dplyr::recode(sample_data(gene.ps)$Constipation, Y=1, N=0)

# define target KO groups
target_ko <- intersect(ancom.gene$result.summary$Feature[ancom.gene$result.summary$FDR < 0.05 & !is.na(ancom.gene$result.summary$FDR)],
                       lm.gene$result.summary$Feature[lm.gene$result.summary$FDR < 0.05 & !is.na(lm.gene$result.summary$FDR)])
target_ko <- target_ko[grep('sporulation', target_ko)]

# collapse KO group relative abundances into one group
ra.mod <- data.frame(otu_table(transform_sample_counts(gene.ps, function(x){x/sum(x)})), check.names=F)
ra.mod <- data.frame(`Sporulation KOs`=rowSums(ra.mod[,colnames(ra.mod) %in% target_ko]),
                     ra.mod[,!(colnames(ra.mod) %in% target_ko)], check.names=F)

# log or log2 transform
log2.ra <- data.frame(apply(ra.mod, 2, log2.trans), check.names=F)

# begin result data.frame
mod.results <- data.frame(`Subject group`="",
              Beta1="Results for constipation",
              SE1="",
              P1="",
              FC1="",
              Beta2="Results for case status",
              SE2="",
              P2="",
              FC2="",
              check.names=F)
mod.results <- rbind(mod.results,
              data.frame(`Subject group`="Subject group",
              Beta1="Beta",
              SE1="SE",
              P1="P",
              FC1="FC",
              Beta2="Beta",
              SE2="SE",
              P2="P",
              FC2="FC",
              check.names=F))

### PD ###
# subset for only PD
ra.sub <- ra.mod[rownames(ra.mod) %in% sample_names(subset_samples(gene.ps, Case_status == 1)),]
log2.sub <- log2.ra[rownames(log2.ra) %in% sample_names(subset_samples(gene.ps, Case_status == 1)),]
ps.sub <- subset_samples(gene.ps, Case_status == 1)

# perform linear regression
ra.lm <- lm(log2.sub$`Sporulation KOs` ~ Constipation + collection_method + total_seqs_scaled,
            data=data.frame(sample_data(ps.sub)))

# report results
mod.results <- rbind(mod.results,
              data.frame(`Subject group`="PD",
              Beta1=round(summary(ra.lm)$coefficients[2,1],2),
              SE1=round(summary(ra.lm)$coefficients[2,2],2),
              P1=formatC(summary(ra.lm)$coefficients[2,4],format='e',digits=1),
              FC1=round(2^summary(ra.lm)$coefficients[2,1],2),
              Beta2="-",
              SE2="-",
              P2="-",
              FC2="-",
              check.names=F))

### NHC ###
# subset for only NHC
ra.sub <- ra.mod[rownames(ra.mod) %in% sample_names(subset_samples(gene.ps, Case_status == 0)),]
log2.sub <- log2.ra[rownames(log2.ra) %in% sample_names(subset_samples(gene.ps, Case_status == 0)),]
ps.sub <- subset_samples(gene.ps, Case_status == 0)

# perform linear regression
ra.lm <- lm(log2.sub$`Sporulation KOs` ~ Constipation + collection_method + total_seqs_scaled,
            data=data.frame(sample_data(ps.sub)))

# report results
mod.results <- rbind(mod.results,
              data.frame(`Subject group`="NHC",
              Beta1=round(summary(ra.lm)$coefficients[2,1],2),
              SE1=round(summary(ra.lm)$coefficients[2,2],2),
              P1=formatC(summary(ra.lm)$coefficients[2,4],format='e',digits=1),
              FC1=round(2^summary(ra.lm)$coefficients[2,1],2),
              Beta2="-",
              SE2="-",
              P2="-",
              FC2="-",
              check.names=F))

### PD and NHC ###
# perform linear regression
ra.lm <- lm(log2.ra$`Sporulation KOs` ~ Constipation + Case_status + collection_method + total_seqs_scaled,
            data=data.frame(sample_data(gene.ps)))

# report results
mod.results <- rbind(mod.results,
               data.frame(`Subject group`="PD and NHC",
              Beta1=round(summary(ra.lm)$coefficients[2,1],2),
              SE1=round(summary(ra.lm)$coefficients[2,2],2),
              P1=formatC(summary(ra.lm)$coefficients[2,4],format='e',digits=1),
              FC1=round(2^summary(ra.lm)$coefficients[2,1],2),
              Beta2=round(summary(ra.lm)$coefficients[3,1],2),
              SE2=round(summary(ra.lm)$coefficients[3,2],2),
              P2=formatC(summary(ra.lm)$coefficients[3,4],format='e',digits=1),
              FC2=round(2^summary(ra.lm)$coefficients[3,1],2),
              check.names=F))

# print results
wb <- createWorkbook()

addWorksheet(wb, 'Sporulation KO groups')
writeData(wb, 'Sporulation KO groups', mod.results, keepNA=T, colNames=F)
setColWidths(wb, 'Sporulation KO groups', cols=1:ncol(mod.results), widths=c(20, rep(10,4), rep(10,4))) ### format cells
addStyle(wb, 'Sporulation KO groups', cols=1:ncol(mod.results), rows=1:(nrow(mod.results)+1), gridExpand=T, style=center, stack=T)
addStyle(wb, 'Sporulation KO groups', cols=1:ncol(mod.results), rows=1:2, gridExpand=T, style=bold, stack=T) ### font
addStyle(wb, 'Sporulation KO groups', cols=1:ncol(mod.results), rows=c(1,3,(nrow(mod.results)+1)), ### borders
         gridExpand=T, style=horizontal_border_med, stack=T)
convertNum(mod.results, wb, 'Sporulation KO groups', FALSE) ### convert numbers from strings back to numbers

saveWorkbook(wb, 'output/5.Gene_pathway_associations/Additional_association_analyses.xlsx', ### write out
             overwrite=T)

# print table
kable(mod.results, col.names=NULL, row.names=F, align='c', caption = 'Association of sporulation KO groups with constipation and PD') %>%
  kable_styling(font_size=11, bootstrap_options=c("condensed", "responsive")) %>% row_spec(1,bold=T,hline_after=F) %>% row_spec(2,bold=T,hline_after=T) %>% scroll_box(height='200px', width='1000px')
```

* To summarize differential abundance analysis results, plotted the relative abundances and fold changes of select KOs and pathways that were chosen based on relevance to current PD literature. See file **5.Gene_pathway_associations/KO_group_pathway_figure.pdf** for plot. This is **Figure 3** of the manuscript.

```{r ko_pathway_fig, echo=T, fig.width=12, fig.height=20}
# grab relative abundances and bias-adjusted abundances of KO groups and pathways
ra.gene <- ra.mod
ba.gene <- data.frame(otu_table(ancom.gene$bias.corrected.ps), check.names=F)
ra.path <- data.frame(otu_table(transform_sample_counts(path.ps, function(x){x/sum(x)})), check.names=F)
ba.path <- data.frame(otu_table(ancom.path$bias.corrected.ps), check.names=F)

# extract only IDs
colnames(ra.gene) <- sapply(strsplit(colnames(ra.gene), ": "), function(x){x[1]})
colnames(ba.gene) <- sapply(strsplit(colnames(ba.gene), ": "), function(x){x[1]})
colnames(ra.path) <- sapply(strsplit(colnames(ra.path), ": "), function(x){x[1]})
colnames(ba.path) <- sapply(strsplit(colnames(ba.path), ": "), function(x){x[1]})

### pull out plotting data for selected KOs and pathways in order of common categories and proteins ###

## elevated immunogenic bacterial components
# LPS/lipid A
targets <- c('K02535','K09949','K04744','KDO-NAGLIPASYN-PWY','PWY0-881','PWY-6285','ECASYN-PWY')

lps.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
lps.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(lps.ra.fc.data) <- sapply(strsplit(rownames(lps.ra.fc.data), ": "), function(x){x[1]})
lps.ra.fc.data <- lps.ra.fc.data[order(lps.ra.fc.data$FC, decreasing=F),]
lps.ra.data <- lps.ra.data[,rownames(lps.ra.fc.data), FALSE]
lps.ra.fc.data <- data.frame(sub_group='LPS/ lipid A', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(lps.ra.fc.data), lps.ra.fc.data)
lps.ra.data <- data.frame(sub_group='LPS/ lipid A', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, lps.ra.data)

lps.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
lps.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(lps.ba.fc.data) <- sapply(strsplit(rownames(lps.ba.fc.data), ": "), function(x){x[1]})
lps.ba.fc.data <- lps.ba.fc.data[rownames(lps.ra.fc.data),, FALSE]
lps.ba.data <- lps.ba.data[,rownames(lps.ra.fc.data), FALSE]
lps.ba.fc.data <- data.frame(sub_group='LPS/ lipid A', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(lps.ba.fc.data), lps.ba.fc.data)
lps.ba.data <- data.frame(sub_group='LPS/ lipid A', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, lps.ba.data)

# LTA
targets <- c('K19005','K03739')

lta.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
lta.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(lta.ra.fc.data) <- sapply(strsplit(rownames(lta.ra.fc.data), ": "), function(x){x[1]})
lta.ra.fc.data <- lta.ra.fc.data[order(lta.ra.fc.data$FC, decreasing=F),]
lta.ra.data <- lta.ra.data[,rownames(lta.ra.fc.data), FALSE]
lta.ra.fc.data <- data.frame(sub_group='LTA', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(lta.ra.fc.data), lta.ra.fc.data)
lta.ra.data <- data.frame(sub_group='LTA', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, lta.ra.data)

lta.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
lta.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(lta.ba.fc.data) <- sapply(strsplit(rownames(lta.ba.fc.data), ": "), function(x){x[1]})
lta.ba.fc.data <- lta.ba.fc.data[rownames(lta.ra.fc.data),, FALSE]
lta.ba.data <- lta.ba.data[,rownames(lta.ra.fc.data), FALSE]
lta.ba.fc.data <- data.frame(sub_group='LTA', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(lta.ba.fc.data), lta.ba.fc.data)
lta.ba.data <- data.frame(sub_group='LTA', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, lta.ba.data)

# BLP
targets <- c('K06078')

blp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
blp.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(blp.ra.fc.data) <- sapply(strsplit(rownames(blp.ra.fc.data), ": "), function(x){x[1]})
blp.ra.fc.data <- blp.ra.fc.data[order(blp.ra.fc.data$FC, decreasing=F),]
blp.ra.data <- blp.ra.data[,rownames(blp.ra.fc.data), FALSE]
blp.ra.fc.data <- data.frame(sub_group='BLP', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(blp.ra.fc.data), blp.ra.fc.data)
blp.ra.data <- data.frame(sub_group='BLP', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, blp.ra.data)

blp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
blp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(blp.ba.fc.data) <- sapply(strsplit(rownames(blp.ba.fc.data), ": "), function(x){x[1]})
blp.ba.fc.data <- blp.ba.fc.data[rownames(blp.ra.fc.data),, FALSE]
blp.ba.data <- blp.ba.data[,rownames(blp.ra.fc.data), FALSE]
blp.ba.fc.data <- data.frame(sub_group='BLP', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(blp.ba.fc.data), blp.ba.fc.data)
blp.ba.data <- data.frame(sub_group='BLP', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, blp.ba.data)

# combine
fc.plot.data <- data.frame(group='Elevated immunogenic bacterial components', rbind(lps.ra.fc.data, lta.ra.fc.data, blp.ra.fc.data,
                                                                                    lps.ba.fc.data, lta.ba.fc.data, blp.ba.fc.data))
ab.plot.data <- merge(
                 data.frame(group='Elevated immunogenic bacterial components',
                            merge(lps.ra.data, merge(lta.ra.data, blp.ra.data, all=T, sort=F),all=T, sort=F)),
                 data.frame(group='Elevated immunogenic bacterial components',
                            merge(lps.ba.data, merge(lta.ba.data, blp.ba.data, all=T, sort=F),all=T, sort=F)),
                      all=T,sort=F)
colnames(ab.plot.data) <- gsub('\\.','-',colnames(ab.plot.data))

## reduced polysaccharide metabolism and loss of SCFA
targets <- c('K17236','K17234','K16213','K00702','PWY-7456','PWY-6527','PWY-7237',
             'PWY-7242','GALACTUROCAT-PWY','GLUCUROCAT-PWY','GALACT-GLUCUROCAT-PWY')

pbm.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
pbm.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(pbm.ra.fc.data) <- sapply(strsplit(rownames(pbm.ra.fc.data), ": "), function(x){x[1]})
pbm.ra.fc.data <- pbm.ra.fc.data[order(pbm.ra.fc.data$FC, decreasing=T),]
pbm.ra.data <- pbm.ra.data[,rownames(pbm.ra.fc.data), FALSE]
pbm.ra.fc.data <- data.frame(sub_group='---', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(pbm.ra.fc.data), pbm.ra.fc.data)
pbm.ra.data <- data.frame(sub_group='---', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, pbm.ra.data)

pbm.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
pbm.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(pbm.ba.fc.data) <- sapply(strsplit(rownames(pbm.ba.fc.data), ": "), function(x){x[1]})
pbm.ba.fc.data <- pbm.ba.fc.data[rownames(pbm.ra.fc.data),, FALSE]
pbm.ba.data <- pbm.ba.data[,rownames(pbm.ra.fc.data), FALSE]
pbm.ba.fc.data <- data.frame(sub_group='---', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(pbm.ba.fc.data), pbm.ba.fc.data)
pbm.ba.data <- data.frame(sub_group='---', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, pbm.ba.data)

# combine
fc.plot.data <- rbind(fc.plot.data, data.frame(group='Reduced plant-based polysaccharide degradation and SCFA production',
                                               rbind(pbm.ra.fc.data, pbm.ba.fc.data)))
ab.plot.data <- merge(ab.plot.data,
                      merge(data.frame(group='Reduced plant-based polysaccharide degradation and SCFA production', pbm.ra.data),
                            data.frame(group='Reduced plant-based polysaccharide degradation and SCFA production', pbm.ba.data),
                            all=T, sort=F),
                      all=T, sort=F)
colnames(ab.plot.data) <- gsub('\\.','-',colnames(ab.plot.data))

## elevated proteolytic pathways
targets <- c('ORNARGDEG-PWY','ORNDEG-PWY','THREOCAT-PWY')

pdp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
pdp.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(pdp.ra.fc.data) <- sapply(strsplit(rownames(pdp.ra.fc.data), ": "), function(x){x[1]})
pdp.ra.fc.data <- pdp.ra.fc.data[order(pdp.ra.fc.data$FC, decreasing=F),]
pdp.ra.data <- pdp.ra.data[,rownames(pdp.ra.fc.data), FALSE]
pdp.ra.fc.data <- data.frame(sub_group='---', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(pdp.ra.fc.data), pdp.ra.fc.data)
pdp.ra.data <- data.frame(sub_group='---', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, pdp.ra.data)

pdp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
pdp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(pdp.ba.fc.data) <- sapply(strsplit(rownames(pdp.ba.fc.data), ": "), function(x){x[1]})
pdp.ba.fc.data <- pdp.ba.fc.data[rownames(pdp.ra.fc.data),, FALSE]
pdp.ba.data <- pdp.ba.data[,rownames(pdp.ra.fc.data), FALSE]
pdp.ba.fc.data <- data.frame(sub_group='---', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(pdp.ba.fc.data), pdp.ba.fc.data)
pdp.ba.data <- data.frame(sub_group='---', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, pdp.ba.data)

# combine
fc.plot.data <- rbind(fc.plot.data, data.frame(group='Increased protein degradation', rbind(pdp.ra.fc.data, pdp.ba.fc.data)))
ab.plot.data <- merge(ab.plot.data,
                      merge(data.frame(group='Increased protein degradation', pdp.ra.data),
                            data.frame(group='Increased protein degradation', pdp.ba.data),
                            all=T, sort=F),
                      all=T, sort=F)
colnames(ab.plot.data) <- gsub('\\.','-',colnames(ab.plot.data))

## dysregulated neuroactive signaling
# dopamine synthesis
targets <- c('COMPLETE-ARO-PWY')

dsp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
dsp.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(dsp.ra.fc.data) <- sapply(strsplit(rownames(dsp.ra.fc.data), ": "), function(x){x[1]})
dsp.ra.fc.data <- dsp.ra.fc.data[order(dsp.ra.fc.data$FC, decreasing=T),]
dsp.ra.data <- dsp.ra.data[,rownames(dsp.ra.fc.data), FALSE]
dsp.ra.fc.data <- data.frame(sub_group='dopamine synthesis', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(dsp.ra.fc.data), dsp.ra.fc.data)
dsp.ra.data <- data.frame(sub_group='dopamine synthesis', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, dsp.ra.data)

dsp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
dsp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(dsp.ba.fc.data) <- sapply(strsplit(rownames(dsp.ba.fc.data), ": "), function(x){x[1]})
dsp.ba.fc.data <- dsp.ba.fc.data[rownames(dsp.ra.fc.data),, FALSE]
dsp.ba.data <- dsp.ba.data[,rownames(dsp.ra.fc.data), FALSE]
dsp.ba.fc.data <- data.frame(sub_group='dopamine synthesis', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(dsp.ba.fc.data), dsp.ba.fc.data)
dsp.ba.data <- data.frame(sub_group='dopamine synthesis', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, dsp.ba.data)

# glutamate synthesis
targets <- c('K00266','PWY-5505')

gsp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
gsp.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(gsp.ra.fc.data) <- sapply(strsplit(rownames(gsp.ra.fc.data), ": "), function(x){x[1]})
gsp.ra.fc.data <- gsp.ra.fc.data[order(gsp.ra.fc.data$FC, decreasing=T),]
gsp.ra.data <- gsp.ra.data[,rownames(gsp.ra.fc.data), FALSE]
gsp.ra.fc.data <- data.frame(sub_group='glutamate synthesis', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(gsp.ra.fc.data), gsp.ra.fc.data)
gsp.ra.data <- data.frame(sub_group='glutamate synthesis', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, gsp.ra.data)

gsp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
gsp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(gsp.ba.fc.data) <- sapply(strsplit(rownames(gsp.ba.fc.data), ": "), function(x){x[1]})
gsp.ba.fc.data <- gsp.ba.fc.data[rownames(gsp.ra.fc.data),, FALSE]
gsp.ba.data <- gsp.ba.data[,rownames(gsp.ra.fc.data), FALSE]
gsp.ba.fc.data <- data.frame(sub_group='glutamate synthesis', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(gsp.ba.fc.data), gsp.ba.fc.data)
gsp.ba.data <- data.frame(sub_group='glutamate synthesis', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, gsp.ba.data)

# serotonin synthesis
targets <- c('Sporulation KOs','TRPSYN-PWY')

ssp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
ssp.ra.fc.data <- rbind(data.frame(FC=2^(coef(lm(log2.ra$`Sporulation KOs` ~ Case_status + collection_method + total_seqs_scaled,
                                              data=data.frame(sample_data(gene.ps))))[2]),
                                   FC_lower=2^(confint(lm(log2.ra$`Sporulation KOs` ~ Case_status + collection_method + total_seqs_scaled,
                                                          data=data.frame(sample_data(gene.ps))))[2,1]),
                                   FC_upper=2^(confint(lm(log2.ra$`Sporulation KOs` ~ Case_status + collection_method + total_seqs_scaled,
                                                          data=data.frame(sample_data(gene.ps))))[2,2]),
                                   row.names='Sporulation KOs'),
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(ssp.ra.fc.data) <- sapply(strsplit(rownames(ssp.ra.fc.data), ": "), function(x){x[1]})
ssp.ra.fc.data <- ssp.ra.fc.data[order(ssp.ra.fc.data$FC, decreasing=T),]
ssp.ra.data <- ssp.ra.data[,rownames(ssp.ra.fc.data), FALSE]
ssp.ra.fc.data <- data.frame(sub_group='serotonin synthesis', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(ssp.ra.fc.data), ssp.ra.fc.data)
ssp.ra.data <- data.frame(sub_group='serotonin synthesis', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, ssp.ra.data)

ssp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
ssp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(ssp.ba.fc.data) <- sapply(strsplit(rownames(ssp.ba.fc.data), ": "), function(x){x[1]})
#ssp.ba.fc.data <- ssp.ba.fc.data[rownames(ssp.ra.fc.data),, FALSE]
#ssp.ba.data <- ssp.ba.data[,rownames(ssp.ra.fc.data), FALSE]
ssp.ba.fc.data <- data.frame(sub_group='serotonin synthesis', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(ssp.ba.fc.data), ssp.ba.fc.data)
ssp.ba.data <- data.frame(sub_group='serotonin synthesis', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, ssp.ba.data)

# dopamine inhibition
targets <- c('K18933')

dip.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
dip.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(dip.ra.fc.data) <- sapply(strsplit(rownames(dip.ra.fc.data), ": "), function(x){x[1]})
dip.ra.fc.data <- dip.ra.fc.data[order(dip.ra.fc.data$FC, decreasing=F),]
dip.ra.data <- dip.ra.data[,rownames(dip.ra.fc.data), FALSE]
dip.ra.fc.data <- data.frame(sub_group='dopamine inhibition', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(dip.ra.fc.data), dip.ra.fc.data)
dip.ra.data <- data.frame(sub_group='dopamine inhibition', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, dip.ra.data)

dip.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
dip.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(dip.ba.fc.data) <- sapply(strsplit(rownames(dip.ba.fc.data), ": "), function(x){x[1]})
dip.ba.fc.data <- dip.ba.fc.data[rownames(dip.ra.fc.data),, FALSE]
dip.ba.data <- dip.ba.data[,rownames(dip.ra.fc.data), FALSE]
dip.ba.fc.data <- data.frame(sub_group='dopamine inhibition', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(dip.ba.fc.data), dip.ba.fc.data)
dip.ba.data <- data.frame(sub_group='dopamine inhibition', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, dip.ba.data)

# glutamate/GABA degradation
targets <- c('PWY-5088','ARGDEG-PWY')

ggd.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
ggd.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(ggd.ra.fc.data) <- sapply(strsplit(rownames(ggd.ra.fc.data), ": "), function(x){x[1]})
ggd.ra.fc.data <- ggd.ra.fc.data[order(ggd.ra.fc.data$FC, decreasing=F),]
ggd.ra.data <- ggd.ra.data[,rownames(ggd.ra.fc.data), FALSE]
ggd.ra.fc.data <- data.frame(sub_group='glutamate/GABA degradation', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(ggd.ra.fc.data), ggd.ra.fc.data)
ggd.ra.data <- data.frame(sub_group='glutamate/GABA degradation', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, ggd.ra.data)

ggd.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
ggd.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(ggd.ba.fc.data) <- sapply(strsplit(rownames(ggd.ba.fc.data), ": "), function(x){x[1]})
ggd.ba.fc.data <- ggd.ba.fc.data[rownames(ggd.ra.fc.data),, FALSE]
ggd.ba.data <- ggd.ba.data[,rownames(ggd.ra.fc.data), FALSE]
ggd.ba.fc.data <- data.frame(sub_group='glutamate/GABA degradation', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(ggd.ba.fc.data), ggd.ba.fc.data)
ggd.ba.data <- data.frame(sub_group='glutamate/GABA degradation', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, ggd.ba.data)

# combine
fc.plot.data <- rbind(fc.plot.data,
                      data.frame(group='Dysregulated neuroactive signaling',
                                 rbind(dsp.ra.fc.data, gsp.ra.fc.data, ssp.ra.fc.data, dip.ra.fc.data, ggd.ra.fc.data,
                                       dsp.ba.fc.data, gsp.ba.fc.data, ssp.ba.fc.data, dip.ba.fc.data, ggd.ba.fc.data)))
ab.plot.data <- merge(ab.plot.data,
                      merge(
                       data.frame(group='Dysregulated neuroactive signaling',
                                  merge(dsp.ra.data,
                                    merge(gsp.ra.data,
                                      merge(ssp.ra.data,
                                        merge(dip.ra.data, ggd.ra.data,
                                              all=T, sort=F),
                                            all=T, sort=F),
                                          all=T, sort=F),
                                        all=T, sort=F)),
                       data.frame(group='Dysregulated neuroactive signaling',
                                  merge(dsp.ba.data,
                                    merge(gsp.ba.data,
                                      merge(ssp.ba.data,
                                        merge(dip.ba.data, ggd.ba.data,
                                              all=T, sort=F),
                                            all=T, sort=F),
                                          all=T, sort=F),
                                        all=T, sort=F)),
                            all=T, sort=F),
                       all=T, sort=F)
colnames(ab.plot.data) <- gsub('\\.','-',colnames(ab.plot.data))
colnames(ab.plot.data) <- gsub('Sporulation-KOs','Sporulation KOs',colnames(ab.plot.data))

## reduced neuroprotective molecules
# nicotinamide degradation
targets <- c('K08281')

ndp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
ndp.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(ndp.ra.fc.data) <- sapply(strsplit(rownames(ndp.ra.fc.data), ": "), function(x){x[1]})
ndp.ra.fc.data <- ndp.ra.fc.data[order(ndp.ra.fc.data$FC, decreasing=F),]
ndp.ra.data <- ndp.ra.data[,rownames(ndp.ra.fc.data), FALSE]
ndp.ra.fc.data <- data.frame(sub_group='nicotinamide degradation', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(ndp.ra.fc.data), ndp.ra.fc.data)
ndp.ra.data <- data.frame(sub_group='nicotinamide degradation', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, ndp.ra.data)

ndp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
ndp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(ndp.ba.fc.data) <- sapply(strsplit(rownames(ndp.ba.fc.data), ": "), function(x){x[1]})
ndp.ba.fc.data <- ndp.ba.fc.data[rownames(ndp.ra.fc.data),, FALSE]
ndp.ba.data <- ndp.ba.data[,rownames(ndp.ra.fc.data), FALSE]
ndp.ba.fc.data <- data.frame(sub_group='nicotinamide degradation', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(ndp.ba.fc.data), ndp.ba.fc.data)
ndp.ba.data <- data.frame(sub_group='nicotinamide degradation', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, ndp.ba.data)

# trehalose degradation
targets <- c('K00697','K01087','PWY-2723')

tdp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
tdp.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(tdp.ra.fc.data) <- sapply(strsplit(rownames(tdp.ra.fc.data), ": "), function(x){x[1]})
tdp.ra.fc.data <- tdp.ra.fc.data[order(tdp.ra.fc.data$FC, decreasing=F),]
tdp.ra.data <- tdp.ra.data[,rownames(tdp.ra.fc.data), FALSE]
tdp.ra.fc.data <- data.frame(sub_group='trehalose degradation and metabolism', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(tdp.ra.fc.data), tdp.ra.fc.data)
tdp.ra.data <- data.frame(sub_group='trehalose degradation and metabolism', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, tdp.ra.data)

tdp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
tdp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(tdp.ba.fc.data) <- sapply(strsplit(rownames(tdp.ba.fc.data), ": "), function(x){x[1]})
tdp.ba.fc.data <- tdp.ba.fc.data[rownames(tdp.ra.fc.data),, FALSE]
tdp.ba.data <- tdp.ba.data[,rownames(tdp.ra.fc.data), FALSE]
tdp.ba.fc.data <- data.frame(sub_group='trehalose degradation and metabolism', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(tdp.ba.fc.data), tdp.ba.fc.data)
tdp.ba.data <- data.frame(sub_group='trehalose degradation and metabolism', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, tdp.ba.data)

# combine
fc.plot.data <- rbind(fc.plot.data,
                      data.frame(group='Reduced neuroprotective molecules', rbind(ndp.ra.fc.data, tdp.ra.fc.data,
                                                                                  ndp.ba.fc.data, tdp.ba.fc.data)))
ab.plot.data <- merge(ab.plot.data,
                      merge(data.frame(group='Reduced neuroprotective molecules', merge(ndp.ra.data, tdp.ra.data, all=T, sort=F)),
                            data.frame(group='Reduced neuroprotective molecules', merge(ndp.ba.data, tdp.ba.data, all=T, sort=F)),
                            all=T, sort=F),
                      all=T, sort=F)
colnames(ab.plot.data) <- gsub('\\.','-',colnames(ab.plot.data))

## elevated bacterial amyloid curli
# curli production
targets <- c('K04334','K04336','K06214')

cpp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
cpp.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(cpp.ra.fc.data) <- sapply(strsplit(rownames(cpp.ra.fc.data), ": "), function(x){x[1]})
cpp.ra.fc.data <- cpp.ra.fc.data[order(cpp.ra.fc.data$FC, decreasing=F),]
cpp.ra.data <- cpp.ra.data[,rownames(cpp.ra.fc.data), FALSE]
cpp.ra.fc.data <- data.frame(sub_group='curli production', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(cpp.ra.fc.data), cpp.ra.fc.data)
cpp.ra.data <- data.frame(sub_group='curli production', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, cpp.ra.data)

cpp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
cpp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(cpp.ba.fc.data) <- sapply(strsplit(rownames(cpp.ba.fc.data), ": "), function(x){x[1]})
cpp.ba.fc.data <- cpp.ba.fc.data[rownames(cpp.ra.fc.data),, FALSE]
cpp.ba.data <- cpp.ba.data[,rownames(cpp.ra.fc.data), FALSE]
cpp.ba.fc.data <- data.frame(sub_group='curli production', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(cpp.ba.fc.data), cpp.ba.fc.data)
cpp.ba.data <- data.frame(sub_group='curli production', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, cpp.ba.data)

# curli regulation
targets <- c('K21963')

crp.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
crp.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(crp.ra.fc.data) <- sapply(strsplit(rownames(crp.ra.fc.data), ": "), function(x){x[1]})
crp.ra.fc.data <- crp.ra.fc.data[order(crp.ra.fc.data$FC, decreasing=F),]
crp.ra.data <- crp.ra.data[,rownames(crp.ra.fc.data), FALSE]
crp.ra.fc.data <- data.frame(sub_group='curli regulation', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(crp.ra.fc.data), crp.ra.fc.data)
crp.ra.data <- data.frame(sub_group='curli regulation', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, crp.ra.data)

crp.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
crp.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(crp.ba.fc.data) <- sapply(strsplit(rownames(crp.ba.fc.data), ": "), function(x){x[1]})
crp.ba.fc.data <- crp.ba.fc.data[rownames(crp.ra.fc.data),, FALSE]
crp.ba.data <- crp.ba.data[,rownames(crp.ra.fc.data), FALSE]
crp.ba.fc.data <- data.frame(sub_group='curli regulation', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(crp.ba.fc.data), crp.ba.fc.data)
crp.ba.data <- data.frame(sub_group='curli regulation', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, crp.ba.data)

# combine
fc.plot.data <- rbind(fc.plot.data,
                      data.frame(group='Elevated curli, a bacterial amyloid, triggers alpha-synuclein pathology',
                                 rbind(cpp.ra.fc.data, crp.ra.fc.data, cpp.ba.fc.data, crp.ba.fc.data)))
ab.plot.data <- merge(ab.plot.data,
                      merge(data.frame(group='Elevated curli, a bacterial amyloid, triggers alpha-synuclein pathology',
                                       merge(cpp.ra.data, crp.ra.data, all=T, sort=F)),
                            data.frame(group='Elevated curli, a bacterial amyloid, triggers alpha-synuclein pathology',
                                       merge(cpp.ba.data, crp.ba.data, all=T, sort=F)),
                            all=T, sort=F),
                      all=T, sort=F)
colnames(ab.plot.data) <- gsub('\\.','-',colnames(ab.plot.data))

## elevated toxic metabolite
# TMA from choline
targets <- c('K20038')

tch.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
tch.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(tch.ra.fc.data) <- sapply(strsplit(rownames(tch.ra.fc.data), ": "), function(x){x[1]})
tch.ra.fc.data <- tch.ra.fc.data[order(tch.ra.fc.data$FC, decreasing=F),]
tch.ra.data <- tch.ra.data[,rownames(tch.ra.fc.data), FALSE]
tch.ra.fc.data <- data.frame(sub_group='TMA from choline', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(tch.ra.fc.data), tch.ra.fc.data)
tch.ra.data <- data.frame(sub_group='TMA from choline', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, tch.ra.data)

tch.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
tch.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(tch.ba.fc.data) <- sapply(strsplit(rownames(tch.ba.fc.data), ": "), function(x){x[1]})
tch.ba.fc.data <- tch.ba.fc.data[rownames(tch.ra.fc.data),, FALSE]
tch.ba.data <- tch.ba.data[,rownames(tch.ra.fc.data), FALSE]
tch.ba.fc.data <- data.frame(sub_group='TMA from choline', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(tch.ba.fc.data), tch.ba.fc.data)
tch.ba.data <- data.frame(sub_group='TMA from choline', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, tch.ba.data)

# TMA from carnitine
targets <- c('K05245')

tca.ra.data <- cbind(ra.gene[,colnames(ra.gene) %in% targets, FALSE], ra.path[,colnames(ra.path) %in% targets, FALSE])
tca.ra.fc.data <- rbind(lm.gene$result.summary[sapply(strsplit(lm.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        lm.path$result.summary[sapply(strsplit(lm.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               lm.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(tca.ra.fc.data) <- sapply(strsplit(rownames(tca.ra.fc.data), ": "), function(x){x[1]})
tca.ra.fc.data <- tca.ra.fc.data[order(tca.ra.fc.data$FC, decreasing=F),]
tca.ra.data <- tca.ra.data[,rownames(tca.ra.fc.data), FALSE]
tca.ra.fc.data <- data.frame(sub_group='TMA from carnitine', plot='Absolute fold change with 95%CI', line='MaAsLin2',
                             variable=rownames(tca.ra.fc.data), tca.ra.fc.data)
tca.ra.data <- data.frame(sub_group='TMA from carnitine', plot='log2(Relative abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, tca.ra.data)

tca.ba.data <- cbind(ba.gene[,colnames(ba.gene) %in% targets, FALSE], ba.path[,colnames(ba.path) %in% targets, FALSE])
tca.ba.fc.data <- rbind(ancom.gene$result.summary[sapply(strsplit(ancom.gene$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.gene$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')],
                        ancom.path$result.summary[sapply(strsplit(ancom.path$result.summary$Feature, ": "), function(x){x[1]}) %in% targets &
                                               ancom.path$result.summary$Variable == 'Case_status',c('FC','FC_lower','FC_upper')])
rownames(tca.ba.fc.data) <- sapply(strsplit(rownames(tca.ba.fc.data), ": "), function(x){x[1]})
tca.ba.fc.data <- tca.ba.fc.data[rownames(tca.ra.fc.data),, FALSE]
tca.ba.data <- tca.ba.data[,rownames(tca.ra.fc.data), FALSE]
tca.ba.fc.data <- data.frame(sub_group='TMA from carnitine', plot='Absolute fold change with 95%CI', line='ANCOM-BC',
                             variable=rownames(tca.ba.fc.data), tca.ba.fc.data)
tca.ba.data <- data.frame(sub_group='TMA from carnitine', plot='log(Bias-adjusted abundances)',
                          Case_status=sample_data(gene.ps)$Case_status, tca.ba.data)

# combine
fc.plot.data <- rbind(fc.plot.data,
                      data.frame(group='Elevated toxic metabolite', rbind(tch.ra.fc.data, tca.ra.fc.data,
                                                                          tch.ba.fc.data, tca.ba.fc.data)))
ab.plot.data <- merge(ab.plot.data,
                      merge(data.frame(group='Elevated toxic metabolite', merge(tch.ra.data, tca.ra.data, all=T, sort=F)),
                            data.frame(group='Elevated toxic metabolite', merge(tch.ba.data, tca.ba.data, all=T, sort=F)),
                            all=T, sort=F),
                      all=T, sort=F)
colnames(ab.plot.data) <- gsub('\\.','-',colnames(ab.plot.data))

# prep fold change data for plotting
fc.plot.data$group <- factor(fc.plot.data$group, levels=unique(fc.plot.data$group))
fc.plot.data$sub_group <- factor(fc.plot.data$sub_group, levels=unique(fc.plot.data$sub_group))
fc.plot.data$line <- factor(fc.plot.data$line, levels=rev(unique(fc.plot.data$line)))
fc.plot.data$variable <- factor(fc.plot.data$variable, levels=unique(fc.plot.data$variable))
fc.plot.data$color[fc.plot.data$FC < 1] <- 'elevated'
fc.plot.data$color[fc.plot.data$FC > 1] <- 'depleted'
fc.plot.data$FC_mod[fc.plot.data$FC > 1] <- fc.plot.data$FC[fc.plot.data$FC > 1]-1
fc.plot.data$FC_mod[fc.plot.data$FC < 1] <- -((1/fc.plot.data$FC[fc.plot.data$FC < 1])-1)
fc.plot.data$FC_lower_mod[fc.plot.data$FC_lower > 1] <- fc.plot.data$FC_lower[fc.plot.data$FC_lower > 1]-1
fc.plot.data$FC_lower_mod[fc.plot.data$FC_lower < 1] <- -((1/fc.plot.data$FC_lower[fc.plot.data$FC_lower < 1])-1)
fc.plot.data$FC_upper_mod[fc.plot.data$FC_upper > 1] <- fc.plot.data$FC_upper[fc.plot.data$FC_upper > 1]-1
fc.plot.data$FC_upper_mod[fc.plot.data$FC_upper < 1] <- -((1/fc.plot.data$FC_upper[fc.plot.data$FC_upper < 1])-1)

# prep abundance data for plotting
ab.plot.data$group <- factor(ab.plot.data$group, levels=unique(ab.plot.data$group))
ab.plot.data$sub_group <- factor(ab.plot.data$sub_group, levels=unique(ab.plot.data$sub_group))
ab.plot.data$Case_status <- dplyr::recode(ab.plot.data$Case_status, '1' = 'PD', '0' = 'NHC')
ab.plot.data$Case_status <- factor(ab.plot.data$Case_status, levels=rev(unique(ab.plot.data$Case_status)))
ab.plot.data$plot <- factor(ab.plot.data$plot, levels=unique(ab.plot.data$plot))
ab.plot.data.melt <- reshape2::melt(ab.plot.data)
ab.plot.data.melt <- ab.plot.data.melt[!is.na(ab.plot.data.melt$value),]
ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log2(Relative abundances)'] <-
      log2(replace(ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log2(Relative abundances)'],
                   ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log2(Relative abundances)']==0,
                   min(ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log2(Relative abundances)' &
                                               ab.plot.data.melt$value > 0])/2))
ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log(Bias-adjusted abundances)'] <-
      ab.plot.data.melt$value[ab.plot.data.melt$plot == 'log(Bias-adjusted abundances)']+10

# merge data
plot.data <- merge(ab.plot.data.melt, fc.plot.data, all=T, sort=F)
plot.data$plot <- factor(plot.data$plot, levels=unique(plot.data$plot))

# add gene annotations to KOs
plot.data$variable <- factor(gsub('K02535', 'K02535 (lpxC)', plot.data$variable),
                             levels=gsub('K02535', 'K02535 (lpxC)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K04744', 'K04744 (lptD, imp, ostA)', plot.data$variable),
                             levels=gsub('K04744', 'K04744 (lptD, imp, ostA)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K09949', 'K09949 (lpxI)', plot.data$variable),
                             levels=gsub('K09949', 'K09949 (lpxI)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K19005', 'K19005 (ltaS)', plot.data$variable),
                             levels=gsub('K19005', 'K19005 (ltaS)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K03739', 'K03739 (dltB)', plot.data$variable),
                             levels=gsub('K03739', 'K03739 (dltB)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K06078', 'K06078 (lpp)', plot.data$variable),
                             levels=gsub('K06078', 'K06078 (lpp)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K17236', 'K17236 (araQ)', plot.data$variable),
                             levels=gsub('K17236', 'K17236 (araQ)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K16213', 'K16213 (cbe, mbe)', plot.data$variable),
                             levels=gsub('K16213', 'K16213 (cbe, mbe)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K17234', 'K17234 (araN)', plot.data$variable),
                             levels=gsub('K17234', 'K17234 (araN)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K00702', 'K00702 (E2.4.1.20)', plot.data$variable),
                             levels=gsub('K00702', 'K00702 (E2.4.1.20)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K00266', 'K00266 (gltD)', plot.data$variable),
                             levels=gsub('K00266', 'K00266 (gltD)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K18933', 'K18933 (mfnA, adc)', plot.data$variable),
                             levels=gsub('K18933', 'K18933 (mfnA, adc)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K08281', 'K08281 (pncA)', plot.data$variable),
                             levels=gsub('K08281', 'K08281 (pncA)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K01087', 'K01087 (otsB)', plot.data$variable),
                             levels=gsub('K01087', 'K01087 (otsB)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K00697', 'K00697 (otsA)', plot.data$variable),
                             levels=gsub('K00697', 'K00697 (otsA)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K06214', 'K06214 (csgG)', plot.data$variable),
                             levels=gsub('K06214', 'K06214 (csgG)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K04336', 'K04336 (csgC)', plot.data$variable),
                             levels=gsub('K04336', 'K04336 (csgC)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K04334', 'K04334 (csgA)', plot.data$variable),
                             levels=gsub('K04334', 'K04334 (csgA)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K21963', 'K21963 (ecpR, matA)', plot.data$variable),
                             levels=gsub('K21963', 'K21963 (ecpR, matA)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K20038', 'K20038 (cutC)', plot.data$variable),
                             levels=gsub('K20038', 'K20038 (cutC)', unique(plot.data$variable)))
plot.data$variable <- factor(gsub('K05245', 'K05245 (caiT)', plot.data$variable),
                             levels=gsub('K05245', 'K05245 (caiT)', unique(plot.data$variable)))

# plot data
breaks <- c(-25,-20,-15,-10,-3,-2,-1,0,1,2,7.5,10,15,20)
break_labels <- c(paste(breaks[1:4],'\n(',gsub('e-0','e-',formatC(2^breaks[1:4],format='e',digits=0)),')',sep=''),
                  paste(gsub('1','0', abs(breaks[5:10])+1), 'x',sep=''),
                  paste(breaks[11:14]-10,'\n(',round(exp(breaks[11:14]-10),1),')',sep=''))

g <- ggplot(data=plot.data[grep('log', plot.data$plot),],
            aes(x=variable, y=value, fill=as.character(Case_status))) +
        geom_boxplot(notch=F, outlier.size = 0.5) +
        geom_errorbar(inherit.aes=F, data=plot.data[plot.data$plot=='Absolute fold change with 95%CI',],
                      aes(x=variable, ymin=FC_lower_mod, ymax=FC_upper_mod, color=color, linetype=line),
                      width=0, position=position_dodge(0.75), size=0.75) +
        geom_point(inherit.aes=F, data=plot.data[plot.data$plot=='Absolute fold change with 95%CI',],
                   aes(x=variable, y=FC_mod, color=color, pch=line), position=position_dodge(0.75), size=1.75) +
        geom_hline(data=plot.data[plot.data$plot=='Absolute fold change with 95%CI',], aes(yintercept=0),
                   size=0.5, linetype='dashed', alpha=0.5) +
        facet_nested(group + sub_group ~ plot, scales='free', space='free_y', switch='y',
                     strip=strip_nested(text_y=list(element_text(angle=0))),
                     labeller=labeller(group=label_wrap_gen(width=10), sub_group=label_wrap_gen(width=10))) +
        scale_x_discrete(position='bottom') +
        scale_y_continuous(position='right', breaks=breaks, labels=break_labels) +
        coord_flip() +
        scale_fill_manual(values=c("#E69F00", "#00BFC4")) +
        scale_color_manual(values=c("blue", "red"), labels=c("elevated","depleted")) +
        scale_linetype_manual(values=c("11", "solid")) +
        scale_shape_manual(values=c(16, 15)) +
        guides(fill=guide_legend(order=1, title="Subject group", title.position="top"),
               color=guide_legend(order=2, title="Fold change direction", title.position="top"),
               linetype=guide_legend(title="Fold change source", title.position="top", reverse=T),
               pch=guide_legend(title="Fold change source", title.position="top", reverse=T)) +
        theme(legend.position="top", legend.key=element_blank(), legend.title=element_text(size=8), legend.text=element_text(size=8),
              axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.y=element_text(size=8),
              strip.text=element_text(size=8), strip.background=element_rect(fill='gray90', color='gray'), strip.placement="outside",
              panel.spacing.y=unit(0.5, "lines"))
ggsave('output/5.Gene_pathway_associations/KO_group_pathway_figure.pdf', g, device='pdf', width=12, height=20)

g
```
